---
title: "An R interface to COMPASS, the gene expression compendia GraphQL endpoint"
author: Paolo Sonego
affiliation: https://github.com/onertipaday/rcompass
package: rcompass
abstract: > 
  How to query COMPASS from R
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{An R interface to COMPASS, the gene expression compendia GraphQL endpoint}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteKeywords{Bioinformatics, Gene Expression, data}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```
# Introduction

## Installation

`rcompass` installation from github requires the devtools package to be installed.

```{r install, eval=FALSE}
if(!requireNamespace("devtools")) install.packages("devtools")
devtools::install_github("onertipaday/rcompass")
```

## Getting help

To get help open an issue on the `rcompass`
[github page](https://github.com/onertipaday/rcompass/issues).

## The resource

[COMPASS](http://compass.fmach.it/graphql/) GraphQL Endpoint
[COMPASS documentation](https://compass-.readthedocs.io/en/latest/index.html)

## The package

COMPASS (COMpendia Programmatic Access Support Software) is a software layer that provides a GraphQL endpoint to query compendia built using COMMAND>_ technology [1](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6348648/). 
COMPASS is meant to be the barebone interface on top of a compendium database from a
single location with a unified output format. The `rcompass`
package make this possible from within R. Do do so, it relies on the
`httr` package to query the GraphQL interface, and access
and retrieve data.

# A Brief rCOMPASS overview

The `rcompass` package is build around a bunch of functions
enable to query the compendium of interest and retrieve, store and manipulate data. Each of
these functions are described in more details in their respective manual
pages. We start by loading the package.

```{r setup}
library(rcompass)
```

# Case Study

Here we illustrate one of the most common usages of rCOMPASS: how to create a module from a set of known genes and automatically selecting condition contrasts. To demonstrate all this we will focus on a set of genes regulated by the phytohormone abscisic acid (ABA) in pre-v√©raison berries in Vitis vinifera.

## Step 1 - create the initial module

We start by creating a module with these 55 ABA modulated genes and refining it by selecting the most co-expressed genes. The module will be manually split to distinguish different condition contrasts, corresponding to different biological contexts, in which ABA affects gene expression. Finally, we will extend these modules to include other relevant genes by coexpression (which could hint at similar forms of regulation). 

```{r}
aba_genes <- c("VIT_05s0077g00430", "VIT_14s0060g00790", "VIT_14s0066g00810", "VIT_16s0050g00390", "VIT_18s0001g10450", "VIT_11s0016g03180", "VIT_08s0058g00470", "VIT_17s0000g01080", "VIT_06s0004g04860", "VIT_00s0322g00030", "VIT_06s0061g00730", "VIT_07s0005g00140", "VIT_00s0934g00010", "VIT_18s0001g04800", "VIT_01s0011g04820", "VIT_00s0216g00060", "VIT_18s0072g01220", "VIT_15s0048g02870", "VIT_00s0203g00070", "VIT_19s0014g03290", "VIT_01s0026g02710", "VIT_17s0000g08080", "VIT_00s0429g00040", "VIT_00s1206g00010", "VIT_09s0002g03690", "VIT_13s0067g00240", "VIT_07s0104g00350", "VIT_04s0023g02480", "VIT_18s0001g08490", "VIT_13s0019g02110", "VIT_04s0008g01120", "VIT_10s0003g00390", "VIT_03s0017g01410", "VIT_08s0007g08030", "VIT_08s0040g01820", "VIT_08s0058g01260", "VIT_09s0002g00630", "VIT_14s0068g01620", "VIT_18s0001g01030", "VIT_16s0050g01880", "VIT_08s0007g08280", "VIT_09s0002g06790", "VIT_09s0002g00640", "VIT_03s0063g01790", "VIT_04s0044g01230", "VIT_08s0040g02610", "VIT_01s0010g03680", "VIT_01s0137g00780", "VIT_02s0025g00500", "VIT_06s0004g05650", "VIT_07s0005g01080", "VIT_13s0156g00100", "VIT_17s0000g01120", "VIT_18s0001g07840", "VIT_18s0001g09290")
#QuickSearch_ABAcoreset <- create_module(compendium="vitis_vinifera", biofeatureNames=aba_genes)
```

Plot the heatmap

```{r}

```



# Session information

```{r si, echo=FALSE}
print(sessionInfo(), locale = FALSE)
```
