---
title: "An R interface to COMPASS, the gene expression compendia GraphQL endpoint"
author: Paolo Sonego
affiliation: https://github.com/onertipaday/rcompass
package: rcompass
abstract: > 
  How to query COMPASS from R
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{An R interface to COMPASS, the gene expression compendia GraphQL endpoint}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteKeywords{Bioinformatics, Gene Expression, data}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
options(scipen = 2, digits = 6)
```
# Introduction

## Installation

`rcompass` installation from github requires the devtools package to be installed.

```{r install, eval=FALSE}
if(!requireNamespace("devtools")) install.packages("devtools")
devtools::install_github("onertipaday/rcompass")
```

## Getting help

To get help open an issue on the `rcompass`
[github page](https://github.com/onertipaday/rcompass/issues).

## The resource

[COMPASS](http://compass.fmach.it/graphql/) GraphQL Endpoint
[COMPASS documentation](https://compass-.readthedocs.io/en/latest/index.html)

## The package

COMPASS (COMpendia Programmatic Access Support Software) is a software layer that provides a GraphQL endpoint to query compendia built using COMMAND>_ technology [1](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6348648/). 
COMPASS is meant to be the barebone interface on top of a compendium database from a
single location with a unified output format. The `rcompass` package make this possible from within R. It relies on both the `httr` and `ghql` packages for querying the GraphQL interface, access and retrieve data.

# A Brief rCOMPASS overview

The `rcompass` package is build around a bunch of functions enable to query the compendium of interest and retrieve, store and manipulate data. Each of these functions are described in more details in their respective manual pages. We start by loading the package.

```{r setup}
library(rcompass)
```

The COMPASS GraphQL endpoint might hosts different compendia. At the moment there is only the VESPUCCI compendium, but there are different version of VESPUCCI, and each version might have data normalized in different ways. In this case there are 2 versions of VESPUCCI, version 1.0 (legacy) and version 2.0 (latest). The latter has data normalized in 2 different ways, TPM normalization and LIMMA (the default one) while the legacy version has the legacy normalization only (i.e. per-sample logratios). For every query we will need to indicate the compendium we want to use, if no version, no normalization and no database is specified the default values will be used.

```{r}
get_available_compendia()
```

Now we can use `rcompass` to retrieve some stats about VESPUCCI. The Vitis gene expression compendium version `r paste(get_available_compendia()$versions[[1]], collapse = " ")` normalized contains values for `r totalCounts(version="1.0", aggregate_type="biofeatures")` biological features (genes), measured for `r totalCounts(version="1.0", aggregate_type="sampleSets")` sample sets. This corresponds to a total of `r totalCounts(aggregate_type = "experiments")` experiments and `r totalCounts(aggregate_type = "samples")` samples measured on `r  totalCounts(aggregate_type = "platforms")` different platforms.

# Session information

```{r si, echo=FALSE}
print(sessionInfo(), locale = FALSE)
```
