<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Use Case 02 • rcompass</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Use Case 02">
<meta property="og:description" content="rcompass">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">rcompass</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/rcompass.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/usecase01.html">Use Case 01</a>
    </li>
    <li>
      <a href="../articles/usecase02.html">Use Case 02</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/onertipaday/rcompass/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="usecase02_files/header-attrs-2.7/header-attrs.js"></script><script src="usecase02_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Use Case 02</h1>
                        <h4 class="author">Paolo Sonego</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/onertipaday/rcompass/blob/master/vignettes/usecase02.Rmd"><code>vignettes/usecase02.Rmd</code></a></small>
      <div class="hidden name"><code>usecase02.Rmd</code></div>

    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>Beyond basics</p>
    </div>
    
<p><a href="https://vespucci.readthedocs.io/">VESPUCCI</a> is the gene expression database for grapevine and we can access it via its GraphQL interface, called <a href="https://compass-.readthedocs.io/">COMPASS</a>. The <a href="https://onertipaday.github.io/rcompass/">rCOMPASS</a> package is a R package that wraps some functionalities to simplify communication with the COMPASS interface.</p>
<p>In this second exercise we will use <a href="https://vespucci.readthedocs.io/">VESPUCCI</a> together with some dedicate R packages to create an integrated workflow that goes beyond the functionality that <a href="https://vespucci.readthedocs.io/">VESPUCCI</a> alone provides.</p>
<p>The goal here is to answer the following question: <em>which are the “biological processes” that get shut down in fruit?</em> To answer this question we will need to perform a GO enrichment analysis (using <a href="http://www.bioconductor.org/packages/release/bioc/html/topGO.html">topGO</a> package) on a subset of genes that are not expressed in fruit tissue.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># devtools::install_github("onertipaday/rcompass")</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">rcompass</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; This is 'rCOMPASS' version 0.0.4 based on COMPASS version 0.8.7.</span></code></pre></div>
<p>The COMPASS GraphQL endpoint might hosts different compendia. At the moment there is only the VESPUCCI compendium, but there are different version of VESPUCCI, and each version might have data normalized in different ways. In this case there are 2 versions of VESPUCCI, version 1.0 (legacy) and version 2.0 (latest). The latter has data normalized in 2 different ways, TPM normalization and LIMMA (the default one) while the legacy version has the legacy normalization only (i.e. per-sample logratios). For every query we will need to indicate the compendium we want to use, if no version, no normalization and no database is specified the default values will be used.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/get_available_compendia.html">get_available_compendia</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; $info</span>
<span class="co">#&gt;                                   name                               fullName </span>
<span class="co">#&gt;                             "vespucci"                             "VESPUCCI" </span>
<span class="co">#&gt;                            description </span>
<span class="co">#&gt; "The Vitis gene expression compendium" </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $versions</span>
<span class="co">#&gt; $versions[[1]]</span>
<span class="co">#&gt;            versionNumber           databases.name databases.normalizations </span>
<span class="co">#&gt;                    "1.0"         "vitis_vinifera"       "legacy (default)" </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $versions[[2]]</span>
<span class="co">#&gt;             versionNumber            databases.name databases.normalizations1 </span>
<span class="co">#&gt;                     "2.0"          "vitis_vinifera"                     "tpm" </span>
<span class="co">#&gt; databases.normalizations2 </span>
<span class="co">#&gt;         "limma (default)"</span></code></pre></div>
<p>For this case study we decide to use the TPM normalized values: versionNumber=“2.0” and databases.normalizations1=“tpm_sample”.</p>
<div id="select-berry-samples" class="section level3">
<h3 class="hasAnchor">
<a href="#select-berry-samples" class="anchor"></a>Select berry samples</h3>
<p>Using a SPARQL query we can select only the samples annotated as berry using the following triple template where <a href="http://www.ontobee.org/ontology/PO?iri=http://purl.obolibrary.org/obo/PO_0030108">PO_0030108</a> is the <a href="https://www.ebi.ac.uk/ols/ontologies/po">Plant Ontology</a> term for <strong><em>berry fruit</em></strong>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sparql</span> <span class="op">=</span> <span class="st">"SELECT ?s ?p ?o WHERE { ?s &lt;http://www.w3.org/1999/02/22-rdf-syntax-ns#type&gt; &lt;http://purl.obolibrary.org/obo/PO_0030108&gt;}"</span>

<span class="va">berry_samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_sparql_annotation_triples.html">get_sparql_annotation_triples</a></span><span class="op">(</span>target <span class="op">=</span> <span class="st">"sample"</span>, 
                                               normalization <span class="op">=</span> <span class="st">"tpm"</span>, 
                                               query <span class="op">=</span> <span class="va">sparql</span><span class="op">)</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span></code></pre></div>
</div>
<div id="get-the-samplesets" class="section level3">
<h3 class="hasAnchor">
<a href="#get-the-samplesets" class="anchor"></a>Get the SampleSets</h3>
<p>Now that we have the samples we need to retrieve the <strong>SampleSets</strong> objects for this specific normalization (the TPM normalization).</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">berry_ss</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_sampleset_id.html">get_sampleset_id</a></span><span class="op">(</span>name_In <span class="op">=</span> <span class="va">berry_samples</span>, 
                             normalization <span class="op">=</span> <span class="st">"tpm"</span>,
                             useIds <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
</div>
<div id="get-the-genes" class="section level3">
<h3 class="hasAnchor">
<a href="#get-the-genes" class="anchor"></a>Get the genes</h3>
<p>We will need to use all genes</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_biofeature_id.html">get_biofeature_id</a></span><span class="op">(</span>name_In <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></code></pre></div>
</div>
<div id="create-the-module" class="section level3">
<h3 class="hasAnchor">
<a href="#create-the-module" class="anchor"></a>Create the module</h3>
<p>Now that we have both the genes (<strong>BiologicalFeatures</strong>) and the <strong>SampleSets</strong>, we can create our <strong>Module</strong> and retrieve the TPM-normalized values.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">module</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_module.html">create_module</a></span><span class="op">(</span>compendium <span class="op">=</span> <span class="st">"vespucci"</span>,
                        normalization <span class="op">=</span> <span class="st">"tpm"</span>,
                        biofeaturesNames <span class="op">=</span> <span class="va">genes</span><span class="op">$</span><span class="va">id</span>, 
                        samplesetNames <span class="op">=</span> <span class="va">berry_ss</span><span class="op">$</span><span class="va">id</span>, 
                        useIds <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
</div>
<div id="format-the-values" class="section level3">
<h3 class="hasAnchor">
<a href="#format-the-values" class="anchor"></a>Format the values</h3>
<p>In order to better handle the module values and get rid of NaNs, we will create a data.frame starting from our Module values (we are using log2), with genes as rows and SampleSets as columns.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># df = pd.DataFrame(np.log2(module.values), columns=[ss.name for ss in module.sample_sets], index=[bf.name for bf in module.biological_features])</span>
<span class="co"># df = df.replace([np.inf, -np.inf], np.nan)</span>
<span class="co"># df = df.T.fillna(df.mean(axis=1)).T</span>
<span class="co"># data = df.replace([np.inf, -np.inf], np.nan).dropna().values</span>
<span class="co">#module_df &lt;- sapply(as.data.frame(module), function(x) replace(x, is.na(x), mean(x, na.rm = TRUE)))</span>
<span class="co">#</span>
<span class="co"># alternatively (much, much faster!)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://zoo.R-Forge.R-project.org/">zoo</a></span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'zoo'</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     as.Date, as.Date.numeric</span>
<span class="va">module_df</span> <span class="op">&lt;-</span> <span class="fu">zoo</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/zoo/man/na.aggregate.html">na.aggregate</a></span><span class="op">(</span><span class="va">module</span><span class="op">)</span>
<span class="co"># or</span>
<span class="co"># module_df &lt;- imputeTS::na_mean(module)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">module_df</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">berry_ss</span><span class="op">$</span><span class="va">name</span>; <span class="fu"><a href="https://rdrr.io/r/base/row.names.html">row.names</a></span><span class="op">(</span><span class="va">module_df</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">genes</span><span class="op">$</span><span class="va">name</span></code></pre></div>
</div>
<div id="gaussian-mixture-model-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#gaussian-mixture-model-analysis" class="anchor"></a>Gaussian Mixture Model Analysis</h1>
<div id="fit-a-2-component-gaussian-mixture-model" class="section level2">
<h2 class="hasAnchor">
<a href="#fit-a-2-component-gaussian-mixture-model" class="anchor"></a>Fit a 2-component Gaussian Mixture Model</h2>
<p>The idea to fit a 2-component Gaussian Mixture Model comes from the <a href="https://www.embopress.org/doi/full/10.1038/msb.2011.28">following paper</a>: Hebenstreit, D., Fang, M., Gu, M., Charoensawan, V., van Oudenaarden, A., &amp; Teichmann, S. A. (2011). <em>RNA sequencing reveals two major classes of gene expression levels in metazoan cells</em>. Molecular systems biology, 7(1).</p>
<div id="plot-the-data-distribution-and-the-fitted-model" class="section level3">
<h3 class="hasAnchor">
<a href="#plot-the-data-distribution-and-the-fitted-model" class="anchor"></a>Plot the data distribution and the fitted model</h3>
</div>
</div>
</div>
<div id="session-information" class="section level1">
<h1 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session information</h1>
<pre><code>#&gt; R version 4.0.4 (2021-02-15)
#&gt; Platform: x86_64-redhat-linux-gnu (64-bit)
#&gt; Running under: Fedora 33 (Workstation Edition)
#&gt; 
#&gt; Matrix products: default
#&gt; BLAS/LAPACK: /usr/lib64/libflexiblas.so.3.0
#&gt; 
#&gt; attached base packages:
#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     
#&gt; 
#&gt; other attached packages:
#&gt; [1] zoo_1.8-9      rcompass_0.0.4
#&gt; 
#&gt; loaded via a namespace (and not attached):
#&gt;  [1] Rcpp_1.0.6        knitr_1.31        magrittr_2.0.1    graphql_1.5      
#&gt;  [5] lattice_0.20-41   R6_2.5.0          ragg_1.1.2        rlang_0.4.10     
#&gt;  [9] fastmap_1.1.0     stringr_1.4.0     tools_4.0.4       grid_4.0.4       
#&gt; [13] xfun_0.22         jquerylib_0.1.3   htmltools_0.5.1.1 systemfonts_1.0.1
#&gt; [17] yaml_2.2.1        digest_0.6.27     rprojroot_2.0.2   httpcode_0.3.0   
#&gt; [21] pkgdown_1.6.1     crayon_1.4.1      RJSONIO_1.3-1.4   textshaping_0.3.3
#&gt; [25] sass_0.3.1        fs_1.5.0          triebeard_0.3.0   curl_4.3         
#&gt; [29] crul_1.1.0        ghql_0.1.0        memoise_2.0.0     cachem_1.0.4     
#&gt; [33] evaluate_0.14     rmarkdown_2.7     stringi_1.5.3     compiler_4.0.4   
#&gt; [37] bslib_0.2.4       urltools_1.7.3    desc_1.3.0        jsonlite_1.7.2</code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Paolo Sonego.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
