<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Use Case 01 • rcompass</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Use Case 01">
<meta property="og:description" content="rcompass">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">rcompass</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/rcompass.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/usecase01.html">Use Case 01</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/onertipaday/rcompass/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="usecase01_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Use Case 01</h1>
                        <h4 class="author">Paolo Sonego</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/onertipaday/rcompass/blob/master/vignettes/usecase01.Rmd"><code>vignettes/usecase01.Rmd</code></a></small>
      <div class="hidden name"><code>usecase01.Rmd</code></div>

    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>How to query COMPASS from R</p>
    </div>
    
<p><a href="https://vespucci.readthedocs.io/">VESPUCCI</a> is the gene expression database for grapevine and we can access it via its GraphQL interface, called <a href="https://compass-.readthedocs.io/">COMPASS</a>. The <a href="https://onertipaday.github.io/rcompass/">rCOMPASS</a> package is a R package that wraps some functionalities to simplify communication with the COMPASS interface.</p>
<p>In this first exercise we will perform few basic operations with VESPUCCI. We will create a Module starting from few genes and then automatically extend it by adding more genes. We will also have a look at gene and samples annotations.:</p>
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="co"># devtools::install_github("onertipaday/rcompass")</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">rcompass</span>)
<span class="co">#&gt; </span>
<span class="co">#&gt; This is 'rCOMPASS' version 0.0.3 based on COMPASS version 0.8.5.</span></pre></body></html></div>
<p>The COMPASS GraphQL endpoint might hosts different compendia. At the moment there is only the VESPUCCI compendium, but there are different version of VESPUCCI, and each version might have data normalized in different ways. In this case there are 2 versions of VESPUCCI, version 1.0 (legacy) and version 2.0 (latest). The latter has data normalized in 2 different ways, TPM normalization and LIMMA (the default one) while the legacy version has the legacy normalization only (i.e. per-sample logratios). For every query we will need to indicate the compendium we want to use, if no version, no normalization and no database is specified the default values will be used.</p>
<div class="sourceCode" id="cb2"><html><body><pre class="r"><span class="fu"><a href="../reference/get_available_compendia.html">get_available_compendia</a></span>()
<span class="co">#&gt; $info</span>
<span class="co">#&gt;                                   name                               fullName </span>
<span class="co">#&gt;                             "vespucci"                             "VESPUCCI" </span>
<span class="co">#&gt;                            description </span>
<span class="co">#&gt; "The Vitis gene expression compendium" </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $versions</span>
<span class="co">#&gt; $versions[[1]]</span>
<span class="co">#&gt;            versionNumber           databases.name databases.normalizations </span>
<span class="co">#&gt;                    "1.0"         "vitis_vinifera"       "legacy (default)" </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $versions[[2]]</span>
<span class="co">#&gt;             versionNumber            databases.name databases.normalizations1 </span>
<span class="co">#&gt;                     "2.0"          "vitis_vinifera"              "tpm_sample" </span>
<span class="co">#&gt; databases.normalizations2 </span>
<span class="co">#&gt;         "limma (default)"</span></pre></body></html></div>
<div id="genes" class="section level1">
<h1 class="hasAnchor">
<a href="#genes" class="anchor"></a>Genes</h1>
<p>Let’s build our module starting from a bunch of genes that might come from a previous analysis. Gene name used are the VIT_ ids</p>
<div class="sourceCode" id="cb3"><html><body><pre class="r"><span class="no">gene_names</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">'VIT_00s0246g00220'</span>,<span class="st">'VIT_00s0332g00060'</span>,<span class="st">'VIT_00s0332g00110'</span>,<span class="st">'VIT_00s0332g00160'</span>,<span class="st">'VIT_00s0396g00010'</span>,<span class="st">'VIT_00s0505g00030'</span>,<span class="st">'VIT_00s0505g00060'</span>,<span class="st">'VIT_00s0873g00020'</span>,<span class="st">'VIT_00s0904g00010'</span>)</pre></body></html></div>
<p>We can query the compendium with the list of gene names and get a list of BiologicalFeature objects that represents our genes of interest. The easiest way to know which are the valid filter values to be used is to use the autocompletion (ALT + SPACEBAR) in the COMPASS GraphQL interface at <a href="http://compass.fmach.it/graphql">http://compass.fmach.it/graphql</a>.</p>
<div class="sourceCode" id="cb4"><html><body><pre class="r"><span class="no">bf_ids</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/get_biofeature_id.html">get_biofeature_id</a></span>(<span class="kw">compendium</span> <span class="kw">=</span> <span class="st">"vespucci"</span>, <span class="kw">name_In</span> <span class="kw">=</span> <span class="no">gene_names</span>)
<span class="kw pkg">knitr</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span>(<span class="no">bf_ids</span>)</pre></body></html></div>
<table class="table">
<colgroup>
<col width="27%">
<col width="17%">
<col width="55%">
</colgroup>
<thead><tr class="header">
<th align="left">id</th>
<th align="left">name</th>
<th align="left">description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">QmlvRmVhdHVyZVR5cGU6Mjc0Mzk=</td>
<td align="left">VIT_00s0246g00220</td>
<td align="left">ATPase subunit 8</td>
</tr>
<tr class="even">
<td align="left">QmlvRmVhdHVyZVR5cGU6Mjc4OTI=</td>
<td align="left">VIT_00s0332g00060</td>
<td align="left">ATP synthase subunit beta (Fragment)</td>
</tr>
<tr class="odd">
<td align="left">QmlvRmVhdHVyZVR5cGU6Mjc4OTY=</td>
<td align="left">VIT_00s0332g00110</td>
<td align="left">PSII reaction center subunit X</td>
</tr>
<tr class="even">
<td align="left">QmlvRmVhdHVyZVR5cGU6Mjc5MDA=</td>
<td align="left">VIT_00s0332g00160</td>
<td align="left">NADH dehydrogenase subunit 7</td>
</tr>
<tr class="odd">
<td align="left">QmlvRmVhdHVyZVR5cGU6MjgwODE=</td>
<td align="left">VIT_00s0396g00010</td>
<td align="left">Photosystem II CP43 chlorophyll apoprotein (Fragment)</td>
</tr>
<tr class="even">
<td align="left">QmlvRmVhdHVyZVR5cGU6MjgyNzg=</td>
<td align="left">VIT_00s0505g00030</td>
<td align="left">Orf300 protein</td>
</tr>
<tr class="odd">
<td align="left">QmlvRmVhdHVyZVR5cGU6MjgyODE=</td>
<td align="left">VIT_00s0505g00060</td>
<td align="left">PSII N protein</td>
</tr>
<tr class="even">
<td align="left">QmlvRmVhdHVyZVR5cGU6Mjg1OTg=</td>
<td align="left">VIT_00s0873g00020</td>
<td align="left">NADH-ubiquinone oxidoreductase chain 3</td>
</tr>
<tr class="odd">
<td align="left">QmlvRmVhdHVyZVR5cGU6Mjg2MDk=</td>
<td align="left">VIT_00s0904g00010</td>
<td align="left">Oxygen-evolving enhancer protein 3, chloroplast, putative</td>
</tr>
</tbody>
</table>
<p>Each BiologicalFeature object has several fields. In our case, since the BiologicalFeature represent a gene, we have sequence. The id is the identifier used internally in the database.</p>
<div class="sourceCode" id="cb5"><html><body><pre class="r"><span class="kw pkg">knitr</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span>(<span class="fu"><a href="../reference/get_biofeature_by_name.html">get_biofeature_by_name</a></span>(<span class="kw">name_In</span> <span class="kw">=</span> <span class="no">gene_names</span>))</pre></body></html></div>
<table class="table">
<colgroup>
<col width="0%">
<col width="1%">
<col width="98%">
</colgroup>
<thead><tr class="header">
<th align="left">name</th>
<th align="left">id</th>
<th align="left">value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">VIT_00s0246g00220</td>
<td align="left">QmlvRmVhdHVyZVZhbHVlVHlwZToyNzQzOQ==</td>
<td align="left">ATGCCTCAACTGGATAAATTCACTTATTTCACACAATTCTTCTGGTCATGCCTTTTCCTCTTTACTTTCTATATTCCCATATGCAATGATGGAGATGGAGTACTTGGGATCAGCAGAATTCTAAAACTACGGAACCAACTGGTTTCACACCGGGGGAACAACATCCGGAGCAACGACCCCAACAGTTTGGAAGATATCTTGAGAAAAGGTTTTAGCACCGGTGTATCCTATATGTACTCTAGTTTATTCGAAGTATCCCAATGGTGTAACGTCGTCGACTTATTGGGAAAAAGGAGTCAAATAACATTGATCTCTTGTTTCGGAGAAATAAGTGGCTCACGAGGAATGGAAAGAAACATATTCTATTTGATCTCGAAGTCCTCATATAGCACTTCTTCCAATCCTGGATGGGGGATCACTTGTAGGAATGACATAATGCTAATCCATGTTCCACACGGCCAAGGAAACATCGTTTTTTAATCTCATTATGACCGGAAGAAATTCTTTCTCGATCAGAAAAGTGGAATGGAAGGCACTACAAGCAAGAAAGATATACTCCTTTTCAAATCGCCCCGCGAAGACGGACGTTCAGAAAGGTTCTCGAGATTCTCAATCGCTCTAGTGGGGAGGGATAGTGCTTGAAGGGAGCGAGACCAAGCCGAGCCACTAGGAGAGTAAGCCCTTCCCACGTGTCAAGTTGAATAAATGAATGCAGCCTCAACCAGAGAGATCAACCAGCGCCTTTTATTTTAGGCCGTCGGCGGTGCAGAACGCACTAATCCGCGACCAACAAGTTTTCCGAAACCGAACTGAATGGAATTCTTACTTTAAAAAAATGCTTGCTGAAAATCAAAGAAAGAAGGTCCATTTTCCCACGTAGTTCGTCGGTCAAACCAACGATTCTCAAAGTAATAGAGAGATCTTTTTCTAGTTAGACTTCTATCAATGCAATGAAAGAACCATCCCTTCCTATTTGTTTGTCCTGTCAGATATAAAGAAAATTAGACCCCAAAGAGCCCTTCTTTACCACTTTAGGGGGTGGGGGTGAAGGGGGGGTTTACATACAACCAAGGCAAAGTGGTTTATGATTGAATCTCAGAGGCATTCTTTTCATTTGGTAGATCCAAGTCCATGGCCTATTTCGGGTTCACTCGGAGCTTTGGCAACCACCGTAGAGGTGTGATGTACATGCACCCATTTCAAGGGGGTGCAACACTTCTCAGTTTGGCATCATATTTATCCTATATACCATGTTCGTATGCTGGCGCGATGTTATACGTGAATCAACGTTGGGAGGACATCATACCAAAGTCGTACA</td>
</tr>
<tr class="even">
<td align="left">VIT_00s0332g00060</td>
<td align="left">QmlvRmVhdHVyZVZhbHVlVHlwZToyNzg5Mg==</td>
<td align="left">ACTAAAAATTTAGCTCATTGAATAAGTAAACAATTGAATTGGATTCGGTTGGATGGTACCAACGAAATCGAGTGCTAACTCCCATTTCTTATTGAATTAACCGATCAACTTTCTATCGAACTTTTGGATTCGATAATTTTCGCAAAAAATTTCGTTTAATTTGACTTTATTATTATTATTATGAGAATAAATCCTACTACTTCTGGTCCTAGGGTTTCCACGCTTGAAAAAAAAAACCTGGGGCGTATCGCTCAAATCATTGGTCCGATCCTGGATGTAGCTTTTCCCCCGGGCAAGATGCCTAATATTTACAATGCGCTGGTAGTTAAGGGTCGAGATACTGTCGGTCAACAAATTAATGTGACTTGTGAGGTACAGCAATTATTAGGAAATAATCGAGTTAGAGCTGTGGCTATGAGTGCTACAGATGGTCTAATGAGAGGAATGGAAGTGATTGACACGGGAGCTCCTCTAAGTGTTCCAGTCGGTGGAGCGACTCTCGGACGAATTTTCAATGTGCTTGGAGAGCTTGTTGATAATTTAGGTCCTGTAGATACTCGCACAACATCTCCTATTCATAGATCTGCGCCCGCATTTATACAGTTAGATACAAAATTCTCTATTTTTGAAACAGGAATTCAAGTTGTAGATCTTTTAGCCCCTTATCGCCTTGGAGGAAAAATCGAACTACTCGGGTTGGCTGGTTGGAAGGATTGCTTTCTGCCGTCTCTCGTTCTTGACAGATCCGATCGGGTGTTCATAATCTGGAGTAAAAGGATTCGAACCTTTGCATGCCGGTACCAAAAACCGATGCCTTACCACTTGGCTATACTCCATACGGCCTTAGGCCTTTTGATAACTAATTAGAGTTGACATGAAATGGGTCACGAAAGAAGACGTATCCGATGAATGAATGATTCATTCCCCTCCCACTCACACGGGTTCTTCTATTGAAGAGGTTCCCCTTCTTCTATGTGAGGTGGGGCGTACGTAAGAGCGGAACCTAGATAGAACGCGGAGCGCCGGCCCCGGCCGCCGATACAGTTACCATGCTTACCAGCTCTTACCATTGCCGCCTATATATATAGATGGGAGGTAGGCGAGGGTAGCTTGCTTCTCTTCTCCTAGGCTACCTGCACATGTTATTCGCGAAGAAAAGGCAGCGAGCGGTTCTTCGCTTAGTAGACGGCCGCCGGACGACGACCGCTTCTTGTTCTCGCCCAGCCGCTTCTTTTTTTTTCTTATTATTTAGAATCCTAAACTAAAGAAAGAAGCTCCTGAATCAACGCAGGGTGGTCGTAGATCAGCAGGAGAACTGTACTAACCTGCCGCCGGTGACTTTATCATATCAGGGCTCCGCAGGAGAAGAAAGAAGGTCCGGGTCCAATTTATAATGAAGCGAAGGTCCCCCTTACTCCCTATTCTCTCTTAAAGCTTTATAAAGCTCGGCGAGAAAGGGTTGGTTAAGAACTATTGACTGTAAACCATAGCAGTTACAGTCACTCAATGGAGATGTTCGCCTTTGGAATGAAGATGGATGAGCAGGCACTTGAGCCTGGAACTGATTTAATTCGGGGAAAACATGGAACCGGTTACTATACTGGGGGAGGCGAGACATGACCGCGACTTAAGATTCAAGTACCGTTGAAAAGACTAAAGAAACGGGGGGA</td>
</tr>
<tr class="odd">
<td align="left">VIT_00s0332g00110</td>
<td align="left">QmlvRmVhdHVyZVZhbHVlVHlwZToyNzg5Ng==</td>
<td align="left">CAGTCATTATCCTTATAGAAGGCACTCACAAAGATAGAGTTACAGAGGTAGAAAAAAAAATGGATTCATCATCTGTTACATCTACCCATCCCGTGATTCCGAATCAACAGACCCAATTTCCGAGATATGGCAACTCTTTTATAAAGTAATAGTAATAAGAACTGTGTCGTGTAACTTCATTCACAAATGAAACTCCGGAAAAGCAAAATGGAAAATAATTTTTTTTTTTTCAAATTGATTTTCTTTTTCAATTGTATCAAATCCATTTTCTATTTTGGAACGAACAATGACCAACTAAGACTTTGACTTGACTTTAACTCATTAGATTCAGTAGTAGGTTCATTAATTGCACATAATATCTCGAATGAGAAGAAATTATCGCACGATCGCTCGAATAAATCAAACCTTTATTTCGGTCTAACTCGATTCTAAGACTACTAATTCCTATGTTCTTTTCCTTTTAGGTTCTCTTTTATTTCAATATTAGGGAGTCCCACCCTTGTAGCTAGGTCAAGAAAAAACCTTTGGATCTAATACAGCAATGCTTGCATCACGAATATTGATTGTTCCAATTATTTTATTGTTTGTTCTGTTTCTTTCATCGAAGACTATCTACTAGAATACTTGTTATTGTACGACCAACCAATTCCTTAAAATGAAAGGATTCGAACAAAAAGAAAAAAACCTTTTCTACAACCATGAAAAGGGGCTTTCTAGATTTCGCATCTAATTGAATTGTGGGAATATGAGAATCTCTTTCATGAATTATTAGAACCCAACTCGTCGGACTCAATCTTTACCAGATCTTCAGTTTCTAGTCTGAAGCCAGTAATGGAAAGAAACCCTATACTAAAGAAGAGGAATTTTCTATTGAATGGCACATGGTTCTGCATTGACAAGCAACAAGAAAGGAAGAAAGAAATGTTGTATCACTTCTTTTCTATACTGATTGAAATTGCGTTGCTGTGTCAGAAGAAGGATAGCTATACTGATTCGGTATACTCACACCCTCGGTACACTATTGAGGATCTCACAAAGATGAAATCTCAGTGAATTTCCATTTACTGATCCCATCTTTCACATAATAAATCCCCCCTTTGACTGTACATGAATATGTGGAGCTCAGCATGTCTGGAAGCACGGGAGAATGTTCTTTTGCTGATATTATTACCTGTATTCGATACTGGGTCATTCATAGCAAACCTATACCTTCCCTATTTACTCGACGAATTAAGGGGGTTGGTTATTCGTCAGCACGGGTTTTGCTTCCTCTTCTGGTATAGGTTTGGGAGCCCTCGGCCAAACGAGTATTTTACAGAGAGCCGACAAGGAATGGAATTAATAACTGGCCTGGCCGTTTTGATCCTTTGGCACAACTCGATTCATTTAGTAGATCCTTTTAGAAGGTCTCCCTCTTAGTAAGCTTAGATCGAACCTATCCCCTTTTTACAGTGCGATGGTTGGCTGTTCACGGACTAGCTGTACCTACCCTTTCTTTTTTGGGGTCAATATCAGCAATGCAGTTCATCCAACGAGAAACCTAATCCGAATTATAGAGCTACGACACAATCAAACCCGAACAAACAAAATGTTGAATTGAATCGTACCAGTCTCTACTGGGGGTTCTTACCCTATATATGACGAGCTGAACTAGCTGTTTTCTTTTAAAATTCTTTCTAAAATGAAGAGAAAGAAAGAGAATAGTAATAGTAGGAATTCTCTTATCCCATTCGGAAGGATCTCATCTCATAATTATCCATGACTGTTTATGTCTCTAGCATGACCACTTGATGAAATGTGGAGGGAAGTGGGGTAAATGGCCGATACTACCGGAAGGATTCCTCTTTGGCTGATAGGTACTGTAACTGGTATTCCTGTGATCGGTTCAATAGGTCTTTTCTTTTACGGTTCATATTCCGGATTGGGTTCATCCCTGTAGTAATCGGATGAACTGGGTTGT</td>
</tr>
<tr class="even">
<td align="left">VIT_00s0332g00160</td>
<td align="left">QmlvRmVhdHVyZVZhbHVlVHlwZToyNzkwMA==</td>
<td align="left">CGGAATGATCGGTCGGACAGAGGCTAGGGCTGCTCCTTCCCACCGGCGTCCTTCTTGTGTGTCGGAGATATAAAGCGAGTGCACGGAAAGAACGGGAACTGGGTCGATCTATTCTATGGCGAAGCATCCGAAGCATAACTGCACACTCACACGATCTTTGCCGAGAGATAGGAGCATTCGGTGGAACCGGTGAACTACACTTGCTTCTGGATAGATGTGTGGGACAGAGGGCTCGTGGTACCTGCTGCCCACCCTTCCCCCTCTGCTTTGAGAACCGTGTGAACGGAGAGTGGGCAGAAGGGAAGGAGGTCCTCATACGGAGTCGCACACTTACTTGAGCAGTGCGGGAGACTGGGGAATGGGTCGAGTAAACTCCCCTGGGGCCGAAAAAATAACAGACGAAGCTTGACTTAGATAGGGCTTTGATTTGATTGGTATTGATTTTTTCTTAGTGATTGGAAAGAAGGGCGCATGGGTTATAGAAAGGGAAGGCGGAGGTAGTACTTCGAATGGCGAAGAGAGGCGGCTCGGCAGGGAAGGAATGGGAAATCGTCAAGGATGACTTCTTTGTTGGCGAAACGAAGGGCTAAATAGCGCCAGTGATTCTCTGAGCCGGTCTGGATTTCCAACGCCTCGGGAAACTGGTCGAGATAGATGACAGCACCTTTTTCCTCTCTTCCTTACCGGGAGGGCAATCTTATCTTATGGCCTTCACCTCCAGCCCGGCCCAGAAATAGAACCCAGCCCCTTTTTGATCCATTCATTTCTGCAAGCAGGGAGGATCCAGAGCTAAGCCCCCCTCCTATTCGAGGTCGGGTGGCCTCGCCCCACAAGCACCTAACCTTCCTTTTGCTCTTCCTTCGGTTTGCCTCCACGAACGCGCCACCTAGGAGCCCTACTCATATTCCAAAGGCGCTACTTCAATTCAAGCGCAAGCCCCCCTTCTATTGCATAACCTAAAAAAGTGTTAAGCAAAGTACCAAAGTGGTTGCGGGAACGATACGCTTTGGTTACCGGCGAACGCTTCCAAATCTCGAGTTTCCGGCTGTTTCCTAAATTGAAGTAGCCTTTCGTCGCCCTACCAAATGAAAGAAGTCACTATCAAACAGCTCGCCCTACTGAAGTACCAAAGGTGCGCTCCGCCCGGTGACTAAGAAATAAATTGGTTTGCGCTTTCATTGAAGTGATGAGGGTGATCTACGATCAGGCTGGTCGGAACGATCAGGGGGAAGTAGGGCTCCTATTGAAACGCTTTCTCTCCCTCAAAAGGCGACCAGCTTTCAATACTAAGCCGTGTTATGGTTTCGTTGTTCGTTCCGTCGTCGACCCTATCTATGTTTCTACGATGGCCCAAGAGCACGCTCATTCTTCAGCCGTAGAGAGACTTTTGAATTGCGAGGTACCATTACGAGCTCAATATATACGAGTGTTATTCCGTGAAATAACTCGAATTTCAAATCATTCACTTGCTTTAACTACTCATGCTATGGATGTGGGAGCATCAACTCCGTCCCTGTGGGCTTCTGAGGAGCGGGAGAAATTGTTGGAATTCTATGAAAGAGTCTCGAGAGCCAGGATGCATGCCAGTTTCATACGACCAGGTGGAGTGGCACAAGATCTGCCTCTTGGCTTATGTCGAGATATTGATTCCTCCACACAACAATTTGCTTCTCGTATCGACGGATTAGAAGAGATGTCAACCGGCAACCGTATCTGGAAACAACGATTAGTGGATATTGGTACTGTCACTGCACAGCAAGCAAAGGATTGGGGATTCAGTGGTGTAATGTTAAGAGGTCGTGTACCAGAGTGGATCGCTATGATCGTTACTGTATCCGTATCGAAGATGATGCGACAAAGTGTTCGAATCATTGTGCAATGTCCTAATCAAATGCCTAGTGGCATGATCAAAGCCGATGATCGTAAGCTATGTCCTCCATCACGATGTCGAATGAAACTATCCATGGAATCACTCAGGTTAAGCTCCCATTGGGAACCCCAATTTCTTTCTTCGGAGCCGTTTCTTTTCCCGTCCCCCCGCCCCGGCATAGCGCTTCGCTTCCGGTTCTTCGGAAGAATCAACTGA</td>
</tr>
<tr class="odd">
<td align="left">VIT_00s0396g00010</td>
<td align="left">QmlvRmVhdHVyZVZhbHVlVHlwZToyODA4MQ==</td>
<td align="left">TTTCCATTCTTCGGTTATGTATGGAAAGATAGAAAGATAATGACCACAATTTTTGGTATTCACTTAATCTTGTTAGGTATAGGTGTTTTTCTTCTAGTATTCAAGGCTCTTTATTTTGGGGGCGTATATGATACCTGGGCTCCCGGAGGGTTAAGAAAAATTACCAACTTGACCCTTAGCCCAAGTGTTATATTTGGTTATTTACAAAAATCGCCCTTTGGAGGAGAAGGATGGATTGTTAGTGTGGACGATTTTGAAGATATAATTGGAGGACATGTATGGTTAGGTTCCATTTGTATACTTGGTGGAATCTGGCATATCTTAACCAAACCCTTTGCATGGGCTCGCCGTGCACTTGTATGGTCTGGAGAGGCTTACTTGTCTTATAGTTTAGGTGCTTTATCTGTCTTTGGTTTCATTGCTTGTTGCTTTGTCTGGTTCAATAATACCGCTTATCCTAGTGAGTTTTATGGGCCAACTGGACCAGAAGCGTCTCAAGCTCAAGCATTTACTTTTCTAGTGAGAGACCAACGTATTGGGGCTAACGTGGGATCCGCTCAAGGACCTACTTGGTTTAGGTAAATATCTCATGCGTTCCCCGACTGGAGAGGGTGGTCTACGATCAGATGGAGGAGAAACTATGCGCTTTTCAAAAATGGTCGGAAGAACGGGTTTCAAAAAATATATAAATATATATATATATATATATATATATATATTTATATATTACCGGTTGGCTGGTTTTTATGCAAAAAAGACAAAACGGGATTGGATTTCCACTCATAAGGAAGGAAGGTTAGATACCTTTGATAGGGTTGTATTTTTGGTTGAAATGGGATGGTGTTCAAACTCCCGAAATGCAGTCTATCTAGCTGATCTACGACCACCCTTTCTGACTGGACTGACTCGGGGAAGGGTCAATCTCCTCCGGAGCATGCTTCACAGCCACTCATTCGTCCTGACCAAATAGTAGAATATATCTCTCTGCGATTCTTTTCTCCGCTAATCACCCCTTCCCAACCAGCCCGCCCGATCGATTTTGTAAGATCGGCTAATTCAAAGGGGTTAATCTGGTCCGAAGTTGTCGGAACGAATCATTTGCTTTATCGAACAGGTTGGCCCCCTATTTTCAACCATTACAGCTGGCGTCGACCAGCTTTGCGATACGGACGGACACGAAGAAGAAGGCAGGCAGCGGAAATGCAAAAGAGAAGAGCAAAGATTCCCGACCCAGAGCATGTTATTGACTCAACCACAAATCTGTTGAATGAAGGCCATTTAGGGCTTACTCTCAGCCACTAGTTAGAAGGAAATCCCGTGATCTACAACCACCCTTATGCAGTAAAGAAAGCAAGTGAGGTGGGCTAAGATTCCTAACGTAACAGGATTCGATGTTGCACTATCTTCAACTCTTCTCGGGATCCGGAGTTTTTCGAGAAAAGGTCCCATCCTTCAATATCATGATTGGGTCGACCAGGCCAGATCATGAAATCATGAGTGAATAGAAAATCGAAAATGTCAATAGCAGTTGCTGTTCCAGCTGAAATACTTGGAATAATTCTACCACTTCTACTAGGAGTAGCCTTTTTAGTGCTAGCTGAACGTAAAGTA</td>
</tr>
<tr class="even">
<td align="left">VIT_00s0505g00030</td>
<td align="left">QmlvRmVhdHVyZVZhbHVlVHlwZToyODI3OA==</td>
<td align="left">CTTCAGAGAAGAGAAAGGAGGCATTCGTCAGCGTAGGAAGGCCAACCGAGCGAAGCTCTTTCTTTTTTTTTGGTCGTGCCGTGAAGTTCAATTGTATCGTATGTTAGTTAGAGAGGTTGGCGAACTACTACGATCTATAGATTCCCCATCTATCTATATCCAATCCCAACGAGAATAGAAGCAAGTCGCTTCCGGCTTGGCTTGTAGTCGTAGTCTTTTAGCTTATGTAGTGGTCGGCCTTCCTACACGCACGCGCCCGCCAAAATGCCTCCTTGGTTCTGGAGACGAAGCCAAGCCGATACATACGATAGGCGAAGGAAAGACCCCCCATTTCATAGCACCTGGGGAACGGGGGTTCGCGTCTATACCTATAGTGTGAGAGTTGGGCCTACCCACCCTATTTGTTCCATGATCTATGGGTCTACTGGAGCTACCCACTTCGATCAATTAGCCAAGATTTTGACCGGATACGAAATCACTGGTGCTCGATCTAGTGGTCTTTTTATGGGGATTCTCTCTATCCCTGTAGGATTCCTATTGAAGATCACTGCAAAAAACCGCCTGGAAGTCCGAGGACCTTTAGTACCGTACCCTACCCTCGAACTAGCAGCCTTCGCGCCAAGCAAGACCGCCCTTGTCCCTTTCCTTTCTCCATTCCGCCTCCTTCTTTGCTTTTTTCGTTTCCGGGTTTATGGATTGGATTCAGTCAGCCCTTTCGTCGCCCTACCAAACGAAAGAAAAGTGGTCGTAGATCATTTTATTTGTTTTTCATCGAAAACGAAGAAGGCCGAGGATGGCCTACGCTGCCTTCGGGCTTTGCCTGCCCTTTATAATAAAGAATTCCGGCTCGGCCCGGGAAAGCGCTGGCAACAACAGAAAGGAAGGGGTCCATGTAGCTGCTGCGCCCGCCCCCTTCTTAGTCAATGGGGCAGCAGGTTCGGCATCTACTATAAAAGAGAGAATCCACTTCAGATAACCACGCCTCTGCTAGGCAGCCCCCGAAGAAAAGCCCCCTCCCCTTCGGACTCCATATCTCTTTTGACTTTATATATGTGGGCACCTGATATCTATGAGGGTTCACCCACCCCGGTTACAGCATTCCTTTTTATTGCGCCTAAAATCTCTATTTCTGCTAATATTTCACGTGTTTCTATTTATGGTTCCTATGGAGCTACATTGCAACAAATCTTCTTTTTCTGCAGCATTGCTTCTATGATCTTAGGAGCACTGGCCGCCATGGCCCAAACGAAAGTAAAAAGACCTCTAGCTCATAGTTCAATTGGACATGTAGGTTATATTCGTACTGGTTTATCATGTGGAACCATAGAAGGAATTCAATCACTACTAATTGGTATCTTTATTTATGCATCAATGACGATAGATGCATTCGCCATAGTTTTAGCATTACAGCAAACCTGTGTCAAATATATAGCGGATTTGGGCGCTCTAGCCAAAACGAATCCTATTTCGGCTATTACCTTCTCCATTACTATGTTCTCATACGCAGGAATACCCCCGTTAGCCGGCTTTTGTAGCAAATTCTATTTGTTCTTCGCCGCTTTGGGTTGTGGGGCTTACTTCCTAGCCCCAGTGGGAGTAGTGACTAGCGTTATAGGTCGTTGGGCGGTCGGAAGGTTGCCACGACTTACCGAATCAGTTGCGACACGGATGGGAATGCATGCTACGAAACATAGGGTCGAGTCTAATACATCAACCGTCTACTCAATATCCTTGTACGAGTCCACAATCACTACACGAGATGAACCTTGGTTTGGTGAATTGAAGTTGGCCTTAGGTGTAATAGGACTCCCAGTTACTGCGCGCGATCGTATACTGAGGTGCTCCCCGCCGGTTGTTGGAACGACGCGAGCCGGGCCGGGCCCCGATTCAAAAAGATGAAGGGCCAAAAAGTATAAATAGGGGGTTAAAAATTCCCCATCTCATTAGGGGAGGAAAACGAATCGGCATCTCGATGTGATACAGCCTTTTCTATTTGAGTTGGGAAAGAACGGCGAAGTCCATCCGAACCGTCCAATGAAGAATAAGAGGAGAGCAAAGCGCCAATGGCGCGCGAAGCGCATGCGGAACGGGCACGGAAAAAAAATAAAGAAGTGTGGAGGGTGGTCGTAGATCAGTAGGGCTCATTCCCTTCGCTTCCTGGGCCCAAAGCAGTGCAGTCTTTCCTGGCCAAATCAAGGATTTGGGGCTTCTTGCTACGCTACTTAACTATATAAATCCATTTTCTTTAGTAATATATATGAATAGAAAGATAGATCCATCCATCTATCCTATCCGATTTAGATTTTGATATCTAAAAAACAATAGATTTCATTCAACGTTTGATTCAAAGAACTGCGCTTAGCCCCCCCGCTCATGAAACGGCTCTGCTGCAATGGATGGCAGAGGGTCCGTAGTACCCAAAGCACTGGAGTGATCCAGTAGCCGGGAAGGGGCCTAGAAGTGCCTACTACTACACCACACTACACTTGGCTCTACACATTTACAGAGCTAACCCCTGTCCAGTGCCTGGCAGAGCTAAAGGGGCTTCAATCCTTACTCCTTATCCCCATCTTCGCCCAGGCTAACGGGGCCTTCCTTATTCAGGGGGGAGAGTGGAGCCTCGAGAAGCACTGTTGAGAGGAAGATCCTTGGCCCCTCTTCATTCTCTACAGGGTTCCAAACCTTTCTTCAACATAGGTGACAACGAGCAGGGCAGAGATGGAAGAGATCGATCACGGGAATAAGAAGCAAGCTCGCCTTCCTTTTTGATCATTTTGATAGAGGGGGATGGAGAAAGTGGACAAAAAAGACTCGCATTTCCCAGTCGAGGGTTCCTTTTTCGTCTCGCATTTATGGAAAAGAAGAATATTGAAAACGCTCCTAACCCAACCCCTTCCTTCGTAGAGCCGTGTATTGTAAGTGATCCGAACCTGCCCGGAGTGAGCCCCCCATAGAGGCAAGTGAAGTTGGTGAGCCGTATGATGGGCAACTATCTCCTGCGGTTCGGAGAGGACTCAGCTGTTAGTTAGTACCCCCTTGGATTCGGGGTGGACCCTTTCACTCTATTTTATTATATACGCTTCGCGAAAAGAATGTTTTTTGATACACCTAGGACATGGATTCTATATGAACCAATGGATCGTGACAAGTCGTTACTACTAGCAATGACTTCCTCTTTCATTACTTCATCCTTTCTATATCCCTCTCCCTTGTTCTCAGTTACTCATCAAATGGCACTCAGTTCATATCTTTAAGTTTGATCATTGACAAGGTTCAAAGAAAGGGTTGGCCATTGACAA</td>
</tr>
<tr class="odd">
<td align="left">VIT_00s0505g00060</td>
<td align="left">QmlvRmVhdHVyZVZhbHVlVHlwZToyODI4MQ==</td>
<td align="left">ATGGAAACAGCAACCCTAGTCGCCATCTCTATATCAGGTTTACTTTTTAGTTTGACTGGGTACGCCTTATATACCGCCTTTGGGCAGCCCTCTCAACAACTAAGAGATCCATTCGAGGAACACGGGGACTAGTTGAAGTAATGAGCCTCCCCATATTAGGGGGCTCATTACTGAAATTGAGATAATGAAAAAGAATTTCATCTTTTTAGTTTTTTCTTTAGGCGGTTCTCGAAAAAAGATAGCGAAAAAAATTATCCCTAGAGTCGAGACTAAGAGGAATGTATAAACCAATGCTTCCATATATTCGATCGTGGTT</td>
</tr>
<tr class="even">
<td align="left">VIT_00s0873g00020</td>
<td align="left">QmlvRmVhdHVyZVZhbHVlVHlwZToyODU5OA==</td>
<td align="left">GGATCCTGGTCTGGACTCTTATGTGTGTATTTTATCTCGTTTGGCTTTTGGAATCACATCCTGCAGTGGTTGGAACAGCTCGCAAAATCCAAGCACTTCACCTACTTTATTGCCCCCAACCCTTTATCGTACCTTTCTTGAAACAGAATGGTTTCATGCTCCTTCATCGATTGGTTATTCCTCTTCGTTCGTCTCTTTTTTTTCAATGTCGGTCTCGATTAGTTCACAAGATTGAATGGCCAATTCTCCTCCGGACCCTCGATTCGATTATTTTCCAAGAATGTAATGTTAAGCCGGGTATGTATGTATCTGGGAGGAACTTTAAAGAAGGGCTTTCGGTTTTTTGCACCCTCTTTTGGTCTTGCAGCTTTATATTTATATTTCGTATATAATGAAGCGGGGGCTATCCGCCAGTAGCGTTGCGGATTCTCGATGAGGGGGTGGGTTCTTTCTTCAAACACTTTTTTCCGGTATGTCGCTCTGCGAGCAAGGAGCGAGAGAACCAAGTGGGCTATGGTGATGTCAGAATTTGCACCTATTTGTATCTATTTAGTGATCAGTCCGCTAGTTTCTTTAATCCCACTCGGTGTTCCTTTTCCATTTGCTTCCAATAGTTCGACCTATCCAGAAAAATTGTCGGCCTACGAATGTGGTTTCGATCCTTTCGGTGATGCTAGAAGTCGTTTTGATATACGATTTTATCTTGTTTCAATTTTATTTATTATTCCTGATCCGGAAGTAACCTTTTCCTTTCCTTGGGCAGTACCTCCCAACAAGATTGATCCGTTTGGATCTTGGTCCATGATGGCCTTTTTATTGATTTTGACAATTGGATCTCTCTATGAATGGAAAAGGGGTGCTTCGGATCGGGAGTAA</td>
</tr>
<tr class="odd">
<td align="left">VIT_00s0904g00010</td>
<td align="left">QmlvRmVhdHVyZVZhbHVlVHlwZToyODYwOQ==</td>
<td align="left">ATAACCTCTCATCATCTATCATTACAACCTGATCAATACCACAAGCTTGGGGTGAGTTCTAGGCTCTAGAAATGGCACAAGCTATGGCTTCAATGGCTGGGCTACGCGGGTCCTCTCAGGCCGTCCTGGAGGGCAGTCTCCAGATCAGTGGCTCAACCCGATTGAGCAGCGGCAGCAGCAGCAGGGTGGGTGTGGCACGCCCTGGGTTCAGTGTTAGAGCTCAGCAGGCTCCAGTTGAGCCTGAGACTAGCCGTCGTGCAGTGCTGGGCCTCGTTGCTGCTGGCCTGGCCTCGGGCTCTTTTGTTCAGGCTGTGCTTGCTGAGGCCAAGTCCATCAAGGTTGGCCCCCCTCCCCCACCCTCTGGTGGATTGCCTGGTACTCTAAACTCTGATGAGGCAAGGGACCTTGATCTGCCATTGAAGGACCGGTTCTTCATTCAGCCACTGCCCCCAGCTGCAGCAGCAGTTAGGGCGAAGGAGTCAGCTAAGGACATTGTTGGGGTGAAGACATTGATAGACAAGAAGGCTTGGCCATATGTCCAGATGGATCTCCGTCTCAAGGCAGAATATCTCCGGTACGACCTCAACACCGTCATTTCTGCAAAGCCCAAAGACGAGAAGAAATCCCTGAAAGAACTCACTGGAAAGCTCTTCTCGACCATCAGTGATTTGGACCATGCGGCAAAGATTAAAAGCTCCCCTGACGCAGAGAAGTACTATGCAGCGACTGTATCCACCCTGAACGATGTCCTTTCCAAGATTGGTTAAGAAGAATCCACTAATCTTGGTGTAAGCTTTAGTTTTCACTGCATGGGACATGTGCTTGGATGTTATTTTAAGGAAATCTAATTTGAAAAAAAAAAAAACCCCTCATGATA</td>
</tr>
</tbody>
</table>
<div id="get-ids-from-aliases-using-sparql" class="section level2">
<h2 class="hasAnchor">
<a href="#get-ids-from-aliases-using-sparql" class="anchor"></a>Get ids from aliases using SPARQL</h2>
<p>Genes might have different aliases, all of which are defined in VESPUCCI as annotations. Since annotations (for both genes and samples) are represented as RDF triples using different ontologies, we can perform a <a href="https://en.wikipedia.org/wiki/SPARQL">SPARQL</a> query to retrieve all the biological features (genes here) that has an <strong><em>alias</em></strong> (<a href="https://www.ebi.ac.uk/ols/ontologies/ncit/terms?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FNCIT_C41095">NCIT_C41095</a> term on the NCIT Ontology, <a href="http://purl.obolibrary.org/obo/" class="uri">http://purl.obolibrary.org/obo/</a>) equal to the ones specified in the list.</p>
<div class="sourceCode" id="cb6"><html><body><pre class="r"><span class="no">my_query</span> <span class="kw">&lt;-</span> <span class="st">"SELECT ?s ?p ?o WHERE { {?s &lt;http://purl.obolibrary.org/obo/NCIT_C41095&gt; 'B9S8R7'} UNION {?s &lt;http://purl.obolibrary.org/obo/NCIT_C41095&gt; 'Q7M2G6'} UNION {?s &lt;http://purl.obolibrary.org/obo/NCIT_C41095&gt; 'D7SZ93'} UNION {?s &lt;http://purl.obolibrary.org/obo/NCIT_C41095&gt; 'B8XIJ8'} UNION {?s &lt;http://purl.obolibrary.org/obo/NCIT_C41095&gt; 'Vv00s0125g00280'} UNION {?s &lt;http://purl.obolibrary.org/obo/NCIT_C41095&gt; 'Vv00s0187g00140'} UNION {?s &lt;http://purl.obolibrary.org/obo/NCIT_C41095&gt; 'Vv00s0246g00010'} UNION {?s &lt;http://purl.obolibrary.org/obo/NCIT_C41095&gt; 'Vv00s0246g00080'} UNION {?s &lt;http://purl.obolibrary.org/obo/NCIT_C41095&gt; 'Vv00s0438g00020'} UNION {?s &lt;http://purl.obolibrary.org/obo/NCIT_C41095&gt; 'Vv00s0246g00200'} }"</span>
<span class="no">sparql_results</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/get_sparql_annotation_triples.html">get_sparql_annotation_triples</a></span>(<span class="kw">target</span> <span class="kw">=</span> <span class="st">"biofeature"</span>, <span class="kw">query</span> <span class="kw">=</span> <span class="no">my_query</span>)
<span class="no">sparql_results</span>[,<span class="fl">1</span>]
<span class="co">#&gt;  [1] "QmlvRmVhdHVyZVR5cGU6MjcwOTc=" "QmlvRmVhdHVyZVR5cGU6Mjg1MDQ="</span>
<span class="co">#&gt;  [3] "QmlvRmVhdHVyZVR5cGU6Mjc1NTU=" "QmlvRmVhdHVyZVR5cGU6Mjg2MzA="</span>
<span class="co">#&gt;  [5] "QmlvRmVhdHVyZVR5cGU6MjY0MzM=" "QmlvRmVhdHVyZVR5cGU6MjY4ODM="</span>
<span class="co">#&gt;  [7] "QmlvRmVhdHVyZVR5cGU6Mjc0MjM=" "QmlvRmVhdHVyZVR5cGU6Mjc0MjY="</span>
<span class="co">#&gt;  [9] "QmlvRmVhdHVyZVR5cGU6MjgxODM=" "QmlvRmVhdHVyZVR5cGU6Mjc0Mzc="</span></pre></body></html></div>
<p>alternatively we can use the wrapper function <code><a href="../reference/get_ids_from_alias.html">get_ids_from_alias()</a></code>:</p>
<div class="sourceCode" id="cb7"><html><body><pre class="r"><span class="no">aliases</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">'B9S8R7'</span>,<span class="st">'Q7M2G6'</span>,<span class="st">'D7SZ93'</span>,<span class="st">'B8XIJ8'</span>,<span class="st">'Vv00s0125g00280'</span>,<span class="st">'Vv00s0187g00140'</span>,<span class="st">'Vv00s0246g00010'</span>,<span class="st">'Vv00s0246g00080'</span>,<span class="st">'Vv00s0438g00020'</span>,<span class="st">'Vv00s0246g00200'</span>)
<span class="no">sparql_results_check</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/get_ids_from_alias.html">get_ids_from_alias</a></span>(<span class="kw">target</span> <span class="kw">=</span> <span class="st">"biofeature"</span>, <span class="kw">alias_names</span> <span class="kw">=</span> <span class="no">aliases</span>)</pre></body></html></div>
<p>and check results are the same:</p>
<div class="sourceCode" id="cb8"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/identical.html">identical</a></span>(<span class="no">sparql_results</span>[,<span class="fl">1</span>],<span class="no">sparql_results_check</span>)
<span class="co">#&gt; [1] TRUE</span></pre></body></html></div>
<p>As before we can retrieve the sequence associate to our aliases using the ids from the database.</p>
<div class="sourceCode" id="cb9"><html><body><pre class="r"><span class="kw pkg">knitr</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span>(<span class="fu"><a href="../reference/get_biofeature_by_name.html">get_biofeature_by_name</a></span>(<span class="kw">name_In</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span>( <span class="fu"><a href="../reference/get_annotation_triples.html">get_annotation_triples</a></span>(<span class="kw">ids</span> <span class="kw">=</span> <span class="no">sparql_results</span>[,<span class="fl">1</span>])[,<span class="fl">1</span>]))))</pre></body></html></div>
<table class="table">
<colgroup>
<col width="0%">
<col width="1%">
<col width="97%">
</colgroup>
<thead><tr class="header">
<th align="left">name</th>
<th align="left">id</th>
<th align="left">value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">VIT_00s0207g00210</td>
<td align="left">QmlvRmVhdHVyZVZhbHVlVHlwZToyNjQzMw==</td>
<td align="left">CATTTCTGTCCAACTCCCCTGCCAACTCGCCAACAATGTCTCTCACCATCCCCACAAACCTCTCCAAGCCCCTCAAGCCAAGGCTCAGCTCTCCCTCCTTCCAAAAGGCAAGGCCTATGATAGTGTGCTCAGCCTCAACTGAGCCACAGCCCACCAACAAATCTCCACTCCAAGCATTCTCTGCTGCGCTTGCACTCTCCTCCATCATTCTAGCTGCTCCAGTCCCTGCCTATGCTGACATCGCCGGGCTCACACCCTGCAAGGAGTCCAAACAGTTCGCCAAGCGGGAGAAGCAGGCACTGAAGAAGCTGGAGTCATCGCTGAAATTGTACGCGCCGGACTCAGCCCCAGCGCTGGCCATAAACGCGACGATCGAGAAGACAAAGAGGAGGTTCACAAACTATGCGAACCAAGGGTTGTTGTGTGGGTCGGATGGGCTGCCACACTTGATAGTGAGCGGGGACCAGAGGCACTGGGGAGAGTTCATTACACCAGGGATATTGTTTTTGTACATTGCTGGATGGATAGGGTGGGTTGGAAGGAGCTATCTGATAGCCATCAGGGATGATAAGAAGCCAACACAGAAGGAGATCATCATTGATGTGCCTTTGGCTTCTAGGTTGGTCTTCAGAGGCTTTATTTGGCCTGTTGCTGCCTACAGAGAGTTCGTCAATGGTGATCTCATTGCCAAGGATGTCTAATTATTTCTCTTCTAAGTTCTACCATCACCACCACCAGTGGGAAAGACTTTGGCTGACAGGAAGACAAGGCCTGAGATTTGGGAGTTGCATGTGAGAGGAAGAGAGGAGTGGAGGCCAGTGGGGGAGAGGTGAAAGATTTCTTGTATTTATTTTGTAAACTTAATTGTATCTAATGCTTTGTGTTGAACACTAAGTCTGCCTTGTCTTTCCAT</td>
</tr>
<tr class="even">
<td align="left">VIT_00s0733g00010</td>
<td align="left">QmlvRmVhdHVyZVZhbHVlVHlwZToyNjg4Mw==</td>
<td align="left">CTTGATGGGTCAAAAATGCCCAACTTATCAATTTATTCGGTAAACTATTTGGGGCTCATGTAGCCCATGTCGGGTTAATCGTATTTTGGGTCAGAACAATAAACTTATTTGAAGTGACTCATTTCGTACCAAAGAAGCCCATGTATGAAAAATGATTAATTTTACTTCCCTACCTAGTTACTCTAGATTGGGGGTAGGTCCTGATGGGGAAGTTATAGACACCTTTTCATACTTTGTATTTGGAGTATTTCACTTAATTTTCTCCGCAGCATTAGACTTTGGCGGTATTTATCATGCAGAGAAGCTAGATTCACAAAATTTGATATTATTTTATATATGA</td>
</tr>
<tr class="odd">
<td align="left">VIT_00s2608g00010</td>
<td align="left">QmlvRmVhdHVyZVZhbHVlVHlwZToyNzA5Nw==</td>
<td align="left">ATAGTTTGTAAGGTGGTAGGCAGAGACTCCTCCACACTCACTAGCCAGTGATAGCAAAGAAGAGAAAAAAATGGCAGCCTCATTGCAAGCAGCGGCCACGCTGATGCAGCCCACCAAGGTGGGCGTGCCAACTCGTACCAGCCTGCAATTGAGATCTTCCCAGAGTGTATGCAAGGCCTTTGGCTTGGAACCCGCTGGAGCCAGGTTCACTTGCTCTTTGCAGGCTGATCTCAAGGACTTGGCTCACAAGTGCGTGGATGCCACTAAGATCGCCGGCTTTGCTCTTGCCACTTCTGCTCTCCTCGTCTCGGGAGCAAGTGCAGAAGGAGCTCCAAAGCGGCTGACATACGATGAGATCCAGAGTAAGACATACATGGAAGTAAAGGGAACTGGAACTGCTAACCAATGCCCAACCATTGACGGAGGGGTCGATTCATTTGCCTTCAAGTCTGGAAAATACCAACTGAAGAAGTTCTGCTTAGAGCCTACATCATTCACCGTCAAGGCTGAGGGGGTGAGCAAGAATGCTCCACCTGAATTCCAAAACACCAAGCTCATGACTCGGTTAACATACACCCTTGATGAGATTGAGGGACCATTTGAAGTATCTAGTGATGGCACCGTTAAGTTTGAGGAGAAAGATGGAATTGACTATGCTGCTGTCACTGTTCAGCTGCCCGGTGGTGAGCGTGTGCCCTTCCTCTTCACCATCAAACAGCTCGTTGCATCCGGCAAACCAGACAAATTTGGAGGAGAGTTTCTGGTACCATCCTACCGGGGTTCGTCCTTCTTGGACCCAAAGGGAAGGGGTGGTTCAACTGGCTATGACAACGCAGTTGCATTGCCAGCAGGCGGGAGAGGAGATGAGGAAGAACTCAAAAAGGAGAATATAAAGGACGCGTCATCGTCAACAGGGAAAATCACTTTAAGTGTCACCAAAAGCAAGCCTGAGACTGGGGAAGTGATTGGAGTATTTGAGAGCATTCAGCCATCTGACACTGACTTGGGGGCTAAGACACCAAAGGAAGTGAAGATCCAGGGGTTCTGGTACGGTCAGCTTGAGTCGTAGAGCCTTTTCTTTATGGATGTATCTCTCGGTTCAGATTTTGTTTTAAGCTCAAAGGTGTTGCTAAGACTTCCCTGTGACCCCGGACAAGCTGAAACTATTATCTTGTTTGTATATTATTTATTATCATGGCTTCTAATGTAATGTTTGGAGATTTCATAAA</td>
</tr>
<tr class="even">
<td align="left">VIT_00s0965g00010</td>
<td align="left">QmlvRmVhdHVyZVZhbHVlVHlwZToyNzQyMw==</td>
<td align="left">TGGGAGGACCCTACTCTCCTAATGCCACTCATTAGGGAATACCGGAAAGACTCATCTTCTTAGTCTTTCTTTTCTGCCGCTTTCTTTCATAACAGAGCCGAGAATAACGGCTGGCATAGAAGTCTCTCACGCGCAAGGAGATGCAGCTGCTGGATGTCACGGTTCTGGGGCGCCATGATCCTTCTCTCTGGAACAATCGAGGGCACTGCCTTCCTTCAGCTTTCAGAGGTAGGGGCTGCATCAATCATCGCTCAGTGAGCTATTTATTGCCACCAATAGCGCTTCGCTTGCATGCAAGGCGGCACGCCAGGTAGAGCTATCGCGCGCGGGCGAAGGAGATGCATTTCTGGTACTGGTAGTACTGGACAAGCTCTCAGGGAATAATCTCTTTCTTATTTCTGCCTTTCTTTCCCATGACGACTAGGAACGGGCAAATCACAAATTTCACTTCGAATTCCGGACCTCAACATCCTGCTGCTCATGGTGTTTCACGATCAGTATTGGAAATGAACGGAGAAGTGGTGGAACGTGCGGAACCACATATTGGATCACTCCAAGGGACTGAGAAATTAATAGAGTACAAAACTTATCTTCAAGCTTTACCTTATTCTGATCGTTCAGAGGGCGATCGCGGAGAGATAACAAGTAGAGCCAACGACCCACTAAGACTAACCGAGTGCACCGGAAAAGAACGGGAACTGGGTCGATCTATTCTATGGCGAAGCATCCGAAGCATAACTGCACACTCACACGATCTTTGCCGAGAGATAGGAGCATTCGGTGGAACCGGTGAACTACACTTGCTTCTGGATAGATGTGTGGGACAGAGGGCTCGTGGTACCTGCTGCCCACCCTTCCCCCTCTGCTTTGAGAACCGATTGGGGATTCAGTGGTGTAATGTTAAGAGGTCGTGCGACATGAAGACATTGATAGCAATATGGGGGAAGTTCCCATCAGGCAACAACGCCTCAAAAAGCGAAGGCCGAGCTGTAGGTCACGTGACCTGCACCGAGTTGGTGGCTGACTGGGCTTTTTCCTTGATCAAAGCAGATCCGCTCGCCTTCTTGTTATCCAAAAGTAGGTCGAATCGGGCTTGCCCCGGAACGTCGAATGAAATAGCGAACGGTCAGCGTGAAGGTTCAGGACTTAGCCGAGCGTTAGCGAAGCTAGATTCTCATAGCAAGGCGCTTCGAGTTAGCAAAGTGCTGTCTGGAGAGAGATGGGACCGGTTATATAATAGGGGGAGCAGATGCAAGCTTTTTCTTTCAATAGCCGGCCAAATGACTACAGGATCATCGGTCTACTCTACCTCAATTCACCATTTCGAACCTTATACAGAAGGTTTTTTCGTACCAGCTCCTTCTACCTATACCGCAGTTGAAGCACCTAAAGGAGAATTTGGTGTCTTTCTGGTCAGTAATGGAAGCAATCGTCCCTACCGTCGTAAAATAAGAGCACCTGGCTCTGCCCATTCACAAGGACTCGATTCTATGTCCAAACATCACATGCCAGCAGATGTGGTCACCATCATAGGTACTCAAGATATTGTGTCTGGAGAGGTAGATAGATAGGACTACTAGTTGCTCAATCAGGACCCTAGCTTTATTGCGAGCCAAAAACCTTGAT</td>
</tr>
<tr class="odd">
<td align="left">VIT_00s0125g00280</td>
<td align="left">QmlvRmVhdHVyZVZhbHVlVHlwZToyNzQyNg==</td>
<td align="left">ATGAGGGTGATCTACGATCAGGCTGGTCGGAACGATCAGGGGGAAGTAGGGCTCCTATTGAAACACTTTCTCTCCCTCAAAAGGCGACCAGCTTTCAATACTAGTTGTTACGTTACGCTTTTCGTTGTTCGTTCCGTCGTCGACCCTATCTATGTTTCTACGATGGCCCAAGAGCACGCTCATTCTTCAGCCGTAGAGAGACTTTTGAATTGCGAGGTACCATTACGAGCTCAATATATACGAGTGTTATTCCGTGAAATAACTCGAATTTCAAATCATTCACTTGCTTTAACTACTCATGCTATGGATGTGGGAGCATCAACTCCGTCCCTGTGGGCTTCCGAGGAGCGGGAGAAATTGTTGGAATTCTATGAAAGAGTCTCGGGAGCCAGGATGCATGCCAGTTTCATACGACCAGGTGGAGTGGCACAAGATCTGCCTCTTGGCTTATGTCACGTGACCTGCACCGAGTTGGTGGCTGACTGGGCTTTTTCCTTGATCAAAGCAGATCCGCTCGCCTTCTTGTTATCCAAAAGTAGGTCGAATCGGGCTTGCCCTTCCCCCTATCCCTTATGTTTAGGAGCGGGATCCGCCCAAGAACGACCAGTCTTGTGGTGGTACGGAAGGCACGGACTCCGCGAACGGGTATGTTGGGATTCGCGAAGAGCAGCACCTTACGATGTTCATGACCAATCGGATCCTGACGTACCAGTAGGTACCAGAGGGGATCGCTATGATCGTTACTGTATCCGTATCGAAGAGATGCGACAAAGTGTTCGGATCATTGTGCAATGTCCTAATCAAATGCCTAGTGGCATGATCAAAGCCGATGATCGTAAGCTATGTCCTCCATCACGATGTCGAATGAAACTATCCATGGAATCGTGCGCCGTGTGAAACGTAGATCATCGCCGTTCTTAACCGAGACTCAGGTTAAGCTCCGTCTCGGAACCTTGTGGGGTTAGGAGTAAAGCATCCCGAGGTTGGCGCATCTCGTTGGGCGTGGAGAAGCATTGGGAACCCCAATTTCTTTCTTCGGAGCCGTTTCTTTTCCCGTCCCCCCGCCCCGGCATAGCGCTTCGCTTCCGGTTCTTCGGAAGAATCAACTGACTTCTCCCTTCTTCATTGATCTGGGGGAAAAGGAACCGTCTACCAGTTGGGAAGCTAGACATCAAGTAAGTGGCTTGATGAGTATAACTAAGCTGACATGCCGGAGTTGGCTG</td>
</tr>
<tr class="even">
<td align="left">VIT_00s0187g00140</td>
<td align="left">QmlvRmVhdHVyZVZhbHVlVHlwZToyNzQzNw==</td>
<td align="left">AATGAATCGAGATGATAAGGAGTGATCAATGATGCTCGAACATGTACTGTTTGAGTGCCTATTTATTTTCTATCGGTATCTATGGATTGATCATGAGTCGAAATATGGTTAGAGCTCTTATGTGTCTTGAACTTATACTGAATGCAGTTAATATAAATTTCATAACATTTTCTGATTTTTTTTATAATCGACAATTAAAAGGAGACATTTTATCAATTTTTGTTATAGCTATTGCAGCCGCTGAAGCTGCTATTGGACCAGCTATTGTTTCATCAATTTATCGTAATAGAAAATCAACTCGTATCAATCAATCGAATTTGTTGAATAAGTAGTATTAATATAGAAATTTGAATATTCATCAATTCAAATTAGCATGTATAATACGTAACGTATGATTATGTTTGCGAAAACGTAAGAAATCAATCAAAGTATCTTGGCCCTCTCTCATGAGTCGATCCAGAAGTAAATTGATTAGAAAAATTCTGGTAAGTCATTGATTCATCTAGTGTTTAGATATATGATTCATTTATAAATTTACTAGATCGAAATTTTATGACGTTTAAAAATTTATAGATCCAATGTCACATTCAGTAAAGATTTATGATACATGTATAGGGTGTACTCAATGTGTCCGTGCCTGCCCCACAGATGTATTAGAAATGATACCTTGGGACGGATGTAAAGCTAAGCAAATTGCTTATGCTCCAAGGACAGAAGACTGCGTCGGTTGTAAGAGATGTGAATCTGCTTGTCCAACAGATTTCTTGAGTGTTCGGGTTTATTTATGGCATGAAACAACTCGCAGCATGGGTCTAGCTTATTGA</td>
</tr>
<tr class="odd">
<td align="left">VIT_00s0246g00010</td>
<td align="left">QmlvRmVhdHVyZVZhbHVlVHlwZToyODE4Mw==</td>
<td align="left">TACAAGATCGAAAAGAATGCATTGGATTCCAAGTGAGATGCCCAAGATAAAAGGAACGAGGTTGGTCTACGATCTTAAGCAGGCTGATTGTAAATCATTTTATCCGAGAGGGCGAATCCATAACAAAAAATCGTGAAGAAAAAAGATAGCGTGACGAGAATTCTCAACTCGAGATGTTAGAAGGTGCAAAATCAATGGGTGCCGGAGCTGCTACAATTGCTTCAGCGGGAGCTGCAATCGATATTGGAAACGTGTTCAGTTCTTTGATCCATTCCATGGCACGAAATCCATCATTGGCTAAACAATCATTTGGTTATGCCATTTTGGGCTTTGCTCTAACCGAAGCTATTGCATCGTTTGCCCCAATGATGTCCTTTCTGATCTCATCCGTATTCTGATCCATTGGACTTTTAGCGGATTCAGAGTTAGGTGGGGCCCTCTCATTCCTTAGGGAGTGAGAGGGGTAAGGGGAAAGGGTGGGTGGCCCCTTGCGTTAGGATGAGTCCAGTGTAGTCGGATGGAACGGAAGCGCATATCAGAAACGAATCTGGCATCTATGGAACCCACGTGCGAACCTTCGGAGCTCGAAGTGAAACGATAACATCCGGAACTTTGAGTTTTCTATGGTAAACTTAGGCGAGTCCTTATAGACTACCTAGTTATGACACCTTGATGCACGTTGGTGCATTGGGGTGCACTGGTTGACGCGGAAGGCCAGCGTGGGGCCATTATAATGAATTAGGTCACATGATCGGGAGTCGTCTGGGTGGGTAGGTCTATCCATACCCCC</td>
</tr>
<tr class="even">
<td align="left">VIT_00s0246g00080</td>
<td align="left">QmlvRmVhdHVyZVZhbHVlVHlwZToyODUwNA==</td>
<td align="left">GTCAAAACAAAAAGCGACATTCTCCCTTGACTCAATCTTTGAATTCATTATGGAATTCTCTCCCAGAGCTGCGGAACTCACGACTCTATTAGAAAGTAGGATTACCAACTTTTACACGAATTTGAAAGTGGATGAGATCGGTCGAGTGGTCTCAGTTGGAGATGGGATTGCACGTGTTTATGGATTGAACGAGATTCAAGCTGGGGAAATGGTTGAATTTGCCAGCAGTGTGAAAGGAATAGCCTTGAATCTTGATAATGAGAATGTAGGGATTGTTGTCTTTGGTAGTGATACCGCTATTAAAGAAGGAGATCTTGTCAAGCGCACTGGATCTATTGTGGATGTTCCTGCGGGAAAGGCTATGCTAGGGCGTGTGGTCGACGCGTTGGGAGTACCTATTGATGGAAGAGGGGCTCTAAGCGATCACGAGCGAAGACGTGTCGAAGTGAAAGCCCCTGGGATTATTGAACGTAAATCTATGCACGAGCCTATGCAAACAGGCTTAAAAGCGGTAGATAGCCTGGTTCCTATAGGTCGTGGTCAACGAGAACTTATAATTGGGGACCGACAAACTGGAAAAACAGCTATAGCTATCGATACCATATTAAACCAAAAGCTTTTGAACTCAAAGGCCACCTCTGATAGTGATACATTGTATTGTGTCTATGTAGCAATTGGACAGAAACGCTCAACTGTGGCACAATTAGTTCAAATTCTTTCAGAAGCGAATGCTTTGGAATATTCCATTCTTGTAGCAGCCACCGCTTCGGATCCTGCTCCTCTGCAATTTCTGGCCCCATATTCTGGTTGTGCCATGGGGGAATATTTCCGCCATAATGGAATGCACGCATTAATAATCTATGATGATCTTAGTAAACAGGCGGTGGCATATCGACAAATGTCATTATTGTTACGCCGACCACCAGGCCGTGAGGCTTTCCCAGGCGATCTCTTTTATCGCGGAATTAGACCTGCTATTAACGTCGGCTTATCTGTCAGTCGCGTCGGGTCTGCCGCTCAGTTGAAAGCTATGAAACAAGTCTGCGGTAGTTCAAAATTGGAATTGGCACAATATCGCGAAGTGGCCGCCCTTGCTCAATTTGGGTCAGACCTTGATGCTGCGAATCAGGCATTACTCAATAGAGGTGCAAGGCTTACAGAAGTCTCGAAACAACCACAATATGCACCACTTCCAATTGAAAAACAAATTCTAGTCATTTATGCAGCTGTCAATGGATTCTGTGATCGAATGCCACTAGACAGAATTTCTCAATATGAGAGAGCCATTCCAAGTAGTATAAAACCAGAATTTCTACAATCCCTTTTAGAAAAGGGTGGGTTAACTAATGAAAGAAAGATGGAACCAGATGCATTCTTAAAAGAAAGCGCTTTGCCTTACCTATGA</td>
</tr>
<tr class="odd">
<td align="left">VIT_00s0438g00020</td>
<td align="left">QmlvRmVhdHVyZVZhbHVlVHlwZToyODYzMA==</td>
<td align="left">ATTGGGTAGCAGATAAGAGAAGGCGGACTTCTTCAACTCCTCACAAAGACCAAATTACAAACTCATGGCTGTTTCATCAACAGCACCGCCATGTTACTCTCTCTCTACCCTCCATTCTTCCCTTTTCCCTTCCTTCAAATCACGCTCTCTTCTTTCGTTTTCTTCCAAGCCAAAGCGCGCCCAAAGGCGTTTTGTTGTTCAGGCATCTGCAACTCTTGAGTCCAGCAATGGTGCCATTGTTGCTGAAGAGTCCGACAACAGCACTTCATATGGAAGGCAGTTCTTCCCATTGGCTGCCGTGGTTGGTCAGGATGCTATTAAAACTGCTCTGTTGCTTGGGGCTATTGACCGTGAGATCGGAGGGATTGCCATCTCTGGGAAAAGAGGAACTGCCAAGACAGTAATGGCACGTGGACTGCATGCAATTTTGCCCCCAATTGATGTTGTTGTTGGTTCAATAGCAAATGCTGATCCCACCCGCCCAGAAGA</td>
</tr>
<tr class="even">
<td align="left">VIT_00s0246g00200</td>
<td align="left">QmlvRmVhdHVyZVZhbHVlVHlwZToyNzU1NQ==</td>
<td align="left">ATAAACAAAAATACCAAATGAAAGAATAAGTTCAATTGACAGTTTTTGATTCATCAATCAATTATTATAGGCGGCACTAACAAAGATAGAGTGACAGAGGTATAAAAAGAAATCAATTCATTATCTGTGGCATCTACCCATACCGTGATTCCGAATCAACAGACCGAATTTCCGAGATATGGCAACTCTTGAGTAAACTAATAATAAAATAAGAACTGTGCCGTGTAACTTCATTCACAAATGAAACTTTCTTTGCATCACTATGAAACAAAATTATAAAATCATTTCTATTTTGATCATTTCTTTTTTAATTGTATCGAATACATTTTCTATTTTGGAACGAAGAGTGCCCTTATAACAATAAAATCTTTTCTTTTACTTTTTACTTTAATTTTAACTCATTAGATTCAATAGTAGGTTCATTCACATAGTATCTCGAATGAGGAAAAAAAAAAAAAGATATCGCACGATCAGATCACTCGAAAAAAGAAAATCTGTATTTTGGTTCAATTATATTCTAAAAAATTCCTATTATCTTTTCCTTTATAGGTTCCACATTTTGTCTTTCCATATTATAACTTCTAGTTAGGTAGTTAGGTCAAGAAAGAACCCTTAGATCTAATACAACAATGCGTGCTTCGCGAATATTGATTGTTCCCATTATTTTATTCATTGTTCTCTTTCTTTCATCGAATACTATCTACTACTGTTACTTGTTATTGGATAACTAAGAAATTCTTTAAAATGAAAAAAATAAAGATTCAAAAAAAAAACCTCTTCTACAACCACAAAAACGAAAAGGCGATTTCTAGATTTCGTATCAATAAAATTGTGGGAATATGATACTCTCTTTCATGAATTATTAAAAAGAAATTTGTCAGATTCACATTTATTTTACCTGATCTTCAGTTTCTAGTCCGAAACCAATAGTAGAGAGAAACCCTATACTACAAGAATTTGAGAGATTTTCTATTGAATGGCACATGGTTCTACATTGATAAGCAACAAGAAAGGAAGAAAGAACTTTTCTAACACTTCATTTCGATACTGATTGAAATTGCGTTGCTGTGGCAGAAGAAGGATAGATATACTGATTCGGTATACTCTAAAGACACCCTCGGTACAATATTGACGATCTCACAAATATCCATTTTCAGTGAATTTCTATTTACTGATCTCATCTTTTACGAAATCGATCGTCCTTTGACTGTACAAGAATATGTGGAGCTGGGCATGTCTGGAAGCACAGGAGAACGTTCTTTTGCTGATATTATTACCTGTATTCGATACTGGGTCATTCATAGCAAACCTATACCTTCCCTATTTACTCGACGAATTAAGGGGGTTAGTTATTCGTCAGCACGAGTTTTGCTTCCTCTTCTGGTATAGGTTTGGAAGTCCTCGGCCAAACGAGTATTTTACAGAGAGCCGACAAGGAATGGAATTAATAACCGGCCGTTTTGATCCTTTTGTTCCGGGAGGAAGAGCCTCTCATCATATTGCAGCAAGGACATTGGGCATATTAGCAGGGCTATTCCATCTTAGTGTCCGTCCGCCCCAACGTCTATACAAAGGATTACGTATGGGCAATATTGAAACCGTCCTTTCCAGTAGTATCGCTGCTGTCTTTTTTGCAGCTTTTGTTGTTGTTGGAACCATGTGGTATGGTTCAGCAACTACCCCGATCGAATTATTTGGTCCCACTCGTTATCAATGGGATCAGGGATACTTCCAGCAAGAAAGATATCTAAGAGTTGGTGCTGGGCTAGCCGAAAATCAAAGTTTTTCAGAAGCTGGGTCTAAAATTCCTGAAAAATGAGCTTTTTATGATTACTTCGGCAATAATCCGGCAAAAGGGGGATTATTCAGAGCGGGCTCAATGGACAACGGGGATGGAATAGCTGTTGGATGGTTAGGACACCCTATCTTTAGAGATAAATAAGGGCGTGCACTTTTTGTACGTCGTATGCCTACTTTTTTTGAAACATTTCCAGTCGTTTTGGTAGACGGAGACGGAATTGTTAGAGCCGATGTTCCTTTAAGAAGGGCAGAATCGAAGTATAGTGTCGAACAAGTAGTGTAACTGTTGAGTTCTATGGCGGCGAACTCAATGGAGTCAATTATAGCGACCCTGCTACTGTGAAAAAT</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="create-a-module" class="section level1">
<h1 class="hasAnchor">
<a href="#create-a-module" class="anchor"></a>Create a Module</h1>
<p>We are now ready to create our first module. If we pass a list of genes (BiologicalFeature objects) it will automatically retrieve the “best” conditions (SampleSet objects).</p>
<div class="sourceCode" id="cb10"><html><body><pre class="r"><span class="no">module_1</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/create_module.html">create_module</a></span>(<span class="kw">biofeaturesNames</span><span class="kw">=</span><span class="no">gene_names</span>, <span class="kw">normalization</span> <span class="kw">=</span> <span class="st">"legacy"</span>)</pre></body></html></div>
<p>A module can be thought as a matrix composed by BiologicalFeature objects (genes in our case) as rows and SampleSet objects (conditions contrasts in our case) as columns. Each SampleSet, as the name suggests, is composed by 1 or more Sample objects. Since we are using the legacy normalization each SampleSet is composed by 2 samples, one Test sample and one Reference sample and the value associated is the logratio between the Test and Reference for a specific gene (BiologicalFeature).</p>
<div class="sourceCode" id="cb11"><html><body><pre class="r"><span class="kw pkg">knitr</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span>(<span class="no">module_1</span>, <span class="kw">class</span> <span class="kw">=</span> <span class="st">'cell-border stripe'</span>)</pre></body></html></div>
<table class="table">
<colgroup>
<col width="0%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="0%">
<col width="1%">
<col width="0%">
<col width="1%">
<col width="0%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="0%">
<col width="1%">
<col width="1%">
<col width="0%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="0%">
<col width="1%">
<col width="1%">
<col width="0%">
<col width="0%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="0%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="0%">
<col width="0%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
</colgroup>
<thead><tr class="header">
<th align="left"></th>
<th align="right">GSM1532841.ch1-vs-GSM1532832.ch1</th>
<th align="right">GSM1532842.ch1-vs-GSM1532832.ch1</th>
<th align="right">GSM881592.ch1-vs-GSM881586.ch1</th>
<th align="right">ERR3025271-vs-ERR3025258</th>
<th align="right">GSM1532843.ch1-vs-GSM1532832.ch1</th>
<th align="right">ERR3025268-vs-ERR3025256</th>
<th align="right">GSM881594.ch1-vs-GSM881586.ch1</th>
<th align="right">ERR3025266-vs-ERR3025256</th>
<th align="right">GSM1957885.ch1-vs-GSM1957878.ch1</th>
<th align="right">GSM1565329.ch1-vs-GSM1565328.ch1</th>
<th align="right">GSM2572390.ch1-vs-GSM2572412.ch1</th>
<th align="right">GSM1565337.ch1-vs-GSM1565328.ch1</th>
<th align="right">GSM881593.ch1-vs-GSM881586.ch1</th>
<th align="right">GSM1565333.ch1-vs-GSM1565328.ch1</th>
<th align="right">GSM1957882.ch1-vs-GSM1957878.ch1</th>
<th align="right">SRR866576-vs-SRR866568</th>
<th align="right">GSM1532878.ch1-vs-GSM1532868.ch1</th>
<th align="right">GSM1957878.ch1-vs-GSM1957815.ch1</th>
<th align="right">ERR3025260-vs-ERR3025252</th>
<th align="right">GSM1957845.ch1-vs-GSM1957833.ch1</th>
<th align="right">GSM1532890.ch1-vs-GSM1532880.ch1</th>
<th align="right">GSM1654664.ch1-vs-GSM1654646.ch1</th>
<th align="right">GSM2572439.ch1-vs-GSM2572415.ch1</th>
<th align="right">GSM2627799.ch1-vs-GSM2627784.ch1</th>
<th align="right">GSM2627717.ch1-vs-GSM2627691.ch1</th>
<th align="right">GSM1532877.ch1-vs-GSM1532868.ch1</th>
<th align="right">GSM2627802.ch1-vs-GSM2627784.ch1</th>
<th align="right">GSM2627828.ch1-vs-GSM2627823.ch1</th>
<th align="right">GSM1020501.ch1-vs-GSM1020496.ch1</th>
<th align="right">ERR3025264-vs-ERR3025254</th>
<th align="right">GSM881561.ch1-vs-GSM881550.ch1</th>
<th align="right">GSM2627843.ch1-vs-GSM2627823.ch1</th>
<th align="right">ERR3025265-vs-ERR3025254</th>
<th align="right">ERR3025263-vs-ERR3025254</th>
<th align="right">GSM1957879.ch1-vs-GSM1957878.ch1</th>
<th align="right">GSM1532879.ch1-vs-GSM1532868.ch1</th>
<th align="right">GSM2627738.ch1-vs-GSM2627691.ch1</th>
<th align="right">GSM1957886.ch1-vs-GSM1957878.ch1</th>
<th align="right">GSM2572437.ch1-vs-GSM2572412.ch1</th>
<th align="right">GSM2627886.ch1-vs-GSM2627823.ch1</th>
<th align="right">GSM1654673.ch1-vs-GSM1654646.ch1</th>
<th align="right">GSM1532886.ch1-vs-GSM1532880.ch1</th>
<th align="right">GSM2572440.ch1-vs-GSM2572415.ch1</th>
<th align="right">GSM1020450.ch1-vs-GSM1020442.ch1</th>
<th align="right">GSM1654672.ch1-vs-GSM1654664.ch1</th>
<th align="right">GSM2650959.ch1-vs-GSM2650955.ch1</th>
<th align="right">GSM881559.ch1-vs-GSM881550.ch1</th>
<th align="right">GSM881525.ch1-vs-GSM881520.ch1</th>
<th align="right">GSM2627811.ch1-vs-GSM2627784.ch1</th>
<th align="right">GSM2627796.ch1-vs-GSM2627784.ch1</th>
<th align="right">GSM1654670.ch1-vs-GSM1654664.ch1</th>
<th align="right">GSM1957889.ch1-vs-GSM1957878.ch1</th>
<th align="right">GSM2572391.ch1-vs-GSM2572415.ch1</th>
<th align="right">GSM2441412.ch1-vs-GSM2441409.ch1</th>
<th align="right">GSM1020449.ch1-vs-GSM1020442.ch1</th>
<th align="right">GSM1532876.ch1-vs-GSM1532868.ch1</th>
<th align="right">SRR3999588.ch1-vs-SRR3999581.ch1</th>
<th align="right">GSM2627721.ch1-vs-GSM2627691.ch1</th>
<th align="right">ERR3025269-vs-ERR3025258</th>
<th align="right">GSM1654680.ch1-vs-GSM1654673.ch1</th>
<th align="right">GSM2583804.ch1-vs-GSM2583799.ch1</th>
<th align="right">GSM1020468.ch1-vs-GSM1020460.ch1</th>
<th align="right">SRR5851301-vs-SRR5851297</th>
<th align="right">SRR5876345-vs-SRR5851297</th>
<th align="right">GSM1532819.ch1-vs-GSM1532808.ch1</th>
<th align="right">GSM1957887.ch1-vs-GSM1957878.ch1</th>
<th align="right">GSM1654666.ch1-vs-GSM1654664.ch1</th>
<th align="right">GSM2627728.ch1-vs-GSM2627691.ch1</th>
<th align="right">GSM2627709.ch1-vs-GSM2627691.ch1</th>
<th align="right">GSM2650962.ch1-vs-GSM2650955.ch1</th>
<th align="right">GSM712566.ch1-vs-GSM712552.ch1</th>
<th align="right">GSM1020503.ch1-vs-GSM1020496.ch1</th>
<th align="right">SRR5332106.ch1-vs-SRR5332108.ch1</th>
<th align="right">GSM2627736.ch1-vs-GSM2627691.ch1</th>
<th align="right">GSM2583811.ch1-vs-GSM2583799.ch1</th>
<th align="right">GSM1532866.ch1-vs-GSM1532856.ch1</th>
<th align="right">GSM2627800.ch1-vs-GSM2627784.ch1</th>
<th align="right">GSM2650948.ch1-vs-GSM2650955.ch1</th>
<th align="right">GSM1020500.ch1-vs-GSM1020496.ch1</th>
<th align="right">GSM1020548.ch1-vs-GSM1020541.ch1</th>
<th align="right">GSM881677.ch1-vs-GSM881670.ch1</th>
<th align="right">GSM2572397.ch1-vs-GSM2572421.ch1</th>
<th align="right">GSM1532853.ch1-vs-GSM1532844.ch1</th>
<th align="right">GSM1654671.ch1-vs-GSM1654664.ch1</th>
<th align="right">GSM2627719.ch1-vs-GSM2627691.ch1</th>
<th align="right">GSM2572457.ch1-vs-GSM2572415.ch1</th>
<th align="right">GSM1532889.ch1-vs-GSM1532880.ch1</th>
<th align="right">GSM2627741.ch1-vs-GSM2627691.ch1</th>
<th align="right">GSM1654685.ch1-vs-GSM1654682.ch1</th>
<th align="right">GSM2627812.ch1-vs-GSM2627784.ch1</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">VIT_00s0246g00220</td>
<td align="right">4.1394</td>
<td align="right">3.4110</td>
<td align="right">3.03280</td>
<td align="right">8.799400</td>
<td align="right">3.4077</td>
<td align="right">2.7819</td>
<td align="right">3.3710</td>
<td align="right">2.7779</td>
<td align="right">-1.1606</td>
<td align="right">2.1521</td>
<td align="right">-1.968900</td>
<td align="right">2.1642</td>
<td align="right">2.16560</td>
<td align="right">1.9676</td>
<td align="right">-0.42252</td>
<td align="right">2.3707</td>
<td align="right">2.58780</td>
<td align="right">0.010966</td>
<td align="right">1.4741</td>
<td align="right">0.93984</td>
<td align="right">2.2998</td>
<td align="right">-3.41670</td>
<td align="right">1.436000</td>
<td align="right">2.57930</td>
<td align="right">-1.9428</td>
<td align="right">2.78550</td>
<td align="right">2.5028</td>
<td align="right">-2.37120</td>
<td align="right">-2.35980</td>
<td align="right">1.60270</td>
<td align="right">0.61787</td>
<td align="right">-1.43830</td>
<td align="right">1.34610</td>
<td align="right">1.46090</td>
<td align="right">-1.51810</td>
<td align="right">2.26080</td>
<td align="right">-0.77284</td>
<td align="right">-1.2145</td>
<td align="right">-0.73124</td>
<td align="right">-2.2577</td>
<td align="right">-3.18830</td>
<td align="right">2.02220</td>
<td align="right">1.47950</td>
<td align="right">-1.56850</td>
<td align="right">3.11990</td>
<td align="right">0.80889</td>
<td align="right">0.49742</td>
<td align="right">0.47349</td>
<td align="right">2.04220</td>
<td align="right">2.2071</td>
<td align="right">3.051200</td>
<td align="right">-0.60964</td>
<td align="right">3.148700</td>
<td align="right">1.40410</td>
<td align="right">-1.2813</td>
<td align="right">2.57620</td>
<td align="right">2.473800</td>
<td align="right">-1.45590</td>
<td align="right">1.39760</td>
<td align="right">2.97630</td>
<td align="right">-2.18660</td>
<td align="right">-0.33552</td>
<td align="right">-0.11379</td>
<td align="right">0.12534</td>
<td align="right">1.53260</td>
<td align="right">-0.78306</td>
<td align="right">2.93930</td>
<td align="right">-1.09680</td>
<td align="right">-1.83520</td>
<td align="right">0.81107</td>
<td align="right">3.47650</td>
<td align="right">-1.42750</td>
<td align="right">-1.30060</td>
<td align="right">-0.58471</td>
<td align="right">-2.79220</td>
<td align="right">1.87950</td>
<td align="right">2.07450</td>
<td align="right">0.36720</td>
<td align="right">-1.7487</td>
<td align="right">2.48560</td>
<td align="right">3.14550</td>
<td align="right">-0.51020</td>
<td align="right">0.99858</td>
<td align="right">3.00120</td>
<td align="right">-1.30730</td>
<td align="right">2.116200</td>
<td align="right">1.58190</td>
<td align="right">-0.76832</td>
<td align="right">2.63270</td>
<td align="right">1.62780</td>
</tr>
<tr class="even">
<td align="left">VIT_00s0332g00060</td>
<td align="right">4.0925</td>
<td align="right">3.0663</td>
<td align="right">3.35400</td>
<td align="right">0.000000</td>
<td align="right">3.0167</td>
<td align="right">2.2259</td>
<td align="right">3.4117</td>
<td align="right">2.2222</td>
<td align="right">-3.0668</td>
<td align="right">NA</td>
<td align="right">-2.783700</td>
<td align="right">NA</td>
<td align="right">2.12430</td>
<td align="right">NA</td>
<td align="right">-3.27770</td>
<td align="right">2.0919</td>
<td align="right">2.65020</td>
<td align="right">3.311700</td>
<td align="right">1.6846</td>
<td align="right">2.54100</td>
<td align="right">2.5995</td>
<td align="right">-1.92510</td>
<td align="right">2.943800</td>
<td align="right">1.79700</td>
<td align="right">-1.1237</td>
<td align="right">1.63490</td>
<td align="right">1.5554</td>
<td align="right">-2.29180</td>
<td align="right">-1.68250</td>
<td align="right">1.26910</td>
<td align="right">3.22190</td>
<td align="right">-1.29220</td>
<td align="right">1.33410</td>
<td align="right">1.10270</td>
<td align="right">-2.48570</td>
<td align="right">2.14020</td>
<td align="right">-0.69107</td>
<td align="right">-2.0703</td>
<td align="right">-2.38500</td>
<td align="right">-2.8353</td>
<td align="right">-1.33630</td>
<td align="right">3.09660</td>
<td align="right">2.39710</td>
<td align="right">-1.65290</td>
<td align="right">1.76940</td>
<td align="right">-0.91083</td>
<td align="right">2.92590</td>
<td align="right">2.17980</td>
<td align="right">0.98560</td>
<td align="right">1.7860</td>
<td align="right">1.890900</td>
<td align="right">-2.00100</td>
<td align="right">1.947600</td>
<td align="right">2.43630</td>
<td align="right">-1.6380</td>
<td align="right">0.35766</td>
<td align="right">2.149200</td>
<td align="right">-0.96358</td>
<td align="right">1.15230</td>
<td align="right">1.29790</td>
<td align="right">-1.08580</td>
<td align="right">-1.72250</td>
<td align="right">2.59140</td>
<td align="right">2.58610</td>
<td align="right">0.95910</td>
<td align="right">-1.75160</td>
<td align="right">1.65220</td>
<td align="right">-0.60392</td>
<td align="right">-1.19010</td>
<td align="right">-0.77960</td>
<td align="right">NA</td>
<td align="right">-0.96130</td>
<td align="right">-2.10850</td>
<td align="right">-0.54412</td>
<td align="right">-1.02700</td>
<td align="right">-1.06330</td>
<td align="right">1.63060</td>
<td align="right">-0.44400</td>
<td align="right">-1.3552</td>
<td align="right">1.35600</td>
<td align="right">1.73300</td>
<td align="right">-1.66890</td>
<td align="right">2.12540</td>
<td align="right">1.34720</td>
<td align="right">-0.94147</td>
<td align="right">1.826100</td>
<td align="right">-1.21640</td>
<td align="right">-0.64341</td>
<td align="right">1.40320</td>
<td align="right">1.35670</td>
</tr>
<tr class="odd">
<td align="left">VIT_00s0332g00110</td>
<td align="right">3.0292</td>
<td align="right">2.6774</td>
<td align="right">3.20590</td>
<td align="right">9.758500</td>
<td align="right">2.5148</td>
<td align="right">1.8179</td>
<td align="right">3.9399</td>
<td align="right">1.6098</td>
<td align="right">-3.1108</td>
<td align="right">-1.1372</td>
<td align="right">-3.485800</td>
<td align="right">-1.1078</td>
<td align="right">2.80300</td>
<td align="right">-1.2764</td>
<td align="right">-3.23040</td>
<td align="right">2.5257</td>
<td align="right">1.96120</td>
<td align="right">2.806400</td>
<td align="right">1.7198</td>
<td align="right">2.78080</td>
<td align="right">1.8973</td>
<td align="right">-1.13710</td>
<td align="right">2.871400</td>
<td align="right">1.28790</td>
<td align="right">-2.7030</td>
<td align="right">2.33720</td>
<td align="right">0.9282</td>
<td align="right">-2.96110</td>
<td align="right">-1.42400</td>
<td align="right">0.74812</td>
<td align="right">3.25040</td>
<td align="right">-1.82280</td>
<td align="right">0.94685</td>
<td align="right">0.96465</td>
<td align="right">-2.59760</td>
<td align="right">1.94370</td>
<td align="right">-2.21220</td>
<td align="right">-2.0246</td>
<td align="right">-3.43980</td>
<td align="right">-1.9523</td>
<td align="right">-0.76462</td>
<td align="right">0.86233</td>
<td align="right">2.30790</td>
<td align="right">-1.55670</td>
<td align="right">1.47320</td>
<td align="right">-0.76085</td>
<td align="right">2.98080</td>
<td align="right">2.69860</td>
<td align="right">0.56101</td>
<td align="right">1.4174</td>
<td align="right">1.742100</td>
<td align="right">-2.13660</td>
<td align="right">0.375970</td>
<td align="right">2.66480</td>
<td align="right">-1.8175</td>
<td align="right">2.43140</td>
<td align="right">2.007700</td>
<td align="right">-1.69620</td>
<td align="right">0.28288</td>
<td align="right">0.61263</td>
<td align="right">-1.51770</td>
<td align="right">-1.17990</td>
<td align="right">2.41330</td>
<td align="right">2.37930</td>
<td align="right">0.92107</td>
<td align="right">-1.78660</td>
<td align="right">1.26510</td>
<td align="right">-1.95210</td>
<td align="right">-2.06590</td>
<td align="right">-0.34705</td>
<td align="right">NA</td>
<td align="right">-0.89742</td>
<td align="right">-1.63440</td>
<td align="right">-2.06070</td>
<td align="right">-1.41090</td>
<td align="right">0.98436</td>
<td align="right">1.21040</td>
<td align="right">-0.30154</td>
<td align="right">-1.2454</td>
<td align="right">1.60920</td>
<td align="right">0.86018</td>
<td align="right">-1.84160</td>
<td align="right">0.59804</td>
<td align="right">1.19970</td>
<td align="right">-1.80710</td>
<td align="right">2.049700</td>
<td align="right">1.07460</td>
<td align="right">-1.62490</td>
<td align="right">0.66103</td>
<td align="right">0.55152</td>
</tr>
<tr class="even">
<td align="left">VIT_00s0332g00160</td>
<td align="right">4.3616</td>
<td align="right">3.4807</td>
<td align="right">4.15840</td>
<td align="right">0.000000</td>
<td align="right">3.2895</td>
<td align="right">2.5248</td>
<td align="right">3.6967</td>
<td align="right">2.4803</td>
<td align="right">-1.6813</td>
<td align="right">2.6450</td>
<td align="right">-2.312100</td>
<td align="right">2.3961</td>
<td align="right">2.78960</td>
<td align="right">2.4798</td>
<td align="right">-1.23240</td>
<td align="right">2.0372</td>
<td align="right">2.92130</td>
<td align="right">1.421200</td>
<td align="right">1.2663</td>
<td align="right">1.30870</td>
<td align="right">2.6245</td>
<td align="right">-4.72070</td>
<td align="right">2.507000</td>
<td align="right">1.20350</td>
<td align="right">-1.0845</td>
<td align="right">2.81940</td>
<td align="right">1.3175</td>
<td align="right">-1.57550</td>
<td align="right">-2.29660</td>
<td align="right">1.37010</td>
<td align="right">1.58620</td>
<td align="right">-1.17860</td>
<td align="right">1.00370</td>
<td align="right">0.92531</td>
<td align="right">-1.71120</td>
<td align="right">2.68000</td>
<td align="right">-0.69998</td>
<td align="right">-1.3424</td>
<td align="right">-2.28110</td>
<td align="right">-1.5591</td>
<td align="right">-4.33260</td>
<td align="right">2.26370</td>
<td align="right">2.43840</td>
<td align="right">-1.52410</td>
<td align="right">4.42680</td>
<td align="right">0.75333</td>
<td align="right">1.33820</td>
<td align="right">1.02220</td>
<td align="right">0.89325</td>
<td align="right">1.1207</td>
<td align="right">4.391800</td>
<td align="right">-1.06090</td>
<td align="right">3.249200</td>
<td align="right">2.04040</td>
<td align="right">-1.2811</td>
<td align="right">2.18780</td>
<td align="right">1.504100</td>
<td align="right">-1.18510</td>
<td align="right">1.43980</td>
<td align="right">4.65440</td>
<td align="right">-0.97281</td>
<td align="right">-0.67716</td>
<td align="right">1.48910</td>
<td align="right">1.48520</td>
<td align="right">1.75840</td>
<td align="right">-1.05630</td>
<td align="right">3.81080</td>
<td align="right">-0.87440</td>
<td align="right">-1.01640</td>
<td align="right">0.39311</td>
<td align="right">NA</td>
<td align="right">-1.83030</td>
<td align="right">-1.51930</td>
<td align="right">-0.37059</td>
<td align="right">-1.26830</td>
<td align="right">1.63130</td>
<td align="right">0.97022</td>
<td align="right">0.77200</td>
<td align="right">-1.7039</td>
<td align="right">0.87874</td>
<td align="right">2.27770</td>
<td align="right">-1.24620</td>
<td align="right">1.92100</td>
<td align="right">4.09700</td>
<td align="right">-0.90402</td>
<td align="right">1.850400</td>
<td align="right">2.50340</td>
<td align="right">-0.49588</td>
<td align="right">3.77640</td>
<td align="right">0.53108</td>
</tr>
<tr class="odd">
<td align="left">VIT_00s0396g00010</td>
<td align="right">3.7646</td>
<td align="right">2.8414</td>
<td align="right">4.31320</td>
<td align="right">5.277900</td>
<td align="right">2.4906</td>
<td align="right">2.9986</td>
<td align="right">3.5030</td>
<td align="right">2.8764</td>
<td align="right">-3.1878</td>
<td align="right">-2.9072</td>
<td align="right">-3.638900</td>
<td align="right">-3.0570</td>
<td align="right">2.89930</td>
<td align="right">-2.9441</td>
<td align="right">-2.72480</td>
<td align="right">1.9293</td>
<td align="right">2.12520</td>
<td align="right">2.898400</td>
<td align="right">2.0861</td>
<td align="right">2.41010</td>
<td align="right">1.5868</td>
<td align="right">-2.28250</td>
<td align="right">3.202000</td>
<td align="right">2.12140</td>
<td align="right">-1.6655</td>
<td align="right">2.42530</td>
<td align="right">2.0356</td>
<td align="right">-2.37030</td>
<td align="right">-2.59290</td>
<td align="right">1.84800</td>
<td align="right">2.21730</td>
<td align="right">-1.71000</td>
<td align="right">1.88130</td>
<td align="right">1.78240</td>
<td align="right">-2.24730</td>
<td align="right">2.42710</td>
<td align="right">-0.92517</td>
<td align="right">-2.0822</td>
<td align="right">-2.73450</td>
<td align="right">-1.4741</td>
<td align="right">-2.21680</td>
<td align="right">1.73510</td>
<td align="right">2.70640</td>
<td align="right">-1.85970</td>
<td align="right">2.11890</td>
<td align="right">-0.01628</td>
<td align="right">2.05310</td>
<td align="right">2.74810</td>
<td align="right">1.53750</td>
<td align="right">1.9962</td>
<td align="right">2.078500</td>
<td align="right">-2.06380</td>
<td align="right">2.446100</td>
<td align="right">2.15700</td>
<td align="right">-1.7826</td>
<td align="right">2.47110</td>
<td align="right">2.785000</td>
<td align="right">-1.40180</td>
<td align="right">1.54310</td>
<td align="right">2.21970</td>
<td align="right">-1.57730</td>
<td align="right">-1.99940</td>
<td align="right">2.43750</td>
<td align="right">2.43350</td>
<td align="right">1.07900</td>
<td align="right">-1.84320</td>
<td align="right">1.88730</td>
<td align="right">-1.18580</td>
<td align="right">-1.26350</td>
<td align="right">-0.68199</td>
<td align="right">2.49490</td>
<td align="right">-2.16520</td>
<td align="right">-2.03970</td>
<td align="right">-0.76998</td>
<td align="right">-1.57500</td>
<td align="right">-0.17791</td>
<td align="right">1.68210</td>
<td align="right">0.60628</td>
<td align="right">-2.1443</td>
<td align="right">1.44320</td>
<td align="right">1.73370</td>
<td align="right">-2.13880</td>
<td align="right">0.27078</td>
<td align="right">1.63830</td>
<td align="right">-1.17210</td>
<td align="right">2.150300</td>
<td align="right">1.32660</td>
<td align="right">-0.79860</td>
<td align="right">1.95530</td>
<td align="right">1.21540</td>
</tr>
<tr class="even">
<td align="left">VIT_00s0505g00030</td>
<td align="right">3.5150</td>
<td align="right">3.0186</td>
<td align="right">2.12570</td>
<td align="right">4.939700</td>
<td align="right">2.8332</td>
<td align="right">3.8308</td>
<td align="right">1.4118</td>
<td align="right">4.0222</td>
<td align="right">-1.0786</td>
<td align="right">1.6503</td>
<td align="right">0.186190</td>
<td align="right">1.8403</td>
<td align="right">0.72269</td>
<td align="right">1.8435</td>
<td align="right">-0.79985</td>
<td align="right">2.9440</td>
<td align="right">1.57690</td>
<td align="right">1.314800</td>
<td align="right">3.4129</td>
<td align="right">0.62757</td>
<td align="right">2.0120</td>
<td align="right">-0.80238</td>
<td align="right">0.288100</td>
<td align="right">0.84441</td>
<td align="right">-1.6532</td>
<td align="right">1.66320</td>
<td align="right">1.0706</td>
<td align="right">-1.44290</td>
<td align="right">-0.35294</td>
<td align="right">3.23550</td>
<td align="right">0.94906</td>
<td align="right">-0.77656</td>
<td align="right">3.02070</td>
<td align="right">3.06260</td>
<td align="right">-0.98054</td>
<td align="right">1.39980</td>
<td align="right">-0.86832</td>
<td align="right">-0.9452</td>
<td align="right">0.22766</td>
<td align="right">-1.1155</td>
<td align="right">-0.67088</td>
<td align="right">1.30570</td>
<td align="right">0.23235</td>
<td align="right">-0.56582</td>
<td align="right">0.70003</td>
<td align="right">2.39410</td>
<td align="right">0.80835</td>
<td align="right">1.22810</td>
<td align="right">0.71696</td>
<td align="right">1.1959</td>
<td align="right">0.726730</td>
<td align="right">-0.68685</td>
<td align="right">0.633540</td>
<td align="right">0.98118</td>
<td align="right">-0.5433</td>
<td align="right">1.29670</td>
<td align="right">1.778100</td>
<td align="right">-1.19020</td>
<td align="right">3.29600</td>
<td align="right">0.69737</td>
<td align="right">-1.06060</td>
<td align="right">-0.21317</td>
<td align="right">1.09280</td>
<td align="right">1.09530</td>
<td align="right">1.49960</td>
<td align="right">-0.86254</td>
<td align="right">0.64502</td>
<td align="right">-0.72246</td>
<td align="right">-1.19900</td>
<td align="right">2.35640</td>
<td align="right">1.80330</td>
<td align="right">-0.16970</td>
<td align="right">-0.93284</td>
<td align="right">-0.64064</td>
<td align="right">-1.13220</td>
<td align="right">1.69140</td>
<td align="right">0.79138</td>
<td align="right">2.29200</td>
<td align="right">-0.1788</td>
<td align="right">0.29678</td>
<td align="right">0.61320</td>
<td align="right">0.26544</td>
<td align="right">1.68630</td>
<td align="right">0.69730</td>
<td align="right">-1.28570</td>
<td align="right">-0.032303</td>
<td align="right">2.15340</td>
<td align="right">-0.68502</td>
<td align="right">0.60967</td>
<td align="right">0.56684</td>
</tr>
<tr class="odd">
<td align="left">VIT_00s0505g00060</td>
<td align="right">5.2095</td>
<td align="right">4.4270</td>
<td align="right">3.10780</td>
<td align="right">0.025781</td>
<td align="right">3.5454</td>
<td align="right">2.2872</td>
<td align="right">1.1541</td>
<td align="right">2.0220</td>
<td align="right">-2.5693</td>
<td align="right">-4.2649</td>
<td align="right">-3.789000</td>
<td align="right">-4.2613</td>
<td align="right">2.02010</td>
<td align="right">-4.1110</td>
<td align="right">-2.36420</td>
<td align="right">1.0127</td>
<td align="right">1.66700</td>
<td align="right">1.750000</td>
<td align="right">1.8770</td>
<td align="right">2.50800</td>
<td align="right">1.6843</td>
<td align="right">-2.45100</td>
<td align="right">1.576300</td>
<td align="right">1.69270</td>
<td align="right">-1.0032</td>
<td align="right">0.53597</td>
<td align="right">1.1269</td>
<td align="right">-1.38830</td>
<td align="right">-2.24590</td>
<td align="right">0.95411</td>
<td align="right">0.57979</td>
<td align="right">-1.61480</td>
<td align="right">1.12490</td>
<td align="right">0.91563</td>
<td align="right">-1.71420</td>
<td align="right">0.16434</td>
<td align="right">-0.79039</td>
<td align="right">-1.9374</td>
<td align="right">-1.68570</td>
<td align="right">-1.5151</td>
<td align="right">-2.09660</td>
<td align="right">2.32280</td>
<td align="right">1.54080</td>
<td align="right">-1.86440</td>
<td align="right">1.56690</td>
<td align="right">1.27860</td>
<td align="right">0.51829</td>
<td align="right">1.08370</td>
<td align="right">0.63628</td>
<td align="right">1.7791</td>
<td align="right">1.533000</td>
<td align="right">-1.79360</td>
<td align="right">2.108000</td>
<td align="right">1.54850</td>
<td align="right">-1.8733</td>
<td align="right">1.56700</td>
<td align="right">1.717700</td>
<td align="right">-0.77773</td>
<td align="right">1.24090</td>
<td align="right">1.73540</td>
<td align="right">-1.52470</td>
<td align="right">-2.80630</td>
<td align="right">2.70000</td>
<td align="right">2.67430</td>
<td align="right">0.38112</td>
<td align="right">-1.64840</td>
<td align="right">1.64550</td>
<td align="right">-0.64838</td>
<td align="right">-0.82078</td>
<td align="right">1.47980</td>
<td align="right">0.91119</td>
<td align="right">-2.14090</td>
<td align="right">-2.22360</td>
<td align="right">-0.70757</td>
<td align="right">-1.06950</td>
<td align="right">1.95680</td>
<td align="right">1.33390</td>
<td align="right">1.22090</td>
<td align="right">-2.0346</td>
<td align="right">1.13970</td>
<td align="right">1.22730</td>
<td align="right">-2.60330</td>
<td align="right">2.93820</td>
<td align="right">1.39390</td>
<td align="right">-0.88419</td>
<td align="right">1.549000</td>
<td align="right">1.20700</td>
<td align="right">-0.76732</td>
<td align="right">1.72200</td>
<td align="right">0.53279</td>
</tr>
<tr class="even">
<td align="left">VIT_00s0873g00020</td>
<td align="right">1.9944</td>
<td align="right">1.6660</td>
<td align="right">5.51460</td>
<td align="right">0.000000</td>
<td align="right">1.5892</td>
<td align="right">6.5175</td>
<td align="right">4.2501</td>
<td align="right">6.2818</td>
<td align="right">-2.8122</td>
<td align="right">1.6798</td>
<td align="right">-3.175400</td>
<td align="right">1.5844</td>
<td align="right">4.23320</td>
<td align="right">1.3430</td>
<td align="right">-2.94500</td>
<td align="right">2.3445</td>
<td align="right">0.94898</td>
<td align="right">2.754100</td>
<td align="right">3.8784</td>
<td align="right">2.42950</td>
<td align="right">1.1789</td>
<td align="right">-1.94800</td>
<td align="right">3.696800</td>
<td align="right">3.56130</td>
<td align="right">-2.6707</td>
<td align="right">1.06920</td>
<td align="right">3.1746</td>
<td align="right">-2.74080</td>
<td align="right">-2.39640</td>
<td align="right">4.64580</td>
<td align="right">2.35860</td>
<td align="right">-2.40330</td>
<td align="right">4.38710</td>
<td align="right">4.87180</td>
<td align="right">-2.05520</td>
<td align="right">0.90070</td>
<td align="right">-1.78040</td>
<td align="right">-1.6201</td>
<td align="right">-3.32580</td>
<td align="right">-3.0634</td>
<td align="right">-2.08150</td>
<td align="right">0.80846</td>
<td align="right">3.39520</td>
<td align="right">-1.92710</td>
<td align="right">1.20740</td>
<td align="right">2.50200</td>
<td align="right">2.53150</td>
<td align="right">3.25610</td>
<td align="right">3.41830</td>
<td align="right">2.5477</td>
<td align="right">0.937320</td>
<td align="right">-1.69560</td>
<td align="right">2.178000</td>
<td align="right">2.54520</td>
<td align="right">-1.8149</td>
<td align="right">1.69080</td>
<td align="right">1.308500</td>
<td align="right">-2.21260</td>
<td align="right">4.16880</td>
<td align="right">1.37660</td>
<td align="right">-3.04860</td>
<td align="right">-2.14190</td>
<td align="right">1.71350</td>
<td align="right">1.63610</td>
<td align="right">1.55740</td>
<td align="right">-1.43720</td>
<td align="right">1.32330</td>
<td align="right">-2.04990</td>
<td align="right">-2.75170</td>
<td align="right">3.00450</td>
<td align="right">0.62136</td>
<td align="right">-2.15390</td>
<td align="right">-1.27810</td>
<td align="right">-1.59330</td>
<td align="right">-4.01530</td>
<td align="right">1.56660</td>
<td align="right">2.88750</td>
<td align="right">1.71850</td>
<td align="right">-2.0253</td>
<td align="right">2.36400</td>
<td align="right">1.95150</td>
<td align="right">-2.10400</td>
<td align="right">0.96992</td>
<td align="right">0.97977</td>
<td align="right">-2.49650</td>
<td align="right">2.750200</td>
<td align="right">0.43298</td>
<td align="right">-1.57320</td>
<td align="right">1.38720</td>
<td align="right">2.38480</td>
</tr>
<tr class="odd">
<td align="left">VIT_00s0904g00010</td>
<td align="right">-5.3343</td>
<td align="right">-5.0650</td>
<td align="right">-0.50085</td>
<td align="right">0.000000</td>
<td align="right">-4.6958</td>
<td align="right">1.5428</td>
<td align="right">-1.3574</td>
<td align="right">1.4015</td>
<td align="right">-4.0042</td>
<td align="right">-2.6753</td>
<td align="right">0.059764</td>
<td align="right">-2.5555</td>
<td align="right">-1.45260</td>
<td align="right">-2.6451</td>
<td align="right">-3.79670</td>
<td align="right">-3.5342</td>
<td align="right">-3.23390</td>
<td align="right">3.318100</td>
<td align="right">1.9196</td>
<td align="right">3.77230</td>
<td align="right">-3.1865</td>
<td align="right">-0.22479</td>
<td align="right">0.044405</td>
<td align="right">-3.47680</td>
<td align="right">-4.5372</td>
<td align="right">-2.91560</td>
<td align="right">-4.3447</td>
<td align="right">-0.84436</td>
<td align="right">-2.49350</td>
<td align="right">2.16030</td>
<td align="right">-2.81060</td>
<td align="right">-5.35260</td>
<td align="right">2.52130</td>
<td align="right">2.41520</td>
<td align="right">-2.13720</td>
<td align="right">-3.45560</td>
<td align="right">-8.49460</td>
<td align="right">-3.9578</td>
<td align="right">-0.27285</td>
<td align="right">-1.2422</td>
<td align="right">-0.19841</td>
<td align="right">-2.32410</td>
<td align="right">0.21161</td>
<td align="right">-4.16560</td>
<td align="right">-0.24390</td>
<td align="right">-7.18990</td>
<td align="right">-2.94030</td>
<td align="right">-1.87500</td>
<td align="right">-5.73920</td>
<td align="right">-2.3864</td>
<td align="right">-0.067839</td>
<td align="right">-4.14510</td>
<td align="right">-0.081236</td>
<td align="right">0.21687</td>
<td align="right">-3.8409</td>
<td align="right">-1.24240</td>
<td align="right">0.022469</td>
<td align="right">-4.86110</td>
<td align="right">1.20210</td>
<td align="right">0.13036</td>
<td align="right">-2.72490</td>
<td align="right">-4.57210</td>
<td align="right">1.08270</td>
<td align="right">1.08830</td>
<td align="right">-5.66740</td>
<td align="right">-4.17690</td>
<td align="right">0.10886</td>
<td align="right">-6.04510</td>
<td align="right">-3.01150</td>
<td align="right">-5.23260</td>
<td align="right">-0.71329</td>
<td align="right">-3.23250</td>
<td align="right">-1.87390</td>
<td align="right">-7.60410</td>
<td align="right">-0.54999</td>
<td align="right">-3.88660</td>
<td align="right">-2.22410</td>
<td align="right">-7.06470</td>
<td align="right">-2.3227</td>
<td align="right">-3.16590</td>
<td align="right">-1.07620</td>
<td align="right">-2.20850</td>
<td align="right">-3.07110</td>
<td align="right">-0.21459</td>
<td align="right">-3.67570</td>
<td align="right">-0.131950</td>
<td align="right">-2.93390</td>
<td align="right">-7.06100</td>
<td align="right">0.25534</td>
<td align="right">-5.57910</td>
</tr>
</tbody>
</table>
</div>
<div id="plot-a-module" class="section level1">
<h1 class="hasAnchor">
<a href="#plot-a-module" class="anchor"></a>Plot a module</h1>
<p>The easiest way to visualize a module is using a heatmap. The Plot object wraps a module and can return different plot type with different format (the Plotly JSON format or HTML). To see an heatmap embedded in rstudio we can retrieve the HTML code of the Plot (that uses Plotly) and display it directly into the viewer.</p>
<div class="sourceCode" id="cb12"><html><body><pre class="r"><span class="fu"><a href="../reference/view_plot.html">view_plot</a></span>(<span class="kw">module</span> <span class="kw">=</span> <span class="no">module_1</span>, <span class="kw">plotType</span> <span class="kw">=</span> <span class="st">"module_heatmap_expression"</span>, <span class="kw">min</span> <span class="kw">=</span> -<span class="fl">7</span>, <span class="kw">max</span> <span class="kw">=</span> <span class="fl">7</span>)</pre></body></html></div>
<p>Alternatively you can use directly any R function to cluster data and produce an heatmap, e.g.:</p>
<div class="sourceCode" id="cb13"><html><body><pre class="r"><span class="kw pkg">heatmaply</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/heatmaply/man/heatmaply.html">heatmaply</a></span>(<span class="no">module_1</span>,
                     <span class="kw">showticklabels</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">TRUE</span>, <span class="fl">TRUE</span>),
                     <span class="kw">fontsize_col</span> <span class="kw">=</span> <span class="fl">7.5</span>,
                     <span class="kw">col</span> <span class="kw">=</span> <span class="kw pkg">gplots</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/gplots/man/colorpanel.html">greenred</a></span>(<span class="fl">50</span>),
                     <span class="kw">plot_method</span> <span class="kw">=</span> <span class="st">'plotly'</span>)</pre></body></html></div>
<p>or</p>
<div class="sourceCode" id="cb14"><html><body><pre class="r"><span class="no">pheat_cl</span> <span class="kw">&lt;-</span> <span class="kw pkg">pheatmap</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/pheatmap/man/pheatmap.html">pheatmap</a></span>(<span class="kw">mat</span> <span class="kw">=</span>  <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span>(<span class="no">module_1</span>))
<span class="no">pheat_cl</span></pre></body></html></div>
<p>Biological Features (genes) and Sample Sets (contrasts) have no particular order in a module. But since the heatmap is clustered we can retrieve the sorted list of Biological Features and Sample Sets respectively using the <code>sorted = TRUE</code> parameter.</p>
<div class="sourceCode" id="cb15"><html><body><pre class="r"><span class="no">my_sorted_indexes</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/plot_module_heatmap.html">plot_module_heatmap</a></span>(<span class="kw">module</span> <span class="kw">=</span> <span class="no">module_1</span>, <span class="kw">plot</span> <span class="kw">=</span> <span class="fl">FALSE</span>, <span class="kw">sorted</span> <span class="kw">=</span> <span class="fl">TRUE</span>)
<span class="no">unsorted_ss</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/get_sampleset_id.html">get_sampleset_id</a></span>(<span class="kw">name_In</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">module_1</span>), <span class="kw">version</span><span class="kw">=</span><span class="st">"legacy"</span>)
<span class="no">unsorted_bf</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/get_biofeature_id.html">get_biofeature_id</a></span>(<span class="kw">name_In</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/row.names.html">row.names</a></span>(<span class="no">module_1</span>))
<span class="no">sorted_ss</span> <span class="kw">&lt;-</span> <span class="no">unsorted_ss</span>[<span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span>(<span class="no">my_sorted_indexes</span>$<span class="no">sortedSamplesets</span>,<span class="no">unsorted_ss</span>$<span class="no">id</span>),]
<span class="no">sorted_bf</span> <span class="kw">&lt;-</span> <span class="no">unsorted_bf</span>[<span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span>(<span class="no">my_sorted_indexes</span>$<span class="no">sortedBiofeatures</span>,<span class="no">unsorted_bf</span>$<span class="no">id</span>),]
<span class="no">module_1</span>[<span class="no">sorted_bf</span>$<span class="no">name</span>,<span class="no">sorted_ss</span>$<span class="no">name</span>][<span class="fl">1</span>:<span class="fl">5</span>,<span class="fl">1</span>:<span class="fl">5</span>]
<span class="co">#&gt;                   GSM881525.ch1-vs-GSM881520.ch1</span>
<span class="co">#&gt; VIT_00s0873g00020                         3.2561</span>
<span class="co">#&gt; VIT_00s0505g00060                         1.0837</span>
<span class="co">#&gt; VIT_00s0505g00030                         1.2281</span>
<span class="co">#&gt; VIT_00s0396g00010                         2.7481</span>
<span class="co">#&gt; VIT_00s0332g00160                         1.0222</span>
<span class="co">#&gt;                   GSM1532866.ch1-vs-GSM1532856.ch1</span>
<span class="co">#&gt; VIT_00s0873g00020                          1.56660</span>
<span class="co">#&gt; VIT_00s0505g00060                          1.95680</span>
<span class="co">#&gt; VIT_00s0505g00030                          1.69140</span>
<span class="co">#&gt; VIT_00s0396g00010                         -0.17791</span>
<span class="co">#&gt; VIT_00s0332g00160                          1.63130</span>
<span class="co">#&gt;                   GSM1532841.ch1-vs-GSM1532832.ch1</span>
<span class="co">#&gt; VIT_00s0873g00020                           1.9944</span>
<span class="co">#&gt; VIT_00s0505g00060                           5.2095</span>
<span class="co">#&gt; VIT_00s0505g00030                           3.5150</span>
<span class="co">#&gt; VIT_00s0396g00010                           3.7646</span>
<span class="co">#&gt; VIT_00s0332g00160                           4.3616</span>
<span class="co">#&gt;                   GSM1532842.ch1-vs-GSM1532832.ch1</span>
<span class="co">#&gt; VIT_00s0873g00020                           1.6660</span>
<span class="co">#&gt; VIT_00s0505g00060                           4.4270</span>
<span class="co">#&gt; VIT_00s0505g00030                           3.0186</span>
<span class="co">#&gt; VIT_00s0396g00010                           2.8414</span>
<span class="co">#&gt; VIT_00s0332g00160                           3.4807</span>
<span class="co">#&gt;                   GSM1532843.ch1-vs-GSM1532832.ch1</span>
<span class="co">#&gt; VIT_00s0873g00020                           1.5892</span>
<span class="co">#&gt; VIT_00s0505g00060                           3.5454</span>
<span class="co">#&gt; VIT_00s0505g00030                           2.8332</span>
<span class="co">#&gt; VIT_00s0396g00010                           2.4906</span>
<span class="co">#&gt; VIT_00s0332g00160                           3.2895</span></pre></body></html></div>
<p>or, alternatively, you can create a new module with the sorted features (bf and ss) already in place:</p>
<div class="sourceCode" id="cb16"><html><body><pre class="r"><span class="no">module_1_sorted</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/create_module.html">create_module</a></span>(<span class="kw">biofeaturesNames</span> <span class="kw">=</span> <span class="no">my_sorted_indexes</span>$<span class="no">sortedBiofeatures</span>,
                                 <span class="kw">samplesetNames</span> <span class="kw">=</span> <span class="no">my_sorted_indexes</span>$<span class="no">sortedSamplesets</span>,
                                 <span class="kw">normalization</span> <span class="kw">=</span> <span class="st">"legacy"</span>,
                                 <span class="kw">useIds</span> <span class="kw">=</span> <span class="fl">TRUE</span>)</pre></body></html></div>
<p>Alternatively we can use variables containing the order for both rows and columns produced by other heatmap functions.</p>
<div class="sourceCode" id="cb17"><html><body><pre class="r"><span class="kw pkg">knitr</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span>(<span class="no">module_1</span>[<span class="no">pheat_cl</span>$<span class="no">tree_row</span>$<span class="no">order</span>,<span class="no">pheat_cl</span>$<span class="no">tree_col</span>$<span class="no">order</span>])</pre></body></html></div>
</div>
<div id="plot-correlation-network" class="section level1">
<h1 class="hasAnchor">
<a href="#plot-correlation-network" class="anchor"></a>Plot correlation network</h1>
<p>Given a module we can plot the correlation network between Biological Features (genes) by looking at the Pearson correlation coefficient of the module’s Sample Sets (contrast). The default threshold is 0.7, but it can be changed. Edges in orange represent anti-correlation.</p>
<div class="sourceCode" id="cb18"><html><body><pre class="r"><span class="fu"><a href="../reference/view_plot.html">view_plot</a></span>(<span class="kw">module</span> <span class="kw">=</span> <span class="no">module_1</span>, <span class="kw">plotType</span> <span class="kw">=</span> <span class="st">"module_coexpression_network"</span>, <span class="kw">threshold</span> <span class="kw">=</span> <span class="fl">0.2</span>)</pre></body></html></div>
</div>
<div id="automatically-expand-a-module" class="section level1">
<h1 class="hasAnchor">
<a href="#automatically-expand-a-module" class="anchor"></a>Automatically expand a module</h1>
<p>One of the classic way to use VESPUCCI is to start with a set of genes selected from previous analysis, have a look at the top conditions in which they show a consistent (and strong) behavior and then look for other genes that behave the same across those same conditions. The way to do it is to plot the distribution of genes that show a similar behavior of the module’s genes across the same condition in order pick up a suitable threshold.</p>
<div class="sourceCode" id="cb19"><html><body><pre class="r"><span class="fu"><a href="../reference/view_plot.html">view_plot</a></span>(<span class="kw">module</span> <span class="kw">=</span> <span class="no">module_1</span>, <span class="kw">plotType</span> <span class="kw">=</span> <span class="st">"biological_features_uncentered_correlation_distribution"</span>)</pre></body></html></div>
<p>For example, if we pick a threshold around 0.8 (looking at the top plot) we might expect to have around 40 Biological Features (genes) with a score equal or greater than that, to be added in our module. In this case let’s just pick the top 30 genes and create e new module.</p>
<div class="sourceCode" id="cb20"><html><body><pre class="r"><span class="no">top_bf</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/plot_module_distribution.html">plot_module_distribution</a></span>(<span class="kw">module</span> <span class="kw">=</span> <span class="no">module_1</span>,
                                   <span class="kw">plotType</span> <span class="kw">=</span> <span class="st">"biological_features_uncentered_correlation_distribution"</span>,
                                   <span class="kw">getRank</span> <span class="kw">=</span> <span class="fl">TRUE</span>)</pre></body></html></div>
<p>The new module is created by adding Biological Feature objects to the old module.</p>
<div class="sourceCode" id="cb21"><html><body><pre class="r"><span class="no">module_2</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/create_module.html">create_module</a></span>(<span class="kw">biofeaturesNames</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">module_1</span>), <span class="no">top_bf</span>$<span class="no">name</span>[<span class="fl">1</span>:<span class="fl">30</span>]),
                          <span class="kw">normalization</span> <span class="kw">=</span> <span class="st">"legacy"</span>,
                          <span class="kw">useIds</span> <span class="kw">=</span> <span class="fl">FALSE</span>)</pre></body></html></div>
<p>We can have a look at the heatmap of the new module</p>
<div class="sourceCode" id="cb22"><html><body><pre class="r"><span class="fu"><a href="../reference/view_plot.html">view_plot</a></span>(<span class="no">module_2</span>, <span class="kw">plotType</span> <span class="kw">=</span> <span class="st">"module_heatmap_expression"</span> )</pre></body></html></div>
</div>
<div id="annotations" class="section level1">
<h1 class="hasAnchor">
<a href="#annotations" class="anchor"></a>Annotations</h1>
<p>In VESPUCCI, both Biological Feature (gene) and Sample annotations are represented using <a href="https://en.wikipedia.org/wiki/Resource_Description_Framework">RDF</a>. We can visualize annotation as a set of RDF triples.</p>
<div class="sourceCode" id="cb23"><html><body><pre class="r"><span class="no">samples</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/strsplit.html">strsplit</a></span>(<span class="no">sorted_ss</span>$<span class="no">name</span><span class="kw">[[</span><span class="fl">1</span>]],<span class="kw">split</span> <span class="kw">=</span> <span class="st">"-vs-"</span>)<span class="kw">[[</span><span class="fl">1</span>]]
<span class="co"># knitr::kable(get_sample_by_gsm(sampleName_In =  samples))</span>
<span class="kw pkg">knitr</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span>(<span class="fu"><a href="../reference/get_annotation_triples.html">get_annotation_triples</a></span>(<span class="kw">ids</span> <span class="kw">=</span>  <span class="fu"><a href="../reference/get_sample_by_gsm.html">get_sample_by_gsm</a></span>(<span class="kw">sampleName_In</span> <span class="kw">=</span>  <span class="no">samples</span>)$<span class="no">sampleId</span>))
<span class="co">#&gt; Warning in kable_pipe(x = structure(c("GSM881520.ch1", "PO_0009084",</span>
<span class="co">#&gt; "GSM881520.ch1", : The table should have a header (column names)</span></pre></body></html></div>
<table class="table">
<colgroup>
<col width="34%">
<col width="30%">
<col width="34%">
</colgroup>
<tbody>
<tr class="odd">
<td align="left">GSM881520.ch1</td>
<td align="left">whole plant development stage</td>
<td align="left">29</td>
</tr>
<tr class="even">
<td align="left">PO_0009084</td>
<td align="left">name</td>
<td align="left">pericarp</td>
</tr>
<tr class="odd">
<td align="left">GSM881520.ch1</td>
<td align="left">type</td>
<td align="left">PO_0009084</td>
</tr>
<tr class="even">
<td align="left">GSM881520.ch1</td>
<td align="left">Year of Onset</td>
<td align="left">N24b1159352a140d29be6add083b86a9b</td>
</tr>
<tr class="odd">
<td align="left">UO_0000036</td>
<td align="left">name</td>
<td align="left">year</td>
</tr>
<tr class="even">
<td align="left">N24b1159352a140d29be6add083b86a9b</td>
<td align="left">unit</td>
<td align="left">UO_0000036</td>
</tr>
<tr class="odd">
<td align="left">N24b1159352a140d29be6add083b86a9b</td>
<td align="left">value</td>
<td align="left">2008</td>
</tr>
<tr class="even">
<td align="left">GSM881520.ch1</td>
<td align="left">is grown in</td>
<td align="left">N9164f9da3fbb45609ca4428c366fc6ab</td>
</tr>
<tr class="odd">
<td align="left">ENVO_00000116</td>
<td align="left">name</td>
<td align="left">vineyard</td>
</tr>
<tr class="even">
<td align="left">N9164f9da3fbb45609ca4428c366fc6ab</td>
<td align="left">type</td>
<td align="left">ENVO_00000116</td>
</tr>
<tr class="odd">
<td align="left">N9164f9da3fbb45609ca4428c366fc6ab</td>
<td align="left">Geographical Location</td>
<td align="left">N58dc052d55194b94966d7838bc94337d</td>
</tr>
<tr class="even">
<td align="left">N58dc052d55194b94966d7838bc94337d</td>
<td align="left">Description</td>
<td align="left">PM Soave Italy</td>
</tr>
<tr class="odd">
<td align="left">N58dc052d55194b94966d7838bc94337d</td>
<td align="left">Latitude</td>
<td align="left">N8e1b73606e2846bebf209a7b821ca5fc</td>
</tr>
<tr class="even">
<td align="left">N58dc052d55194b94966d7838bc94337d</td>
<td align="left">Longitude</td>
<td align="left">N819972170b3a427b8c6441ec884ef041</td>
</tr>
<tr class="odd">
<td align="left">N819972170b3a427b8c6441ec884ef041</td>
<td align="left">value</td>
<td align="left">11.053889</td>
</tr>
<tr class="even">
<td align="left">UO_0000185</td>
<td align="left">name</td>
<td align="left">degree</td>
</tr>
<tr class="odd">
<td align="left">N819972170b3a427b8c6441ec884ef041</td>
<td align="left">unit</td>
<td align="left">UO_0000185</td>
</tr>
<tr class="even">
<td align="left">UO_0000185</td>
<td align="left">name</td>
<td align="left">degree</td>
</tr>
<tr class="odd">
<td align="left">N8e1b73606e2846bebf209a7b821ca5fc</td>
<td align="left">unit</td>
<td align="left">UO_0000185</td>
</tr>
<tr class="even">
<td align="left">N8e1b73606e2846bebf209a7b821ca5fc</td>
<td align="left">value</td>
<td align="left">45.454722</td>
</tr>
<tr class="odd">
<td align="left">GSM881520.ch1</td>
<td align="left">Year of Onset</td>
<td align="left">N030126a04dfe40b0ae9abc4fc31c92b2</td>
</tr>
<tr class="even">
<td align="left">UO_0000036</td>
<td align="left">name</td>
<td align="left">year</td>
</tr>
<tr class="odd">
<td align="left">N030126a04dfe40b0ae9abc4fc31c92b2</td>
<td align="left">unit</td>
<td align="left">UO_0000036</td>
</tr>
<tr class="even">
<td align="left">N030126a04dfe40b0ae9abc4fc31c92b2</td>
<td align="left">value</td>
<td align="left">2009</td>
</tr>
<tr class="odd">
<td align="left">GSM881520.ch1</td>
<td align="left">cultivar</td>
<td align="left">Corvina</td>
</tr>
<tr class="even">
<td align="left">NCBITaxon_29760</td>
<td align="left">name</td>
<td align="left">Vitis vinifera</td>
</tr>
<tr class="odd">
<td align="left">GSM881520.ch1</td>
<td align="left">Genotype</td>
<td align="left">NCBITaxon_29760</td>
</tr>
<tr class="even">
<td align="left">PO_0009084</td>
<td align="left">name</td>
<td align="left">pericarp</td>
</tr>
<tr class="odd">
<td align="left">GSM881525.ch1</td>
<td align="left">type</td>
<td align="left">PO_0009084</td>
</tr>
<tr class="even">
<td align="left">GSM881525.ch1</td>
<td align="left">Year of Onset</td>
<td align="left">N4554979868c741f8bfff74e2f01cd87d</td>
</tr>
<tr class="odd">
<td align="left">N4554979868c741f8bfff74e2f01cd87d</td>
<td align="left">value</td>
<td align="left">2008</td>
</tr>
<tr class="even">
<td align="left">UO_0000036</td>
<td align="left">name</td>
<td align="left">year</td>
</tr>
<tr class="odd">
<td align="left">N4554979868c741f8bfff74e2f01cd87d</td>
<td align="left">unit</td>
<td align="left">UO_0000036</td>
</tr>
<tr class="even">
<td align="left">GSM881525.ch1</td>
<td align="left">is grown in</td>
<td align="left">N554f48a4c1c4466f8a2c389c3b21b4cb</td>
</tr>
<tr class="odd">
<td align="left">ENVO_00000116</td>
<td align="left">name</td>
<td align="left">vineyard</td>
</tr>
<tr class="even">
<td align="left">N554f48a4c1c4466f8a2c389c3b21b4cb</td>
<td align="left">type</td>
<td align="left">ENVO_00000116</td>
</tr>
<tr class="odd">
<td align="left">N554f48a4c1c4466f8a2c389c3b21b4cb</td>
<td align="left">Geographical Location</td>
<td align="left">N38d451cbf6e1417eb4df6b7c4322579e</td>
</tr>
<tr class="even">
<td align="left">N38d451cbf6e1417eb4df6b7c4322579e</td>
<td align="left">Longitude</td>
<td align="left">N4c6c8d73bd974247af2927eefc94eb34</td>
</tr>
<tr class="odd">
<td align="left">N38d451cbf6e1417eb4df6b7c4322579e</td>
<td align="left">Latitude</td>
<td align="left">N8340be385b7648579fb4044541f88aa8</td>
</tr>
<tr class="even">
<td align="left">N38d451cbf6e1417eb4df6b7c4322579e</td>
<td align="left">Description</td>
<td align="left">PM Soave Italy</td>
</tr>
<tr class="odd">
<td align="left">N8340be385b7648579fb4044541f88aa8</td>
<td align="left">value</td>
<td align="left">45.454722</td>
</tr>
<tr class="even">
<td align="left">UO_0000185</td>
<td align="left">name</td>
<td align="left">degree</td>
</tr>
<tr class="odd">
<td align="left">N8340be385b7648579fb4044541f88aa8</td>
<td align="left">unit</td>
<td align="left">UO_0000185</td>
</tr>
<tr class="even">
<td align="left">UO_0000185</td>
<td align="left">name</td>
<td align="left">degree</td>
</tr>
<tr class="odd">
<td align="left">N4c6c8d73bd974247af2927eefc94eb34</td>
<td align="left">unit</td>
<td align="left">UO_0000185</td>
</tr>
<tr class="even">
<td align="left">N4c6c8d73bd974247af2927eefc94eb34</td>
<td align="left">value</td>
<td align="left">11.053889</td>
</tr>
<tr class="odd">
<td align="left">GSM881525.ch1</td>
<td align="left">Year of Onset</td>
<td align="left">N6eba5a16dc934566a90f255ecb534bf7</td>
</tr>
<tr class="even">
<td align="left">UO_0000036</td>
<td align="left">name</td>
<td align="left">year</td>
</tr>
<tr class="odd">
<td align="left">N6eba5a16dc934566a90f255ecb534bf7</td>
<td align="left">unit</td>
<td align="left">UO_0000036</td>
</tr>
<tr class="even">
<td align="left">N6eba5a16dc934566a90f255ecb534bf7</td>
<td align="left">value</td>
<td align="left">2009</td>
</tr>
<tr class="odd">
<td align="left">GSM881525.ch1</td>
<td align="left">cultivar</td>
<td align="left">Corvina</td>
</tr>
<tr class="even">
<td align="left">NCBITaxon_29760</td>
<td align="left">name</td>
<td align="left">Vitis vinifera</td>
</tr>
<tr class="odd">
<td align="left">GSM881525.ch1</td>
<td align="left">Genotype</td>
<td align="left">NCBITaxon_29760</td>
</tr>
<tr class="even">
<td align="left">GSM881525.ch1</td>
<td align="left">whole plant development stage</td>
<td align="left">32</td>
</tr>
</tbody>
</table>
<p>Another way is to visualize the annotation is with RDF graph.</p>
<p>Annotations can also be used to retrieve Sample or Biological Features (gene) starting from a SPARQL query. For example we might want to select all seed Sample.</p>
<div class="sourceCode" id="cb24"><html><body><pre class="r"><span class="no">sparql</span> <span class="kw">=</span> <span class="st">"SELECT ?s ?p ?o WHERE { ?s &lt;http://www.w3.org/1999/02/22-rdf-syntax-ns#type&gt; &lt;http://purl.obolibrary.org/obo/NCIT_C19157&gt; . ?s &lt;http://www.w3.org/1999/02/22-rdf-syntax-ns#type&gt; &lt;http://purl.obolibrary.org/obo/PO_0009010&gt;}"</span></pre></body></html></div>
<p>The term <a href="http://purl.obolibrary.org/obo/NCIT_C19157">NCIT_C19157</a> indicates <strong><em>specimen</em></strong> while <a href="http://purl.obolibrary.org/obo/PO_0009010">PO_0009010</a> is the Plant Ontology term for <strong><em>seed</em></strong>. To indicate a gene, instead the term <a href="http://purl.obolibrary.org/obo/NCIT_C16612">NCIT_C16612</a> is used.</p>
<p>Here we retrieve all the samples used based on the sparql query:</p>
<div class="sourceCode" id="cb25"><html><body><pre class="r"><span class="no">seed_samples</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/get_sparql_annotation_triples.html">get_sparql_annotation_triples</a></span>(<span class="kw">target</span> <span class="kw">=</span> <span class="st">"sample"</span>, <span class="kw">query</span> <span class="kw">=</span> <span class="no">sparql</span>)[,<span class="fl">1</span>]</pre></body></html></div>
<p>Now we can collect all the Sample Sets in which these samples are used and then create a new module based on Sample Sets only leaving VESPUCCI to figure out which are the genes (Biological Feature objects) that show a consistent up or down-regulated profile.</p>
<div class="sourceCode" id="cb26"><html><body><pre class="r"><span class="no">seed_ss</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/get_sampleset_id.html">get_sampleset_id</a></span>(<span class="kw">name_In</span> <span class="kw">=</span> <span class="no">seed_samples</span>, <span class="kw">useIds</span> <span class="kw">=</span> <span class="fl">TRUE</span>)
<span class="no">module_3</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/create_module.html">create_module</a></span>(<span class="kw">samplesetNames</span> <span class="kw">=</span> <span class="no">seed_ss</span>$<span class="no">name</span>, <span class="kw">useIds</span> <span class="kw">=</span> <span class="fl">FALSE</span>)</pre></body></html></div>
<p>And now we can plot the heatmap</p>
<div class="sourceCode" id="cb27"><html><body><pre class="r"><span class="fu"><a href="../reference/view_plot.html">view_plot</a></span>(<span class="no">module_3</span>, <span class="kw">plotType</span> <span class="kw">=</span> <span class="st">"module_heatmap_expression"</span> )</pre></body></html></div>
</div>
<div id="session-information" class="section level1">
<h1 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session information</h1>
<pre><code>#&gt; R version 4.0.2 (2020-06-22)
#&gt; Platform: x86_64-redhat-linux-gnu (64-bit)
#&gt; Running under: Fedora 32 (Workstation Edition)
#&gt; 
#&gt; Matrix products: default
#&gt; BLAS/LAPACK: /usr/lib64/libopenblas-r0.3.10.so
#&gt; 
#&gt; attached base packages:
#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     
#&gt; 
#&gt; other attached packages:
#&gt; [1] rcompass_0.0.3
#&gt; 
#&gt; loaded via a namespace (and not attached):
#&gt;  [1] Rcpp_1.0.5       rstudioapi_0.11  knitr_1.29       magrittr_1.5    
#&gt;  [5] MASS_7.3-51.6    graphql_1.5      R6_2.4.1         rlang_0.4.7     
#&gt;  [9] highr_0.8        stringr_1.4.0    tools_4.0.2      xfun_0.16       
#&gt; [13] htmltools_0.5.0  yaml_2.2.1       assertthat_0.2.1 digest_0.6.25   
#&gt; [17] rprojroot_1.3-2  httpcode_0.3.0   pkgdown_1.5.1    crayon_1.3.4    
#&gt; [21] RJSONIO_1.3-1.4  fs_1.5.0         triebeard_0.3.0  ghql_0.1.0      
#&gt; [25] curl_4.3         crul_1.0.0       memoise_1.1.0    evaluate_0.14   
#&gt; [29] rmarkdown_2.3    stringi_1.4.6    compiler_4.0.2   urltools_1.7.3  
#&gt; [33] desc_1.2.0       backports_1.1.9  jsonlite_1.7.0</code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Paolo Sonego.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
