<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Use Case 01 • rcompass</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Use Case 01">
<meta property="og:description" content="rcompass">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">rcompass</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.8</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/rcompass.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/usecase01.html">Use Case 01</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/onertipaday/rcompass/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="usecase01_files/header-attrs-2.9/header-attrs.js"></script><script src="usecase01_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Use Case 01</h1>
                        <h4 class="author">Paolo Sonego</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/onertipaday/rcompass/blob/master/vignettes/usecase01.Rmd"><code>vignettes/usecase01.Rmd</code></a></small>
      <div class="hidden name"><code>usecase01.Rmd</code></div>

    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>How to query COMPASS from R</p>
    </div>
    
<p><a href="https://vespucci.readthedocs.io/">VESPUCCI</a> is the gene expression database for grapevine and we can access it via its GraphQL interface, called <a href="https://compass-.readthedocs.io/">COMPASS</a>. The <a href="https://onertipaday.github.io/rcompass/">rCOMPASS</a> package is a R package that wraps some functionalities to simplify communication with the COMPASS interface.</p>
<p>In this first exercise we will perform few basic operations with VESPUCCI. We will create a Module starting from few genes and then automatically extend it by adding more genes. We will also have a look at gene and samples annotations.:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># devtools::install_github("onertipaday/rcompass")</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">rcompass</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; This is 'rCOMPASS' version 0.0.8 based on COMPASS version 0.9.3.</span></code></pre></div>
<p>The COMPASS GraphQL endpoint might hosts different compendia. At the moment there is only the VESPUCCI compendium, but there are different version of VESPUCCI, and each version might have data normalized in different ways. In this case there are 2 versions of VESPUCCI, version 1.0 (legacy) and version 2.0 (latest). The latter has data normalized in 2 different ways, TPM normalization and LIMMA (the default one) while the legacy version has the legacy normalization only (i.e. per-sample logratios). For every query we will need to indicate the compendium we want to use, if no version, no normalization and no database is specified the 2.0 vitis_vinifera tpm limma (default) values will be used.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"The Vitis gene expression compendium version"</span>, <span class="fu"><a href="../reference/get_available_compendia.html">get_available_compendia</a></span><span class="op">(</span><span class="op">)</span><span class="op">$</span><span class="va">versions</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>, <span class="st">"normalized contains values for"</span>, <span class="fu"><a href="../reference/totalCounts.html">totalCounts</a></span><span class="op">(</span>version<span class="op">=</span><span class="st">"1.0"</span>, aggregate_type<span class="op">=</span><span class="st">"biofeatures"</span><span class="op">)</span>, <span class="st">" biological features, measured for"</span>, <span class="fu"><a href="../reference/totalCounts.html">totalCounts</a></span><span class="op">(</span>version<span class="op">=</span><span class="st">"2.0"</span>, aggregate_type<span class="op">=</span><span class="st">"sampleSets"</span><span class="op">)</span>, <span class="st">" sample sets. This corresponds to a total of"</span>, <span class="fu"><a href="../reference/totalCounts.html">totalCounts</a></span><span class="op">(</span>aggregate_type <span class="op">=</span> <span class="st">"experiments"</span><span class="op">)</span>, <span class="st">" experiments and"</span>, <span class="fu"><a href="../reference/totalCounts.html">totalCounts</a></span><span class="op">(</span>aggregate_type <span class="op">=</span> <span class="st">"samples"</span><span class="op">)</span>, <span class="st">" samples measured on"</span>, <span class="fu"><a href="../reference/totalCounts.html">totalCounts</a></span><span class="op">(</span>aggregate_type <span class="op">=</span> <span class="st">"platforms"</span><span class="op">)</span>,<span class="st">" different platforms divided in "</span><span class="op">)</span>
<span class="co">#&gt; [1] "The Vitis gene expression compendium version 2.0 vitis_vinifera tpm limma (default) normalized contains values for 29090  biological features, measured for 1689  sample sets. This corresponds to a total of 160  experiments and 4101  samples measured on 53  different platforms divided in "</span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span><span class="op">(</span> <span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="fu"><a href="../reference/get_platform_information.html">get_platform_information</a></span><span class="op">(</span><span class="op">)</span><span class="op">$</span><span class="va">source</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<div id="genes" class="section level1">
<h1 class="hasAnchor">
<a href="#genes" class="anchor"></a>Genes</h1>
<p>Let’s build our module starting from a bunch of genes that might come from a previous analysis. Gene name used are the VIT_ ids</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gene_names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'VIT_00s0246g00220'</span>,<span class="st">'VIT_00s0332g00060'</span>,<span class="st">'VIT_00s0332g00110'</span>,<span class="st">'VIT_00s0332g00160'</span>,<span class="st">'VIT_00s0396g00010'</span>,<span class="st">'VIT_00s0505g00030'</span>,<span class="st">'VIT_00s0505g00060'</span>,<span class="st">'VIT_00s0873g00020'</span>,<span class="st">'VIT_00s0904g00010'</span><span class="op">)</span></code></pre></div>
<p>We can query the compendium with the list of gene names and get a list of BiologicalFeature objects that represents our genes of interest. The easiest way to know which are the valid filter values to be used is to use the autocompletion (ALT + SPACEBAR) in the COMPASS GraphQL interface at <a href="http://compass.fmach.it/graphql">http://compass.fmach.it/graphql</a>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bf_ids</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_biofeature_id.html">get_biofeature_id</a></span><span class="op">(</span>compendium <span class="op">=</span> <span class="st">"vespucci"</span>, id_In <span class="op">=</span> <span class="va">gene_names</span>, useIds <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> 
<span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span><span class="op">(</span><span class="va">bf_ids</span><span class="op">)</span></code></pre></div>
<table class="table">
<colgroup>
<col width="27%">
<col width="17%">
<col width="55%">
</colgroup>
<thead><tr class="header">
<th align="left">id</th>
<th align="left">name</th>
<th align="left">description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">QmlvRmVhdHVyZVR5cGU6Mjc0Mzk=</td>
<td align="left">VIT_00s0246g00220</td>
<td align="left">ATPase subunit 8</td>
</tr>
<tr class="even">
<td align="left">QmlvRmVhdHVyZVR5cGU6Mjc4OTI=</td>
<td align="left">VIT_00s0332g00060</td>
<td align="left">ATP synthase subunit beta (Fragment)</td>
</tr>
<tr class="odd">
<td align="left">QmlvRmVhdHVyZVR5cGU6Mjc4OTY=</td>
<td align="left">VIT_00s0332g00110</td>
<td align="left">PSII reaction center subunit X</td>
</tr>
<tr class="even">
<td align="left">QmlvRmVhdHVyZVR5cGU6Mjc5MDA=</td>
<td align="left">VIT_00s0332g00160</td>
<td align="left">NADH dehydrogenase subunit 7</td>
</tr>
<tr class="odd">
<td align="left">QmlvRmVhdHVyZVR5cGU6MjgwODE=</td>
<td align="left">VIT_00s0396g00010</td>
<td align="left">Photosystem II CP43 chlorophyll apoprotein (Fragment)</td>
</tr>
<tr class="even">
<td align="left">QmlvRmVhdHVyZVR5cGU6MjgyNzg=</td>
<td align="left">VIT_00s0505g00030</td>
<td align="left">Orf300 protein</td>
</tr>
<tr class="odd">
<td align="left">QmlvRmVhdHVyZVR5cGU6MjgyODE=</td>
<td align="left">VIT_00s0505g00060</td>
<td align="left">PSII N protein</td>
</tr>
<tr class="even">
<td align="left">QmlvRmVhdHVyZVR5cGU6Mjg1OTg=</td>
<td align="left">VIT_00s0873g00020</td>
<td align="left">NADH-ubiquinone oxidoreductase chain 3</td>
</tr>
<tr class="odd">
<td align="left">QmlvRmVhdHVyZVR5cGU6Mjg2MDk=</td>
<td align="left">VIT_00s0904g00010</td>
<td align="left">Oxygen-evolving enhancer protein 3, chloroplast, putative</td>
</tr>
</tbody>
</table>
<p>Each BiologicalFeature object has several fields. In our case, since the BiologicalFeature represent a gene, we have sequence. The id is the identifier used internally in the database.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span><span class="op">(</span><span class="fu"><a href="../reference/get_biofeature_by_name.html">get_biofeature_by_name</a></span><span class="op">(</span>name_In <span class="op">=</span> <span class="va">gene_names</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<table class="table">
<colgroup>
<col width="0%">
<col width="1%">
<col width="98%">
</colgroup>
<thead><tr class="header">
<th align="left">name</th>
<th align="left">id</th>
<th align="left">value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">VIT_00s0246g00220</td>
<td align="left">QmlvRmVhdHVyZVZhbHVlVHlwZToyNzQzOQ==</td>
<td align="left">ATGCCTCAACTGGATAAATTCACTTATTTCACACAATTCTTCTGGTCATGCCTTTTCCTCTTTACTTTCTATATTCCCATATGCAATGATGGAGATGGAGTACTTGGGATCAGCAGAATTCTAAAACTACGGAACCAACTGGTTTCACACCGGGGGAACAACATCCGGAGCAACGACCCCAACAGTTTGGAAGATATCTTGAGAAAAGGTTTTAGCACCGGTGTATCCTATATGTACTCTAGTTTATTCGAAGTATCCCAATGGTGTAACGTCGTCGACTTATTGGGAAAAAGGAGTCAAATAACATTGATCTCTTGTTTCGGAGAAATAAGTGGCTCACGAGGAATGGAAAGAAACATATTCTATTTGATCTCGAAGTCCTCATATAGCACTTCTTCCAATCCTGGATGGGGGATCACTTGTAGGAATGACATAATGCTAATCCATGTTCCACACGGCCAAGGAAACATCGTTTTTTAATCTCATTATGACCGGAAGAAATTCTTTCTCGATCAGAAAAGTGGAATGGAAGGCACTACAAGCAAGAAAGATATACTCCTTTTCAAATCGCCCCGCGAAGACGGACGTTCAGAAAGGTTCTCGAGATTCTCAATCGCTCTAGTGGGGAGGGATAGTGCTTGAAGGGAGCGAGACCAAGCCGAGCCACTAGGAGAGTAAGCCCTTCCCACGTGTCAAGTTGAATAAATGAATGCAGCCTCAACCAGAGAGATCAACCAGCGCCTTTTATTTTAGGCCGTCGGCGGTGCAGAACGCACTAATCCGCGACCAACAAGTTTTCCGAAACCGAACTGAATGGAATTCTTACTTTAAAAAAATGCTTGCTGAAAATCAAAGAAAGAAGGTCCATTTTCCCACGTAGTTCGTCGGTCAAACCAACGATTCTCAAAGTAATAGAGAGATCTTTTTCTAGTTAGACTTCTATCAATGCAATGAAAGAACCATCCCTTCCTATTTGTTTGTCCTGTCAGATATAAAGAAAATTAGACCCCAAAGAGCCCTTCTTTACCACTTTAGGGGGTGGGGGTGAAGGGGGGGTTTACATACAACCAAGGCAAAGTGGTTTATGATTGAATCTCAGAGGCATTCTTTTCATTTGGTAGATCCAAGTCCATGGCCTATTTCGGGTTCACTCGGAGCTTTGGCAACCACCGTAGAGGTGTGATGTACATGCACCCATTTCAAGGGGGTGCAACACTTCTCAGTTTGGCATCATATTTATCCTATATACCATGTTCGTATGCTGGCGCGATGTTATACGTGAATCAACGTTGGGAGGACATCATACCAAAGTCGTACA</td>
</tr>
<tr class="even">
<td align="left">VIT_00s0332g00060</td>
<td align="left">QmlvRmVhdHVyZVZhbHVlVHlwZToyNzg5Mg==</td>
<td align="left">ACTAAAAATTTAGCTCATTGAATAAGTAAACAATTGAATTGGATTCGGTTGGATGGTACCAACGAAATCGAGTGCTAACTCCCATTTCTTATTGAATTAACCGATCAACTTTCTATCGAACTTTTGGATTCGATAATTTTCGCAAAAAATTTCGTTTAATTTGACTTTATTATTATTATTATGAGAATAAATCCTACTACTTCTGGTCCTAGGGTTTCCACGCTTGAAAAAAAAAACCTGGGGCGTATCGCTCAAATCATTGGTCCGATCCTGGATGTAGCTTTTCCCCCGGGCAAGATGCCTAATATTTACAATGCGCTGGTAGTTAAGGGTCGAGATACTGTCGGTCAACAAATTAATGTGACTTGTGAGGTACAGCAATTATTAGGAAATAATCGAGTTAGAGCTGTGGCTATGAGTGCTACAGATGGTCTAATGAGAGGAATGGAAGTGATTGACACGGGAGCTCCTCTAAGTGTTCCAGTCGGTGGAGCGACTCTCGGACGAATTTTCAATGTGCTTGGAGAGCTTGTTGATAATTTAGGTCCTGTAGATACTCGCACAACATCTCCTATTCATAGATCTGCGCCCGCATTTATACAGTTAGATACAAAATTCTCTATTTTTGAAACAGGAATTCAAGTTGTAGATCTTTTAGCCCCTTATCGCCTTGGAGGAAAAATCGAACTACTCGGGTTGGCTGGTTGGAAGGATTGCTTTCTGCCGTCTCTCGTTCTTGACAGATCCGATCGGGTGTTCATAATCTGGAGTAAAAGGATTCGAACCTTTGCATGCCGGTACCAAAAACCGATGCCTTACCACTTGGCTATACTCCATACGGCCTTAGGCCTTTTGATAACTAATTAGAGTTGACATGAAATGGGTCACGAAAGAAGACGTATCCGATGAATGAATGATTCATTCCCCTCCCACTCACACGGGTTCTTCTATTGAAGAGGTTCCCCTTCTTCTATGTGAGGTGGGGCGTACGTAAGAGCGGAACCTAGATAGAACGCGGAGCGCCGGCCCCGGCCGCCGATACAGTTACCATGCTTACCAGCTCTTACCATTGCCGCCTATATATATAGATGGGAGGTAGGCGAGGGTAGCTTGCTTCTCTTCTCCTAGGCTACCTGCACATGTTATTCGCGAAGAAAAGGCAGCGAGCGGTTCTTCGCTTAGTAGACGGCCGCCGGACGACGACCGCTTCTTGTTCTCGCCCAGCCGCTTCTTTTTTTTTCTTATTATTTAGAATCCTAAACTAAAGAAAGAAGCTCCTGAATCAACGCAGGGTGGTCGTAGATCAGCAGGAGAACTGTACTAACCTGCCGCCGGTGACTTTATCATATCAGGGCTCCGCAGGAGAAGAAAGAAGGTCCGGGTCCAATTTATAATGAAGCGAAGGTCCCCCTTACTCCCTATTCTCTCTTAAAGCTTTATAAAGCTCGGCGAGAAAGGGTTGGTTAAGAACTATTGACTGTAAACCATAGCAGTTACAGTCACTCAATGGAGATGTTCGCCTTTGGAATGAAGATGGATGAGCAGGCACTTGAGCCTGGAACTGATTTAATTCGGGGAAAACATGGAACCGGTTACTATACTGGGGGAGGCGAGACATGACCGCGACTTAAGATTCAAGTACCGTTGAAAAGACTAAAGAAACGGGGGGA</td>
</tr>
<tr class="odd">
<td align="left">VIT_00s0332g00110</td>
<td align="left">QmlvRmVhdHVyZVZhbHVlVHlwZToyNzg5Ng==</td>
<td align="left">CAGTCATTATCCTTATAGAAGGCACTCACAAAGATAGAGTTACAGAGGTAGAAAAAAAAATGGATTCATCATCTGTTACATCTACCCATCCCGTGATTCCGAATCAACAGACCCAATTTCCGAGATATGGCAACTCTTTTATAAAGTAATAGTAATAAGAACTGTGTCGTGTAACTTCATTCACAAATGAAACTCCGGAAAAGCAAAATGGAAAATAATTTTTTTTTTTTCAAATTGATTTTCTTTTTCAATTGTATCAAATCCATTTTCTATTTTGGAACGAACAATGACCAACTAAGACTTTGACTTGACTTTAACTCATTAGATTCAGTAGTAGGTTCATTAATTGCACATAATATCTCGAATGAGAAGAAATTATCGCACGATCGCTCGAATAAATCAAACCTTTATTTCGGTCTAACTCGATTCTAAGACTACTAATTCCTATGTTCTTTTCCTTTTAGGTTCTCTTTTATTTCAATATTAGGGAGTCCCACCCTTGTAGCTAGGTCAAGAAAAAACCTTTGGATCTAATACAGCAATGCTTGCATCACGAATATTGATTGTTCCAATTATTTTATTGTTTGTTCTGTTTCTTTCATCGAAGACTATCTACTAGAATACTTGTTATTGTACGACCAACCAATTCCTTAAAATGAAAGGATTCGAACAAAAAGAAAAAAACCTTTTCTACAACCATGAAAAGGGGCTTTCTAGATTTCGCATCTAATTGAATTGTGGGAATATGAGAATCTCTTTCATGAATTATTAGAACCCAACTCGTCGGACTCAATCTTTACCAGATCTTCAGTTTCTAGTCTGAAGCCAGTAATGGAAAGAAACCCTATACTAAAGAAGAGGAATTTTCTATTGAATGGCACATGGTTCTGCATTGACAAGCAACAAGAAAGGAAGAAAGAAATGTTGTATCACTTCTTTTCTATACTGATTGAAATTGCGTTGCTGTGTCAGAAGAAGGATAGCTATACTGATTCGGTATACTCACACCCTCGGTACACTATTGAGGATCTCACAAAGATGAAATCTCAGTGAATTTCCATTTACTGATCCCATCTTTCACATAATAAATCCCCCCTTTGACTGTACATGAATATGTGGAGCTCAGCATGTCTGGAAGCACGGGAGAATGTTCTTTTGCTGATATTATTACCTGTATTCGATACTGGGTCATTCATAGCAAACCTATACCTTCCCTATTTACTCGACGAATTAAGGGGGTTGGTTATTCGTCAGCACGGGTTTTGCTTCCTCTTCTGGTATAGGTTTGGGAGCCCTCGGCCAAACGAGTATTTTACAGAGAGCCGACAAGGAATGGAATTAATAACTGGCCTGGCCGTTTTGATCCTTTGGCACAACTCGATTCATTTAGTAGATCCTTTTAGAAGGTCTCCCTCTTAGTAAGCTTAGATCGAACCTATCCCCTTTTTACAGTGCGATGGTTGGCTGTTCACGGACTAGCTGTACCTACCCTTTCTTTTTTGGGGTCAATATCAGCAATGCAGTTCATCCAACGAGAAACCTAATCCGAATTATAGAGCTACGACACAATCAAACCCGAACAAACAAAATGTTGAATTGAATCGTACCAGTCTCTACTGGGGGTTCTTACCCTATATATGACGAGCTGAACTAGCTGTTTTCTTTTAAAATTCTTTCTAAAATGAAGAGAAAGAAAGAGAATAGTAATAGTAGGAATTCTCTTATCCCATTCGGAAGGATCTCATCTCATAATTATCCATGACTGTTTATGTCTCTAGCATGACCACTTGATGAAATGTGGAGGGAAGTGGGGTAAATGGCCGATACTACCGGAAGGATTCCTCTTTGGCTGATAGGTACTGTAACTGGTATTCCTGTGATCGGTTCAATAGGTCTTTTCTTTTACGGTTCATATTCCGGATTGGGTTCATCCCTGTAGTAATCGGATGAACTGGGTTGT</td>
</tr>
<tr class="even">
<td align="left">VIT_00s0332g00160</td>
<td align="left">QmlvRmVhdHVyZVZhbHVlVHlwZToyNzkwMA==</td>
<td align="left">CGGAATGATCGGTCGGACAGAGGCTAGGGCTGCTCCTTCCCACCGGCGTCCTTCTTGTGTGTCGGAGATATAAAGCGAGTGCACGGAAAGAACGGGAACTGGGTCGATCTATTCTATGGCGAAGCATCCGAAGCATAACTGCACACTCACACGATCTTTGCCGAGAGATAGGAGCATTCGGTGGAACCGGTGAACTACACTTGCTTCTGGATAGATGTGTGGGACAGAGGGCTCGTGGTACCTGCTGCCCACCCTTCCCCCTCTGCTTTGAGAACCGTGTGAACGGAGAGTGGGCAGAAGGGAAGGAGGTCCTCATACGGAGTCGCACACTTACTTGAGCAGTGCGGGAGACTGGGGAATGGGTCGAGTAAACTCCCCTGGGGCCGAAAAAATAACAGACGAAGCTTGACTTAGATAGGGCTTTGATTTGATTGGTATTGATTTTTTCTTAGTGATTGGAAAGAAGGGCGCATGGGTTATAGAAAGGGAAGGCGGAGGTAGTACTTCGAATGGCGAAGAGAGGCGGCTCGGCAGGGAAGGAATGGGAAATCGTCAAGGATGACTTCTTTGTTGGCGAAACGAAGGGCTAAATAGCGCCAGTGATTCTCTGAGCCGGTCTGGATTTCCAACGCCTCGGGAAACTGGTCGAGATAGATGACAGCACCTTTTTCCTCTCTTCCTTACCGGGAGGGCAATCTTATCTTATGGCCTTCACCTCCAGCCCGGCCCAGAAATAGAACCCAGCCCCTTTTTGATCCATTCATTTCTGCAAGCAGGGAGGATCCAGAGCTAAGCCCCCCTCCTATTCGAGGTCGGGTGGCCTCGCCCCACAAGCACCTAACCTTCCTTTTGCTCTTCCTTCGGTTTGCCTCCACGAACGCGCCACCTAGGAGCCCTACTCATATTCCAAAGGCGCTACTTCAATTCAAGCGCAAGCCCCCCTTCTATTGCATAACCTAAAAAAGTGTTAAGCAAAGTACCAAAGTGGTTGCGGGAACGATACGCTTTGGTTACCGGCGAACGCTTCCAAATCTCGAGTTTCCGGCTGTTTCCTAAATTGAAGTAGCCTTTCGTCGCCCTACCAAATGAAAGAAGTCACTATCAAACAGCTCGCCCTACTGAAGTACCAAAGGTGCGCTCCGCCCGGTGACTAAGAAATAAATTGGTTTGCGCTTTCATTGAAGTGATGAGGGTGATCTACGATCAGGCTGGTCGGAACGATCAGGGGGAAGTAGGGCTCCTATTGAAACGCTTTCTCTCCCTCAAAAGGCGACCAGCTTTCAATACTAAGCCGTGTTATGGTTTCGTTGTTCGTTCCGTCGTCGACCCTATCTATGTTTCTACGATGGCCCAAGAGCACGCTCATTCTTCAGCCGTAGAGAGACTTTTGAATTGCGAGGTACCATTACGAGCTCAATATATACGAGTGTTATTCCGTGAAATAACTCGAATTTCAAATCATTCACTTGCTTTAACTACTCATGCTATGGATGTGGGAGCATCAACTCCGTCCCTGTGGGCTTCTGAGGAGCGGGAGAAATTGTTGGAATTCTATGAAAGAGTCTCGAGAGCCAGGATGCATGCCAGTTTCATACGACCAGGTGGAGTGGCACAAGATCTGCCTCTTGGCTTATGTCGAGATATTGATTCCTCCACACAACAATTTGCTTCTCGTATCGACGGATTAGAAGAGATGTCAACCGGCAACCGTATCTGGAAACAACGATTAGTGGATATTGGTACTGTCACTGCACAGCAAGCAAAGGATTGGGGATTCAGTGGTGTAATGTTAAGAGGTCGTGTACCAGAGTGGATCGCTATGATCGTTACTGTATCCGTATCGAAGATGATGCGACAAAGTGTTCGAATCATTGTGCAATGTCCTAATCAAATGCCTAGTGGCATGATCAAAGCCGATGATCGTAAGCTATGTCCTCCATCACGATGTCGAATGAAACTATCCATGGAATCACTCAGGTTAAGCTCCCATTGGGAACCCCAATTTCTTTCTTCGGAGCCGTTTCTTTTCCCGTCCCCCCGCCCCGGCATAGCGCTTCGCTTCCGGTTCTTCGGAAGAATCAACTGA</td>
</tr>
<tr class="odd">
<td align="left">VIT_00s0396g00010</td>
<td align="left">QmlvRmVhdHVyZVZhbHVlVHlwZToyODA4MQ==</td>
<td align="left">TTTCCATTCTTCGGTTATGTATGGAAAGATAGAAAGATAATGACCACAATTTTTGGTATTCACTTAATCTTGTTAGGTATAGGTGTTTTTCTTCTAGTATTCAAGGCTCTTTATTTTGGGGGCGTATATGATACCTGGGCTCCCGGAGGGTTAAGAAAAATTACCAACTTGACCCTTAGCCCAAGTGTTATATTTGGTTATTTACAAAAATCGCCCTTTGGAGGAGAAGGATGGATTGTTAGTGTGGACGATTTTGAAGATATAATTGGAGGACATGTATGGTTAGGTTCCATTTGTATACTTGGTGGAATCTGGCATATCTTAACCAAACCCTTTGCATGGGCTCGCCGTGCACTTGTATGGTCTGGAGAGGCTTACTTGTCTTATAGTTTAGGTGCTTTATCTGTCTTTGGTTTCATTGCTTGTTGCTTTGTCTGGTTCAATAATACCGCTTATCCTAGTGAGTTTTATGGGCCAACTGGACCAGAAGCGTCTCAAGCTCAAGCATTTACTTTTCTAGTGAGAGACCAACGTATTGGGGCTAACGTGGGATCCGCTCAAGGACCTACTTGGTTTAGGTAAATATCTCATGCGTTCCCCGACTGGAGAGGGTGGTCTACGATCAGATGGAGGAGAAACTATGCGCTTTTCAAAAATGGTCGGAAGAACGGGTTTCAAAAAATATATAAATATATATATATATATATATATATATATATTTATATATTACCGGTTGGCTGGTTTTTATGCAAAAAAGACAAAACGGGATTGGATTTCCACTCATAAGGAAGGAAGGTTAGATACCTTTGATAGGGTTGTATTTTTGGTTGAAATGGGATGGTGTTCAAACTCCCGAAATGCAGTCTATCTAGCTGATCTACGACCACCCTTTCTGACTGGACTGACTCGGGGAAGGGTCAATCTCCTCCGGAGCATGCTTCACAGCCACTCATTCGTCCTGACCAAATAGTAGAATATATCTCTCTGCGATTCTTTTCTCCGCTAATCACCCCTTCCCAACCAGCCCGCCCGATCGATTTTGTAAGATCGGCTAATTCAAAGGGGTTAATCTGGTCCGAAGTTGTCGGAACGAATCATTTGCTTTATCGAACAGGTTGGCCCCCTATTTTCAACCATTACAGCTGGCGTCGACCAGCTTTGCGATACGGACGGACACGAAGAAGAAGGCAGGCAGCGGAAATGCAAAAGAGAAGAGCAAAGATTCCCGACCCAGAGCATGTTATTGACTCAACCACAAATCTGTTGAATGAAGGCCATTTAGGGCTTACTCTCAGCCACTAGTTAGAAGGAAATCCCGTGATCTACAACCACCCTTATGCAGTAAAGAAAGCAAGTGAGGTGGGCTAAGATTCCTAACGTAACAGGATTCGATGTTGCACTATCTTCAACTCTTCTCGGGATCCGGAGTTTTTCGAGAAAAGGTCCCATCCTTCAATATCATGATTGGGTCGACCAGGCCAGATCATGAAATCATGAGTGAATAGAAAATCGAAAATGTCAATAGCAGTTGCTGTTCCAGCTGAAATACTTGGAATAATTCTACCACTTCTACTAGGAGTAGCCTTTTTAGTGCTAGCTGAACGTAAAGTA</td>
</tr>
<tr class="even">
<td align="left">VIT_00s0505g00030</td>
<td align="left">QmlvRmVhdHVyZVZhbHVlVHlwZToyODI3OA==</td>
<td align="left">CTTCAGAGAAGAGAAAGGAGGCATTCGTCAGCGTAGGAAGGCCAACCGAGCGAAGCTCTTTCTTTTTTTTTGGTCGTGCCGTGAAGTTCAATTGTATCGTATGTTAGTTAGAGAGGTTGGCGAACTACTACGATCTATAGATTCCCCATCTATCTATATCCAATCCCAACGAGAATAGAAGCAAGTCGCTTCCGGCTTGGCTTGTAGTCGTAGTCTTTTAGCTTATGTAGTGGTCGGCCTTCCTACACGCACGCGCCCGCCAAAATGCCTCCTTGGTTCTGGAGACGAAGCCAAGCCGATACATACGATAGGCGAAGGAAAGACCCCCCATTTCATAGCACCTGGGGAACGGGGGTTCGCGTCTATACCTATAGTGTGAGAGTTGGGCCTACCCACCCTATTTGTTCCATGATCTATGGGTCTACTGGAGCTACCCACTTCGATCAATTAGCCAAGATTTTGACCGGATACGAAATCACTGGTGCTCGATCTAGTGGTCTTTTTATGGGGATTCTCTCTATCCCTGTAGGATTCCTATTGAAGATCACTGCAAAAAACCGCCTGGAAGTCCGAGGACCTTTAGTACCGTACCCTACCCTCGAACTAGCAGCCTTCGCGCCAAGCAAGACCGCCCTTGTCCCTTTCCTTTCTCCATTCCGCCTCCTTCTTTGCTTTTTTCGTTTCCGGGTTTATGGATTGGATTCAGTCAGCCCTTTCGTCGCCCTACCAAACGAAAGAAAAGTGGTCGTAGATCATTTTATTTGTTTTTCATCGAAAACGAAGAAGGCCGAGGATGGCCTACGCTGCCTTCGGGCTTTGCCTGCCCTTTATAATAAAGAATTCCGGCTCGGCCCGGGAAAGCGCTGGCAACAACAGAAAGGAAGGGGTCCATGTAGCTGCTGCGCCCGCCCCCTTCTTAGTCAATGGGGCAGCAGGTTCGGCATCTACTATAAAAGAGAGAATCCACTTCAGATAACCACGCCTCTGCTAGGCAGCCCCCGAAGAAAAGCCCCCTCCCCTTCGGACTCCATATCTCTTTTGACTTTATATATGTGGGCACCTGATATCTATGAGGGTTCACCCACCCCGGTTACAGCATTCCTTTTTATTGCGCCTAAAATCTCTATTTCTGCTAATATTTCACGTGTTTCTATTTATGGTTCCTATGGAGCTACATTGCAACAAATCTTCTTTTTCTGCAGCATTGCTTCTATGATCTTAGGAGCACTGGCCGCCATGGCCCAAACGAAAGTAAAAAGACCTCTAGCTCATAGTTCAATTGGACATGTAGGTTATATTCGTACTGGTTTATCATGTGGAACCATAGAAGGAATTCAATCACTACTAATTGGTATCTTTATTTATGCATCAATGACGATAGATGCATTCGCCATAGTTTTAGCATTACAGCAAACCTGTGTCAAATATATAGCGGATTTGGGCGCTCTAGCCAAAACGAATCCTATTTCGGCTATTACCTTCTCCATTACTATGTTCTCATACGCAGGAATACCCCCGTTAGCCGGCTTTTGTAGCAAATTCTATTTGTTCTTCGCCGCTTTGGGTTGTGGGGCTTACTTCCTAGCCCCAGTGGGAGTAGTGACTAGCGTTATAGGTCGTTGGGCGGTCGGAAGGTTGCCACGACTTACCGAATCAGTTGCGACACGGATGGGAATGCATGCTACGAAACATAGGGTCGAGTCTAATACATCAACCGTCTACTCAATATCCTTGTACGAGTCCACAATCACTACACGAGATGAACCTTGGTTTGGTGAATTGAAGTTGGCCTTAGGTGTAATAGGACTCCCAGTTACTGCGCGCGATCGTATACTGAGGTGCTCCCCGCCGGTTGTTGGAACGACGCGAGCCGGGCCGGGCCCCGATTCAAAAAGATGAAGGGCCAAAAAGTATAAATAGGGGGTTAAAAATTCCCCATCTCATTAGGGGAGGAAAACGAATCGGCATCTCGATGTGATACAGCCTTTTCTATTTGAGTTGGGAAAGAACGGCGAAGTCCATCCGAACCGTCCAATGAAGAATAAGAGGAGAGCAAAGCGCCAATGGCGCGCGAAGCGCATGCGGAACGGGCACGGAAAAAAAATAAAGAAGTGTGGAGGGTGGTCGTAGATCAGTAGGGCTCATTCCCTTCGCTTCCTGGGCCCAAAGCAGTGCAGTCTTTCCTGGCCAAATCAAGGATTTGGGGCTTCTTGCTACGCTACTTAACTATATAAATCCATTTTCTTTAGTAATATATATGAATAGAAAGATAGATCCATCCATCTATCCTATCCGATTTAGATTTTGATATCTAAAAAACAATAGATTTCATTCAACGTTTGATTCAAAGAACTGCGCTTAGCCCCCCCGCTCATGAAACGGCTCTGCTGCAATGGATGGCAGAGGGTCCGTAGTACCCAAAGCACTGGAGTGATCCAGTAGCCGGGAAGGGGCCTAGAAGTGCCTACTACTACACCACACTACACTTGGCTCTACACATTTACAGAGCTAACCCCTGTCCAGTGCCTGGCAGAGCTAAAGGGGCTTCAATCCTTACTCCTTATCCCCATCTTCGCCCAGGCTAACGGGGCCTTCCTTATTCAGGGGGGAGAGTGGAGCCTCGAGAAGCACTGTTGAGAGGAAGATCCTTGGCCCCTCTTCATTCTCTACAGGGTTCCAAACCTTTCTTCAACATAGGTGACAACGAGCAGGGCAGAGATGGAAGAGATCGATCACGGGAATAAGAAGCAAGCTCGCCTTCCTTTTTGATCATTTTGATAGAGGGGGATGGAGAAAGTGGACAAAAAAGACTCGCATTTCCCAGTCGAGGGTTCCTTTTTCGTCTCGCATTTATGGAAAAGAAGAATATTGAAAACGCTCCTAACCCAACCCCTTCCTTCGTAGAGCCGTGTATTGTAAGTGATCCGAACCTGCCCGGAGTGAGCCCCCCATAGAGGCAAGTGAAGTTGGTGAGCCGTATGATGGGCAACTATCTCCTGCGGTTCGGAGAGGACTCAGCTGTTAGTTAGTACCCCCTTGGATTCGGGGTGGACCCTTTCACTCTATTTTATTATATACGCTTCGCGAAAAGAATGTTTTTTGATACACCTAGGACATGGATTCTATATGAACCAATGGATCGTGACAAGTCGTTACTACTAGCAATGACTTCCTCTTTCATTACTTCATCCTTTCTATATCCCTCTCCCTTGTTCTCAGTTACTCATCAAATGGCACTCAGTTCATATCTTTAAGTTTGATCATTGACAAGGTTCAAAGAAAGGGTTGGCCATTGACAA</td>
</tr>
<tr class="odd">
<td align="left">VIT_00s0505g00060</td>
<td align="left">QmlvRmVhdHVyZVZhbHVlVHlwZToyODI4MQ==</td>
<td align="left">ATGGAAACAGCAACCCTAGTCGCCATCTCTATATCAGGTTTACTTTTTAGTTTGACTGGGTACGCCTTATATACCGCCTTTGGGCAGCCCTCTCAACAACTAAGAGATCCATTCGAGGAACACGGGGACTAGTTGAAGTAATGAGCCTCCCCATATTAGGGGGCTCATTACTGAAATTGAGATAATGAAAAAGAATTTCATCTTTTTAGTTTTTTCTTTAGGCGGTTCTCGAAAAAAGATAGCGAAAAAAATTATCCCTAGAGTCGAGACTAAGAGGAATGTATAAACCAATGCTTCCATATATTCGATCGTGGTT</td>
</tr>
<tr class="even">
<td align="left">VIT_00s0873g00020</td>
<td align="left">QmlvRmVhdHVyZVZhbHVlVHlwZToyODU5OA==</td>
<td align="left">GGATCCTGGTCTGGACTCTTATGTGTGTATTTTATCTCGTTTGGCTTTTGGAATCACATCCTGCAGTGGTTGGAACAGCTCGCAAAATCCAAGCACTTCACCTACTTTATTGCCCCCAACCCTTTATCGTACCTTTCTTGAAACAGAATGGTTTCATGCTCCTTCATCGATTGGTTATTCCTCTTCGTTCGTCTCTTTTTTTTCAATGTCGGTCTCGATTAGTTCACAAGATTGAATGGCCAATTCTCCTCCGGACCCTCGATTCGATTATTTTCCAAGAATGTAATGTTAAGCCGGGTATGTATGTATCTGGGAGGAACTTTAAAGAAGGGCTTTCGGTTTTTTGCACCCTCTTTTGGTCTTGCAGCTTTATATTTATATTTCGTATATAATGAAGCGGGGGCTATCCGCCAGTAGCGTTGCGGATTCTCGATGAGGGGGTGGGTTCTTTCTTCAAACACTTTTTTCCGGTATGTCGCTCTGCGAGCAAGGAGCGAGAGAACCAAGTGGGCTATGGTGATGTCAGAATTTGCACCTATTTGTATCTATTTAGTGATCAGTCCGCTAGTTTCTTTAATCCCACTCGGTGTTCCTTTTCCATTTGCTTCCAATAGTTCGACCTATCCAGAAAAATTGTCGGCCTACGAATGTGGTTTCGATCCTTTCGGTGATGCTAGAAGTCGTTTTGATATACGATTTTATCTTGTTTCAATTTTATTTATTATTCCTGATCCGGAAGTAACCTTTTCCTTTCCTTGGGCAGTACCTCCCAACAAGATTGATCCGTTTGGATCTTGGTCCATGATGGCCTTTTTATTGATTTTGACAATTGGATCTCTCTATGAATGGAAAAGGGGTGCTTCGGATCGGGAGTAA</td>
</tr>
<tr class="odd">
<td align="left">VIT_00s0904g00010</td>
<td align="left">QmlvRmVhdHVyZVZhbHVlVHlwZToyODYwOQ==</td>
<td align="left">ATAACCTCTCATCATCTATCATTACAACCTGATCAATACCACAAGCTTGGGGTGAGTTCTAGGCTCTAGAAATGGCACAAGCTATGGCTTCAATGGCTGGGCTACGCGGGTCCTCTCAGGCCGTCCTGGAGGGCAGTCTCCAGATCAGTGGCTCAACCCGATTGAGCAGCGGCAGCAGCAGCAGGGTGGGTGTGGCACGCCCTGGGTTCAGTGTTAGAGCTCAGCAGGCTCCAGTTGAGCCTGAGACTAGCCGTCGTGCAGTGCTGGGCCTCGTTGCTGCTGGCCTGGCCTCGGGCTCTTTTGTTCAGGCTGTGCTTGCTGAGGCCAAGTCCATCAAGGTTGGCCCCCCTCCCCCACCCTCTGGTGGATTGCCTGGTACTCTAAACTCTGATGAGGCAAGGGACCTTGATCTGCCATTGAAGGACCGGTTCTTCATTCAGCCACTGCCCCCAGCTGCAGCAGCAGTTAGGGCGAAGGAGTCAGCTAAGGACATTGTTGGGGTGAAGACATTGATAGACAAGAAGGCTTGGCCATATGTCCAGATGGATCTCCGTCTCAAGGCAGAATATCTCCGGTACGACCTCAACACCGTCATTTCTGCAAAGCCCAAAGACGAGAAGAAATCCCTGAAAGAACTCACTGGAAAGCTCTTCTCGACCATCAGTGATTTGGACCATGCGGCAAAGATTAAAAGCTCCCCTGACGCAGAGAAGTACTATGCAGCGACTGTATCCACCCTGAACGATGTCCTTTCCAAGATTGGTTAAGAAGAATCCACTAATCTTGGTGTAAGCTTTAGTTTTCACTGCATGGGACATGTGCTTGGATGTTATTTTAAGGAAATCTAATTTGAAAAAAAAAAAAACCCCTCATGATA</td>
</tr>
</tbody>
</table>
<div id="get-ids-from-aliases-using-sparql" class="section level2">
<h2 class="hasAnchor">
<a href="#get-ids-from-aliases-using-sparql" class="anchor"></a>Get ids from aliases using SPARQL</h2>
<p>Genes might have different aliases, all of which are defined in VESPUCCI as annotations. Since annotations (for both genes and samples) are represented as RDF triples using different ontologies, we can perform a <a href="https://en.wikipedia.org/wiki/SPARQL">SPARQL</a> query to retrieve all the biological features (genes here) that has an <strong><em>alias</em></strong> <a href="https://www.ebi.ac.uk/ols/ontologies/ncit/terms?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FNCIT_C41095">NCIT_C41095</a> term on the NCIT Ontology, <a href="http://purl.obolibrary.org/obo/" class="uri">http://purl.obolibrary.org/obo/</a>) equal to the ones specified in the list.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">my_query</span> <span class="op">&lt;-</span> <span class="st">"SELECT ?s ?p ?o WHERE { {?s &lt;http://purl.obolibrary.org/obo/NCIT_C41095&gt; 'B9S8R7'} UNION {?s &lt;http://purl.obolibrary.org/obo/NCIT_C41095&gt; 'Q7M2G6'} UNION {?s &lt;http://purl.obolibrary.org/obo/NCIT_C41095&gt; 'D7SZ93'} UNION {?s &lt;http://purl.obolibrary.org/obo/NCIT_C41095&gt; 'B8XIJ8'} UNION {?s &lt;http://purl.obolibrary.org/obo/NCIT_C41095&gt; 'Vv00s0125g00280'} UNION {?s &lt;http://purl.obolibrary.org/obo/NCIT_C41095&gt; 'Vv00s0187g00140'} UNION {?s &lt;http://purl.obolibrary.org/obo/NCIT_C41095&gt; 'Vv00s0246g00010'} UNION {?s &lt;http://purl.obolibrary.org/obo/NCIT_C41095&gt; 'Vv00s0246g00080'} UNION {?s &lt;http://purl.obolibrary.org/obo/NCIT_C41095&gt; 'Vv00s0438g00020'} UNION {?s &lt;http://purl.obolibrary.org/obo/NCIT_C41095&gt; 'Vv00s0246g00200'} }"</span>
<span class="va">sparql_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_sparql_annotation_triples.html">get_sparql_annotation_triples</a></span><span class="op">(</span>target <span class="op">=</span> <span class="st">"biofeature"</span>, query <span class="op">=</span> <span class="va">my_query</span><span class="op">)</span> 
<span class="va">sparql_results</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span>  
<span class="co">#&gt;  [1] "QmlvRmVhdHVyZVR5cGU6MjcwOTc=" "QmlvRmVhdHVyZVR5cGU6Mjg1MDQ="</span>
<span class="co">#&gt;  [3] "QmlvRmVhdHVyZVR5cGU6Mjc1NTU=" "QmlvRmVhdHVyZVR5cGU6Mjg2MzA="</span>
<span class="co">#&gt;  [5] "QmlvRmVhdHVyZVR5cGU6MjY0MzM=" "QmlvRmVhdHVyZVR5cGU6MjY4ODM="</span>
<span class="co">#&gt;  [7] "QmlvRmVhdHVyZVR5cGU6Mjc0MjM=" "QmlvRmVhdHVyZVR5cGU6Mjc0MjY="</span>
<span class="co">#&gt;  [9] "QmlvRmVhdHVyZVR5cGU6MjgxODM=" "QmlvRmVhdHVyZVR5cGU6Mjc0Mzc="</span></code></pre></div>
<p>alternatively we can use the wrapper function <code><a href="../reference/get_ids_from_alias.html">get_ids_from_alias()</a></code>:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">aliases</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'B9S8R7'</span>,<span class="st">'Q7M2G6'</span>,<span class="st">'D7SZ93'</span>,<span class="st">'B8XIJ8'</span>,<span class="st">'Vv00s0125g00280'</span>,<span class="st">'Vv00s0187g00140'</span>,<span class="st">'Vv00s0246g00010'</span>,<span class="st">'Vv00s0246g00080'</span>,<span class="st">'Vv00s0438g00020'</span>,<span class="st">'Vv00s0246g00200'</span><span class="op">)</span>
<span class="va">sparql_results_check</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_ids_from_alias.html">get_ids_from_alias</a></span><span class="op">(</span>target <span class="op">=</span> <span class="st">"biofeature"</span>, alias_names <span class="op">=</span> <span class="va">aliases</span><span class="op">)</span></code></pre></div>
<p>and check results are the same:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/identical.html">identical</a></span><span class="op">(</span><span class="va">sparql_results</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span>,<span class="va">sparql_results_check</span><span class="op">)</span>
<span class="co">#&gt; [1] TRUE</span></code></pre></div>
<p>Now we add the genes id retrieved using either the sparql query or the wrapper function <code><a href="../reference/get_ids_from_alias.html">get_ids_from_alias()</a></code> to the original genes id list.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gene_ids_module</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">bf_ids</span><span class="op">$</span><span class="va">id</span>, <span class="va">sparql_results_check</span><span class="op">)</span>
<span class="va">gene_names_module</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">bf_ids</span><span class="op">$</span><span class="va">name</span>, <span class="fu"><a href="../reference/get_biofeature_id.html">get_biofeature_id</a></span><span class="op">(</span>id_In <span class="op">=</span> <span class="va">sparql_results_check</span><span class="op">)</span><span class="op">$</span><span class="va">name</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="create-a-module" class="section level1">
<h1 class="hasAnchor">
<a href="#create-a-module" class="anchor"></a>Create a Module</h1>
<p>We are now ready to create our first module. If we pass a list of genes (BiologicalFeature objects) it will automatically retrieve the “best” conditions (SampleSet objects).</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">module_1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_module_bf.html">create_module_bf</a></span><span class="op">(</span>biofeaturesNames <span class="op">=</span> <span class="va">gene_ids_module</span>, 
                             normalization <span class="op">=</span> <span class="st">"limma"</span>, 
                             top_n <span class="op">=</span> <span class="fl">50</span>,
                             useIds <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p>A module can be thought as a matrix composed by BiologicalFeature objects (genes in our case) as rows and SampleSet objects (conditions contrasts in our case) as columns. Each SampleSet, as the name suggests, is composed by 1 or more Sample objects. Since we are using the legacy normalization each SampleSet is composed by 2 samples, one Test sample and one Reference sample and the value associated is the logratio between the Test and Reference for a specific gene (BiologicalFeature).</p>
</div>
<div id="plot-a-module" class="section level1">
<h1 class="hasAnchor">
<a href="#plot-a-module" class="anchor"></a>Plot a module</h1>
<p>The easiest way to visualize a module is using a heatmap. The Plot object wraps a module and can return different plot type with different format (the Plotly JSON format or HTML). To see an heatmap embedded in rstudio we can retrieve the HTML code of the Plot (that uses Plotly) and display it directly into the viewer.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/view_plot.html">view_plot</a></span><span class="op">(</span>module <span class="op">=</span> <span class="va">module_1</span>, 
          normalization <span class="op">=</span> <span class="st">"limma"</span>,
          plotType <span class="op">=</span> <span class="st">"module_heatmap_expression"</span>,
          alternativeColoring <span class="op">=</span> <span class="cn">FALSE</span>,
          min <span class="op">=</span> <span class="op">-</span><span class="fl">6</span>, max <span class="op">=</span> <span class="fl">6</span><span class="op">)</span></code></pre></div>
<p>When using the heatmap it is possible to use the button Get Module IDs in order to copy on the clipboard BioFeatures and SampleSets IDs after having selected a part of the module with the left mouse button. For example after having selected one small block of up-regulated genes and clicked the Get Module IDs button, we can paste a JSON structure to be used for create a new module:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">json_selection</span> <span class="op">&lt;-</span> <span class="st">'{"sample_sets":["U2FtcGxlU2V0VHlwZToyMTg3Mg==","U2FtcGxlU2V0VHlwZToyMTg2Mg==","U2FtcGxlU2V0VHlwZToyMTAyNg==","U2FtcGxlU2V0VHlwZTo2NTU2","U2FtcGxlU2V0VHlwZToyMTAyNw==","U2FtcGxlU2V0VHlwZToyMTAyOA==","U2FtcGxlU2V0VHlwZTo1ODc2","U2FtcGxlU2V0VHlwZTo1ODg2","U2FtcGxlU2V0VHlwZToyMTAyOQ==","U2FtcGxlU2V0VHlwZTo1ODU4","U2FtcGxlU2V0VHlwZTo1ODYy","U2FtcGxlU2V0VHlwZTo1OTE3","U2FtcGxlU2V0VHlwZToxOTEyNg==","U2FtcGxlU2V0VHlwZToyMTg1OQ=="],"biological_features":["QmlvRmVhdHVyZVR5cGU6Mjc1NTU=","QmlvRmVhdHVyZVR5cGU6Mjg1MDQ=","QmlvRmVhdHVyZVR5cGU6MjgxODM=","QmlvRmVhdHVyZVR5cGU6Mjc0Mzc=","QmlvRmVhdHVyZVR5cGU6MjgyODE=","QmlvRmVhdHVyZVR5cGU6MjgyNzg=","QmlvRmVhdHVyZVR5cGU6MjgwODE=","QmlvRmVhdHVyZVR5cGU6Mjc5MDA=","QmlvRmVhdHVyZVR5cGU6Mjc4OTY=","QmlvRmVhdHVyZVR5cGU6Mjc4OTI=","QmlvRmVhdHVyZVR5cGU6Mjg1OTg=","QmlvRmVhdHVyZVR5cGU6Mjg2MDk=","QmlvRmVhdHVyZVR5cGU6Mjg2MzA="]}'</span>

<span class="va">submodule_ids</span> <span class="op">&lt;-</span> <span class="fu">RJSONIO</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/RJSONIO/man/fromJSON.html">fromJSON</a></span><span class="op">(</span><span class="va">json_selection</span><span class="op">)</span>

<span class="va">submodule_1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_module.html">create_module</a></span><span class="op">(</span>biofeaturesNames <span class="op">=</span> <span class="va">submodule_ids</span><span class="op">$</span><span class="va">biological_features</span>,
                             samplesetNames <span class="op">=</span> <span class="va">submodule_ids</span><span class="op">$</span><span class="va">sample_sets</span>,
                             normalization <span class="op">=</span> <span class="st">"limma"</span>, useIds <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p>and visualize it</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/view_plot.html">view_plot</a></span><span class="op">(</span>module <span class="op">=</span> <span class="va">submodule_1</span>, 
          normalization <span class="op">=</span> <span class="st">"limma"</span>,
          plotType <span class="op">=</span> <span class="st">"module_heatmap_expression"</span>,
          alternativeColoring <span class="op">=</span> <span class="cn">FALSE</span>,
          min <span class="op">=</span> <span class="op">-</span><span class="fl">6</span>, max <span class="op">=</span> <span class="fl">6</span><span class="op">)</span></code></pre></div>
<p>Biological Features (genes) and Sample Sets (contrasts) have no particular order in a module. But since the heatmap is clustered we can retrieve the sorted list of Biological Features and Sample Sets respectively using the <code>sorted = TRUE</code> parameter.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="va">mod_tmp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_module_bf.html">create_module_bf</a></span><span class="op">(</span>biofeaturesNames <span class="op">=</span> <span class="va">gene_ids_module</span>, 
                            normalization <span class="op">=</span> <span class="st">"limma"</span>, 
                            top_n <span class="op">=</span> <span class="fl">50</span>,
                            sorted <span class="op">=</span> <span class="cn">TRUE</span>,
                            useIds <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co"># my_sorted_indexes &lt;- plot_module_heatmap(module = module_1, normalization = "limma", plot = FALSE, sorted = TRUE)</span>
<span class="co"># mod_tmp = module_1[match(my_sorted_indexes$sortedBiofeatures,rownames(module_1)),</span>
<span class="co">#                    match(my_sorted_indexes$sortedSamplesets,colnames(module_1))]</span></code></pre></div>
<p>Alternatively you can use directly any R function to cluster data and retrieve the sorted rows and columns of the matrix, e.g.:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pheat_cl</span> <span class="op">&lt;-</span> <span class="fu">pheatmap</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/pheatmap/man/pheatmap.html">pheatmap</a></span><span class="op">(</span>mat <span class="op">=</span>  <span class="fu">SummarizedExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">assay</a></span><span class="op">(</span><span class="va">mod_tmp</span><span class="op">)</span>, silent <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="va">mod_tmp</span> <span class="op">&lt;-</span> <span class="va">mod_tmp</span><span class="op">[</span><span class="va">pheat_cl</span><span class="op">$</span><span class="va">tree_row</span><span class="op">$</span><span class="va">order</span>,<span class="va">pheat_cl</span><span class="op">$</span><span class="va">tree_col</span><span class="op">$</span><span class="va">order</span><span class="op">]</span></code></pre></div>
<div id="export-it-in-another-format" class="section level3">
<h3 class="hasAnchor">
<a href="#export-it-in-another-format" class="anchor"></a>Export it in another format</h3>
<p>Since we created a <code>data.frame</code> from the module we can now export it in another format (e.g. <code>.csv</code>, <code>.tsv</code>, etc.) and keep working with it elsewhere with different tools:</p>
</div>
<div id="plot-correlation-network" class="section level2">
<h2 class="hasAnchor">
<a href="#plot-correlation-network" class="anchor"></a>Plot correlation network</h2>
<p>Given a module we can plot the correlation network between Biological Features (genes) by looking at the Pearson correlation coefficient of the module’s Sample Sets (contrast). The default threshold is 0.8, but it can be changed. Edges in orange represent anti-correlation.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/view_plot.html">view_plot</a></span><span class="op">(</span>module <span class="op">=</span> <span class="va">module_1</span>, 
          normalization <span class="op">=</span> <span class="st">"limma"</span>,
          plotType <span class="op">=</span> <span class="st">"module_coexpression_network"</span>, 
          threshold <span class="op">=</span> <span class="fl">0.8</span><span class="op">)</span></code></pre></div>
</div>
<div id="automatically-expand-a-module" class="section level2">
<h2 class="hasAnchor">
<a href="#automatically-expand-a-module" class="anchor"></a>Automatically expand a module</h2>
<p>One of the classic way to use VESPUCCI is to start with a set of genes selected from previous analysis, have a look at the top conditions in which they show a consistent (and strong) behavior and then look for other genes that behave the same across those same conditions. The way to do it is to plot the distribution of genes that show a similar behavior of the module’s genes across the same condition in order pick up a suitable threshold.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/view_plot.html">view_plot</a></span><span class="op">(</span>module <span class="op">=</span> <span class="va">module_1</span>, normalization <span class="op">=</span> <span class="st">"limma"</span>,
          plotType <span class="op">=</span> <span class="st">"biological_features_uncentered_correlation_distribution"</span><span class="op">)</span></code></pre></div>
<p>For example, if we pick a threshold around 0.9 (looking at the top plot) we might expect to have around 15 Biological Features (genes) with a score equal or greater than that, to be added in our module. In this case let’s just pick the top 15 genes and create e new module.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">top_bf</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_module_distribution.html">plot_module_distribution</a></span><span class="op">(</span>module <span class="op">=</span> <span class="va">module_1</span>, 
                                   normalization <span class="op">=</span> <span class="st">"limma"</span>,
                                   plotType <span class="op">=</span> <span class="st">"biological_features_uncentered_correlation_distribution"</span>,
                                   getRank <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p>The new module is created by adding Biological Feature objects to the old module.</p>
<p><strong>NB</strong> Being the ranking based on the full biofeatures set we remove the duplicates ids (keeping unique ids only) and create a new model.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">module_2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_module.html">create_module</a></span><span class="op">(</span>biofeaturesNames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">module_1</span><span class="op">)</span>, <span class="va">top_bf</span><span class="op">$</span><span class="va">id</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">15</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>, 
                             normalization <span class="op">=</span> <span class="st">"limma"</span>, 
                             useIds <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p>We can have a look at the heatmap of the new module</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/view_plot.html">view_plot</a></span><span class="op">(</span>module <span class="op">=</span> <span class="va">module_2</span>, 
          normalization <span class="op">=</span> <span class="st">"limma"</span>,
          plotType <span class="op">=</span> <span class="st">"module_heatmap_expression"</span>,
          alternativeColoring <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="annotations" class="section level1">
<h1 class="hasAnchor">
<a href="#annotations" class="anchor"></a>Annotations</h1>
<p>In VESPUCCI, both Biological Feature (gene) and Sample annotations are represented using <a href="https://en.wikipedia.org/wiki/Resource_Description_Framework">RDF</a>. We can visualize annotation as a set of RDF triples.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">samples</span> <span class="op">&lt;-</span> <span class="va">my_sorted_indexes</span><span class="op">$</span><span class="va">sortedSamplesets</span>
<span class="va">samples_info</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_samples_info.html">get_samples_info</a></span><span class="op">(</span>id_In <span class="op">=</span> <span class="va">samples</span>, normalization <span class="op">=</span> <span class="st">"limma"</span>, useIds <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span><span class="op">(</span><span class="va">samples_info</span><span class="op">)</span>
<span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span><span class="op">(</span><span class="fu"><a href="../reference/get_sample_by_gsm.html">get_sample_by_gsm</a></span><span class="op">(</span>sampleName_In <span class="op">=</span>  <span class="va">samples_info</span><span class="op">$</span><span class="va">sampleName</span><span class="op">)</span><span class="op">)</span>
<span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span><span class="op">(</span><span class="fu"><a href="../reference/get_annotation_triples.html">get_annotation_triples</a></span><span class="op">(</span>ids <span class="op">=</span>  <span class="fu"><a href="../reference/get_sample_by_gsm.html">get_sample_by_gsm</a></span><span class="op">(</span>sampleName_In <span class="op">=</span>  <span class="va">samples_info</span><span class="op">$</span><span class="va">sampleName</span><span class="op">)</span><span class="op">$</span><span class="va">sampleId</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>Another way is to visualize the annotation is with RDF graph. Here we show the RDF graphs for indexed biofeatures.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"> <span class="fu"><a href="../reference/plot_annotation_network.html">plot_annotation_network</a></span><span class="op">(</span>ids <span class="op">=</span> <span class="va">my_sorted_indexes</span><span class="op">$</span><span class="va">sortedBiofeatures</span>, viewer <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></code></pre></div>
<p>Annotations can also be used to retrieve Sample or Biological Features (gene) starting from a SPARQL query. For example we might want to select all seed Sample.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sparql</span> <span class="op">=</span> <span class="st">"SELECT ?s ?p ?o WHERE { ?s &lt;http://www.w3.org/1999/02/22-rdf-syntax-ns#type&gt; &lt;http://purl.obolibrary.org/obo/NCIT_C19157&gt; . ?s &lt;http://www.w3.org/1999/02/22-rdf-syntax-ns#type&gt; &lt;http://purl.obolibrary.org/obo/PO_0009010&gt;}"</span></code></pre></div>
<p>The term <a href="http://purl.obolibrary.org/obo/NCIT_C19157">NCIT_C19157</a> indicates <strong><em>specimen</em></strong> while <a href="http://purl.obolibrary.org/obo/PO_0009010">PO_0009010</a> is the Plant Ontology term for <strong><em>seed</em></strong>. To indicate a gene, instead the term <a href="http://purl.obolibrary.org/obo/NCIT_C16612">NCIT_C16612</a> is used.</p>
<p>Here we retrieve all the samples used based on the sparql query:</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">seed_samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_sparql_annotation_triples.html">get_sparql_annotation_triples</a></span><span class="op">(</span>target <span class="op">=</span> <span class="st">"sample"</span>, 
                                              query <span class="op">=</span> <span class="va">sparql</span>,
                                              normalization <span class="op">=</span> <span class="st">"limma"</span><span class="op">)</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span></code></pre></div>
<p>Now we can collect all the Sample Sets in which these samples are used and then create a new module based on Sample Sets only leaving VESPUCCI to figure out which are the genes (Biological Feature objects) that show a consistent up or down-regulated profile.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">seed_ss</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_sampleset_by_sampleid.html">get_sampleset_by_sampleid</a></span><span class="op">(</span>normalization <span class="op">=</span> <span class="st">"limma"</span>, samples <span class="op">=</span> <span class="va">seed_samples</span><span class="op">)</span>
<span class="va">module_3</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_module.html">create_module</a></span><span class="op">(</span>samplesetNames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">seed_ss</span><span class="op">$</span><span class="va">id</span><span class="op">)</span>, useIds <span class="op">=</span> <span class="cn">TRUE</span>, normalization <span class="op">=</span> <span class="st">"limma"</span><span class="op">)</span></code></pre></div>
<p>And now we can plot the heatmap</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/view_plot.html">view_plot</a></span><span class="op">(</span>module <span class="op">=</span> <span class="va">module_3</span>, 
          normalization <span class="op">=</span> <span class="st">"limma"</span>,
          plotType <span class="op">=</span> <span class="st">"module_heatmap_expression"</span>,
          alternativeColoring <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<div id="module-summary" class="section level2">
<h2 class="hasAnchor">
<a href="#module-summary" class="anchor"></a>Module summary</h2>
<p>One way to have a general understanding about what is contained in a module, in terms of sample conditions, is to use the get the Module Description Summary Object.</p>
</div>
<div id="enrichment" class="section level2">
<h2 class="hasAnchor">
<a href="#enrichment" class="anchor"></a>Enrichment</h2>
<p>Another way to have a general understanding about which genes and conditions are present in our module we can explore the Module Enrichment Object. The enrichment is calculated using an Hypergeometric test with p-values corrected for multiple testing using Benjamini-Hochberg.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">(</span><span class="va">enrichment</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/enrich_module.html">enrich_module</a></span><span class="op">(</span>module <span class="op">=</span> <span class="va">module_1</span>, normalization <span class="op">=</span> <span class="st">"limma"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="sampleset-short-descriptions" class="section level2">
<h2 class="hasAnchor">
<a href="#sampleset-short-descriptions" class="anchor"></a>SampleSet Short Descriptions</h2>
<p>Another way to have a general descriptions of the different conditions is to use the SampleSet short description. It is calculated using Ontological Terms and tries to highlight similarities and differences between sample annotation within a SampleSet if any. In case a SampleSet is a contrast (as in limma and legacy normalizations) the description is divided by a double pipe || with similar terms between reference and test conditions on the left side and different terms on the right side.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span><span class="op">(</span><span class="fu"><a href="../reference/get_sampleset_id.html">get_sampleset_id</a></span><span class="op">(</span>id_In <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">module_1</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, normalization <span class="op">=</span> <span class="st">"limma"</span>, useIds <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="session-information" class="section level1">
<h1 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session information</h1>
<pre><code>#&gt; R version 4.0.5 (2021-03-31)
#&gt; Platform: x86_64-redhat-linux-gnu (64-bit)
#&gt; Running under: Fedora 34 (Workstation Edition)
#&gt; 
#&gt; Matrix products: default
#&gt; BLAS/LAPACK: /usr/lib64/libflexiblas.so.3.0
#&gt; 
#&gt; attached base packages:
#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     
#&gt; 
#&gt; other attached packages:
#&gt; [1] rcompass_0.0.8
#&gt; 
#&gt; loaded via a namespace (and not attached):
#&gt;  [1] SummarizedExperiment_1.20.0 xfun_0.24                  
#&gt;  [3] bslib_0.2.5.1               lattice_0.20-41            
#&gt;  [5] htmltools_0.5.1.1           stats4_4.0.5               
#&gt;  [7] yaml_2.2.1                  rlang_0.4.11               
#&gt;  [9] pkgdown_1.6.1               jquerylib_0.1.4            
#&gt; [11] httpcode_0.3.0              ghql_0.1.0                 
#&gt; [13] BiocGenerics_0.36.1         matrixStats_0.59.0         
#&gt; [15] GenomeInfoDbData_1.2.4      stringr_1.4.0              
#&gt; [17] MatrixGenerics_1.2.1        zlibbioc_1.36.0            
#&gt; [19] graphql_1.5                 ragg_1.1.3.9000            
#&gt; [21] memoise_2.0.0               evaluate_0.14              
#&gt; [23] Biobase_2.50.0              knitr_1.33                 
#&gt; [25] IRanges_2.24.1              fastmap_1.1.0              
#&gt; [27] GenomeInfoDb_1.26.7         parallel_4.0.5             
#&gt; [29] curl_4.3.2                  urltools_1.7.3             
#&gt; [31] triebeard_0.3.0             highr_0.9                  
#&gt; [33] Rcpp_1.0.6                  cachem_1.0.5               
#&gt; [35] DelayedArray_0.16.3         S4Vectors_0.28.1           
#&gt; [37] desc_1.3.0                  jsonlite_1.7.2             
#&gt; [39] XVector_0.30.0              systemfonts_1.0.2          
#&gt; [41] fs_1.5.0                    textshaping_0.3.5          
#&gt; [43] digest_0.6.27               stringi_1.6.2              
#&gt; [45] RJSONIO_1.3-1.4             GenomicRanges_1.42.0       
#&gt; [47] rprojroot_2.0.2             grid_4.0.5                 
#&gt; [49] tools_4.0.5                 bitops_1.0-7               
#&gt; [51] magrittr_2.0.1              sass_0.4.0                 
#&gt; [53] RCurl_1.98-1.3              crul_1.1.0                 
#&gt; [55] crayon_1.4.1                Matrix_1.3-2               
#&gt; [57] rmarkdown_2.9               R6_2.5.0                   
#&gt; [59] compiler_4.0.5</code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Paolo Sonego.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
