<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Use Case 02 • rcompass</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Use Case 02">
<meta property="og:description" content="rcompass">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">rcompass</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.5</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/rcompass.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/usecase01.html">Use Case 01</a>
    </li>
    <li>
      <a href="../articles/usecase02.html">Use Case 02</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/onertipaday/rcompass/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="usecase02_files/header-attrs-2.7/header-attrs.js"></script><script src="usecase02_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Use Case 02</h1>
                        <h4 class="author">Paolo Sonego</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/onertipaday/rcompass/blob/master/vignettes/usecase02.Rmd"><code>vignettes/usecase02.Rmd</code></a></small>
      <div class="hidden name"><code>usecase02.Rmd</code></div>

    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>Beyond basics</p>
    </div>
    
<p><a href="https://vespucci.readthedocs.io/">VESPUCCI</a> is the gene expression database for grapevine and we can access it via its GraphQL interface, called <a href="https://compass-.readthedocs.io/">COMPASS</a>. The <a href="https://onertipaday.github.io/rcompass/">rCOMPASS</a> package is a R package that wraps some functionalities to simplify communication with the COMPASS interface.</p>
<p>In this second exercise we will use <a href="https://vespucci.readthedocs.io/">VESPUCCI</a> together with some dedicate R packages to create an integrated workflow that goes beyond the functionality that <a href="https://vespucci.readthedocs.io/">VESPUCCI</a> alone provides.</p>
<p>The goal here is to answer the following question: <em>which are the “biological processes” that get shut down in fruit?</em> To answer this question we will need to perform a GO enrichment analysis (using, e.g., <a href="http://www.bioconductor.org/packages/release/bioc/html/topGO.html">topGO</a> package) on a subset of genes that are not expressed in fruit tissue.</p>
<p>For this case study we decide to use the TPM normalized values (databases.normalizations1=“tpm”) from VESPUCCI version 2.0 (versionNumber “2.0”). See <code><a href="../reference/get_available_compendia.html">get_available_compendia()</a></code> function to check the different normalization available.</p>
<p>Let’s start loading <code>rcompass</code>.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># devtools::install_github("onertipaday/rcompass")</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">rcompass</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; This is 'rCOMPASS' version 0.0.5 based on COMPASS version 0.8.8.</span></code></pre></div>
<div id="select-the-samples-of-interest" class="section level1">
<h1 class="hasAnchor">
<a href="#select-the-samples-of-interest" class="anchor"></a>Select the samples of interest</h1>
<div id="select-berry-samples" class="section level3">
<h3 class="hasAnchor">
<a href="#select-berry-samples" class="anchor"></a>Select berry samples</h3>
<p>Using a SPARQL query we can select only the samples annotated as berry using the following triple template where <a href="http://www.ontobee.org/ontology/PO?iri=http://purl.obolibrary.org/obo/PO_0030108">PO_0030108</a> is the <a href="https://www.ebi.ac.uk/ols/ontologies/po">Plant Ontology</a> term for <strong><em>berry fruit</em></strong>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sparql</span> <span class="op">=</span> <span class="st">"SELECT ?s ?p ?o WHERE { ?s &lt;http://www.w3.org/1999/02/22-rdf-syntax-ns#type&gt; &lt;http://purl.obolibrary.org/obo/PO_0030108&gt;}"</span>

<span class="va">berry_samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_sparql_annotation_triples.html">get_sparql_annotation_triples</a></span><span class="op">(</span>target <span class="op">=</span> <span class="st">"sample"</span>, 
                                               normalization <span class="op">=</span> <span class="st">"tpm"</span>, 
                                               query <span class="op">=</span> <span class="va">sparql</span><span class="op">)</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span></code></pre></div>
</div>
<div id="get-the-samplesets" class="section level2">
<h2 class="hasAnchor">
<a href="#get-the-samplesets" class="anchor"></a>Get the SampleSets</h2>
<p>Now that we have the samples we need to retrieve the <strong>SampleSets</strong> objects for this specific normalization (‘tpm’ normalization).</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">berry_ss</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_sampleset_by_sampleid.html">get_sampleset_by_sampleid</a></span><span class="op">(</span>samples <span class="op">=</span> <span class="va">berry_samples</span>, normalization <span class="op">=</span> <span class="st">"tpm"</span><span class="op">)</span></code></pre></div>
</div>
<div id="get-the-genes" class="section level2">
<h2 class="hasAnchor">
<a href="#get-the-genes" class="anchor"></a>Get the genes</h2>
<p>We will need all genes</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_biofeature_id.html">get_biofeature_id</a></span><span class="op">(</span>id_In <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></code></pre></div>
</div>
<div id="create-the-module" class="section level2">
<h2 class="hasAnchor">
<a href="#create-the-module" class="anchor"></a>Create the module</h2>
<p>Now that we have both the genes (<strong>BiologicalFeatures</strong>) and the <strong>SampleSets</strong>, we can create our <strong>Module</strong> and retrieve the TPM-normalized values.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">module</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_module.html">create_module</a></span><span class="op">(</span>compendium <span class="op">=</span> <span class="st">"vespucci"</span>,
                        normalization <span class="op">=</span> <span class="st">"tpm"</span>,
                        biofeaturesNames <span class="op">=</span> <span class="va">genes</span><span class="op">$</span><span class="va">id</span>, 
                        samplesetNames <span class="op">=</span> <span class="va">berry_ss</span><span class="op">$</span><span class="va">id</span>, 
                        useIds <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">module</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">berry_ss</span><span class="op">$</span><span class="va">name</span>; <span class="fu"><a href="https://rdrr.io/r/base/row.names.html">row.names</a></span><span class="op">(</span><span class="va">module</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">genes</span><span class="op">$</span><span class="va">name</span></code></pre></div>
</div>
<div id="format-the-values" class="section level2">
<h2 class="hasAnchor">
<a href="#format-the-values" class="anchor"></a>Format the values</h2>
<p>In order to better handle the module values and get rid of NaNs, we will create a data.frame starting from our Module values (we are using log2), with genes as rows and SampleSets as columns.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">df_base</span> <span class="op">&lt;-</span> <span class="va">module</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowSums</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">module</span><span class="op">)</span><span class="op">)</span> <span class="op">!=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span><span class="op">(</span><span class="va">module</span><span class="op">)</span>, <span class="op">]</span> <span class="co"># remove rows with only NAs from the matrix</span>
<span class="va">df</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html">log2</a></span><span class="op">(</span><span class="va">df_base</span> <span class="op">+</span> <span class="fl">0.01</span><span class="op">)</span>
<span class="va">i</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">df</span><span class="op">)</span>, arr.ind <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="va">df</span><span class="op">[</span><span class="va">i</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowMeans</a></span><span class="op">(</span><span class="va">df</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">[</span><span class="va">i</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span><span class="op">]</span>
<span class="va">module_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/na.fail.html">na.omit</a></span><span class="op">(</span><span class="va">df</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/hist.html">hist</a></span><span class="op">(</span><span class="va">module_df</span>, breaks <span class="op">=</span> <span class="fl">100</span><span class="op">)</span></code></pre></div>
<p><img src="usecase02_files/figure-html/format_module-1.png" width="700"></p>
</div>
</div>
<div id="gaussian-mixture-model-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#gaussian-mixture-model-analysis" class="anchor"></a>Gaussian Mixture Model Analysis</h1>
<div id="fit-a-2-component-gaussian-mixture-model" class="section level2">
<h2 class="hasAnchor">
<a href="#fit-a-2-component-gaussian-mixture-model" class="anchor"></a>Fit a 2-component Gaussian Mixture Model</h2>
<p>The idea to fit a 2-component Gaussian Mixture Model comes from the <a href="https://www.embopress.org/doi/full/10.1038/msb.2011.28">following paper</a>: Hebenstreit, D., Fang, M., Gu, M., Charoensawan, V., van Oudenaarden, A., &amp; Teichmann, S. A. (2011). <em>RNA sequencing reveals two major classes of gene expression levels in metazoan cells</em>. Molecular systems biology, 7(1).</p>
</div>
<div id="normal-mixture-models-using-mixtools-package" class="section level2">
<h2 class="hasAnchor">
<a href="#normal-mixture-models-using-mixtools-package" class="anchor"></a>Normal Mixture Models using mixtools package</h2>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">mixtools</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">42</span><span class="op">)</span> 
<span class="va">gmm_fit</span> <span class="op">&lt;-</span>  <span class="fu">mixtools</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/mixtools/man/normalmixEM.html">normalmixEM</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/vector.html">as.vector</a></span><span class="op">(</span><span class="va">module_df</span><span class="op">)</span>, k <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>
<span class="co">#&gt; number of iterations= 116</span></code></pre></div>
<div id="plot-the-data-distribution-and-the-fitted-model" class="section level3">
<h3 class="hasAnchor">
<a href="#plot-the-data-distribution-and-the-fitted-model" class="anchor"></a>Plot the data distribution and the fitted model</h3>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/snoweye/EMCluster">EMCluster</a></span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://plotly-r.com">plotly</a></span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">plotmm</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/plotmm/man/plot_mm.html">plot_mm</a></span><span class="op">(</span><span class="va">gmm_fit</span>, <span class="fl">2</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Berry samples Log2-TPM distribution"</span>,
                subtitle <span class="op">=</span> <span class="st">"Mixtools Object"</span>, 
                x <span class="op">=</span> <span class="st">"log2(TPM)"</span>,
                y <span class="op">=</span> <span class="st">"Density(normalized)"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">theme_minimal</span><span class="op">(</span><span class="op">)</span>
<span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/plotmm/man/plot_cut_point.html">plot_cut_point</a></span><span class="op">(</span><span class="va">gmm_fit</span>, plot <span class="op">=</span> <span class="cn">TRUE</span>, color <span class="op">=</span> <span class="st">"amerika"</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Berry samples Log2-TPM distribution"</span>,
                subtitle <span class="op">=</span> <span class="st">"Mixtools Object"</span>, 
                x <span class="op">=</span> <span class="st">"log2(TPM)"</span>,
                y <span class="op">=</span> <span class="st">"Density"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">theme_minimal</span><span class="op">(</span><span class="op">)</span>
<span class="fu"><a href="https://docs.ropensci.org/plotly/reference/ggplotly.html">ggplotly</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span></code></pre></div>
</div>
<div id="choose-a-cutoff-for-non-expressed-genes" class="section level3">
<h3 class="hasAnchor">
<a href="#choose-a-cutoff-for-non-expressed-genes" class="anchor"></a>Choose a cutoff for non-expressed genes</h3>
<p>From the plot we can (arbitrarily) choose a log2-TPM cutoff value to distinguish the portion of expressed genes from the non-expressed genes. We chose to set this cutoff at 1, that is roughly the median value of the lowly expressed component.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">plotmm</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plotmm/man/plot_cut_point.html">plot_cut_point</a></span><span class="op">(</span><span class="va">gmm_fit</span>, plot <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="co"># [1] 0.8443817</span>
<span class="co"># genes with log2(tpm) &lt; 1 in all samples</span>
<span class="va">never_expressed_in_fruit</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span><span class="op">(</span><span class="va">module_df</span>, <span class="fl">1</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/all.html">all</a></span><span class="op">(</span><span class="va">x</span> <span class="op">&lt;</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span>
<span class="va">never_expressed_in_fruit_names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">never_expressed_in_fruit</span><span class="op">)</span><span class="op">[</span><span class="va">never_expressed_in_fruit</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span><span class="op">(</span><span class="va">module_df</span>, <span class="fl">1</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/all.html">all</a></span><span class="op">(</span><span class="va">x</span> <span class="op">&lt;</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="co"># percentage of genes not expressed in fruit</span>
<span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">never_expressed_in_fruit</span><span class="op">)</span><span class="op">[</span><span class="va">never_expressed_in_fruit</span><span class="op">]</span><span class="op">)</span><span class="op">/</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">module_df</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span> <span class="op">*</span> <span class="fl">100</span>, <span class="st">"%"</span><span class="op">)</span></code></pre></div>
</div>
</div>
</div>
<div id="go-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#go-analysis" class="anchor"></a>GO Analysis</h1>
<div id="retrieve-the-go-annotation" class="section level2">
<h2 class="hasAnchor">
<a href="#retrieve-the-go-annotation" class="anchor"></a>Retrieve the GO annotation</h2>
<p>Now it’s time to retrieve all the GO terms for our genes and see if in the subset of genes never expressed in fruit there are GO terms that are over-represented compared to the full annotation.</p>
<p>We might want to use the Annotation class to retrieve the annotation for all our genes filtering out the GO terms, but to avoid the overhead given by the large number of genes to use, it is better to perform a lower level operation and go directly with a SPARQL query for the annotation terms we are interested in.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"> <span class="fu"><a href="../reference/get_annotation_triples.html">get_annotation_triples</a></span><span class="op">(</span>ids <span class="op">=</span> <span class="va">genes</span><span class="op">$</span><span class="va">id</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span><span class="op">)</span>
<span class="co">#&gt;       [,1]                [,2]         [,3]                            </span>
<span class="co">#&gt;  [1,] "VIT_10s0116g00040" "Gene Alias" "Vv10s0116g00040"               </span>
<span class="co">#&gt;  [2,] "GO_0006810"        "name"       "transport"                     </span>
<span class="co">#&gt;  [3,] "VIT_10s0116g00060" "Annotation" "GO_0006810"                    </span>
<span class="co">#&gt;  [4,] "GO_0019684"        "name"       "photosynthesis, light reaction"</span>
<span class="co">#&gt;  [5,] "VIT_10s0116g00060" "Annotation" "GO_0019684"                    </span>
<span class="co">#&gt;  [6,] "GO_0055114"        "name"       "oxidation-reduction process"   </span>
<span class="co">#&gt;  [7,] "VIT_10s0116g00060" "Annotation" "GO_0055114"                    </span>
<span class="co">#&gt;  [8,] "VIT_10s0116g00060" "Gene Alias" "1.6.5.3"                       </span>
<span class="co">#&gt;  [9,] "VIT_10s0116g00060" "Gene Alias" "EC:1.6.5.3"                    </span>
<span class="co">#&gt; [10,] "VIT_10s0116g00060" "Gene Alias" "VIT_10s0116g00060"             </span>
<span class="co">#&gt; [11,] "VIT_10s0116g00060" "Gene Alias" "Vv10s0116g00060"               </span>
<span class="co">#&gt; [12,] "VIT_10s0116g00070" "Gene Alias" "E0CVJ7"                        </span>
<span class="co">#&gt; [13,] "VIT_10s0116g00070" "Gene Alias" "Vv10s0116g00070"               </span>
<span class="co">#&gt; [14,] "VIT_10s0116g00080" "Gene Alias" "B9HI95"                        </span>
<span class="co">#&gt; [15,] "VIT_10s0116g00080" "Gene Alias" "POPTR_0008s08900g"             </span>
<span class="co">#&gt; [16,] "VIT_10s0116g00080" "Gene Alias" "Vv10s0116g00080"               </span>
<span class="co">#&gt; [17,] "GO_0048544"        "name"       "recognition of pollen"         </span>
<span class="co">#&gt; [18,] "VIT_10s0116g00150" "Annotation" "GO_0048544"                    </span>
<span class="co">#&gt; [19,] "GO_0006468"        "name"       "protein phosphorylation"       </span>
<span class="co">#&gt; [20,] "VIT_10s0116g00150" "Annotation" "GO_0006468"                    </span>
<span class="co">#&gt; [21,] "VIT_10s0116g00150" "Gene Alias" "B9RXT0"                        </span>
<span class="co">#&gt; [22,] "VIT_10s0116g00150" "Gene Alias" "Vv10s0116g00150"</span></code></pre></div>
<p>So let’s avoid the previous code that we ran for only 5 genes and go with the following instead, that is a SPARQL query that will return all the triples where a biofeature is annotated (Annotation is the term NCIT_C44272) with a term starting with ‘GO_’</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sparql</span> <span class="op">=</span> <span class="st">"SELECT ?s ?o WHERE {?s &lt;http://purl.obolibrary.org/obo/NCIT_C44272&gt; ?o FILTER (strstarts(str(?o), 'GO_'))}"</span>

<span class="va">triples</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_sparql_annotation_triples.html">get_sparql_annotation_triples</a></span><span class="op">(</span>compendium <span class="op">=</span> <span class="st">"vespucci"</span>,
                                         target <span class="op">=</span> <span class="st">"biofeature"</span>, 
                                         normalization <span class="op">=</span> <span class="st">"tpm"</span>,
                                         query <span class="op">=</span> <span class="va">sparql</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">triples</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"id"</span>,<span class="st">"GOTERM"</span><span class="op">)</span>
<span class="va">triples</span><span class="op">$</span><span class="va">GOTERM</span> <span class="op">&lt;-</span> <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_replace.html">str_replace</a></span><span class="op">(</span><span class="va">triples</span><span class="op">$</span><span class="va">GOTERM</span>,<span class="st">"_"</span>,<span class="st">":"</span><span class="op">)</span> <span class="co"># get rid of the underscore '_' in the annotation terms and replace it with the ':' symbol.</span>

<span class="va">go_assoc</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">full_join</a></span><span class="op">(</span><span class="va">triples</span>, <span class="va">genes</span><span class="op">)</span>
<span class="co">#&gt; Joining, by = "id"</span></code></pre></div>
</div>
<div id="go-analysis-using-topgo" class="section level2">
<h2 class="hasAnchor">
<a href="#go-analysis-using-topgo" class="anchor"></a>GO analysis using topGO</h2>
<p>To start the GO analysis we will download the basic GO and the Plant Slim GO and create both DAGs objects.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st">"topGO"</span><span class="op">)</span><span class="op">)</span>
<span class="va">GO_plant_slim</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"GO_plant_slim.tsv"</span>, package <span class="op">=</span> <span class="st">"rcompass"</span><span class="op">)</span>, sep<span class="op">=</span><span class="st">"\t"</span>, header<span class="op">=</span><span class="cn">FALSE</span>, stringsAsFactors<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">GO_plant_slim</span><span class="op">)</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ID"</span>,<span class="st">"GOTERM"</span>,<span class="st">"GOdescription"</span>,<span class="st">"Category"</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="http://had.co.nz/plyr">"plyr"</a></span><span class="op">)</span>
<span class="va">tmp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/plyr/man/dlply.html">dlply</a></span><span class="op">(</span><span class="va">GO_plant_slim</span>,<span class="fu"><a href="https://rdrr.io/pkg/plyr/man/quoted.html">.</a></span><span class="op">(</span><span class="va">GOTERM</span><span class="op">)</span><span class="op">)</span>
<span class="va">my_GO_ann_lib</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">tmp</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">x</span>, select<span class="op">=</span><span class="fl">1</span>, drop<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span> <span class="op">}</span><span class="op">)</span> <span class="co"># drop=T ==&gt; genes in a vector of characters</span>
<span class="va">geneID2GO</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/topGO/man/annFUN.html">inverseList</a></span><span class="op">(</span><span class="va">my_GO_ann_lib</span><span class="op">)</span> <span class="co"># we need this object for the annotation since there is no vitis.db</span>
<span class="va">geneNames</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">geneID2GO</span><span class="op">)</span> <span class="co"># As Universe I select all the genes in the genome</span></code></pre></div>
<p>We perform a classical enrichment analysis by testing the over-representation of GO terms within the group of expressed genes. For the method classic each GO category is tested independently. We will look for enriched GOTERMS of the Biological Process (BP) Gene Ontology.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">myInterestingGenes</span> <span class="op">&lt;-</span> <span class="va">never_expressed_in_fruit_names</span>
<span class="va">geneList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/integer.html">as.integer</a></span><span class="op">(</span><span class="va">geneNames</span> <span class="op">%in%</span> <span class="va">myInterestingGenes</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">geneList</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">geneNames</span>
<span class="co">#table(geneList)</span>
<span class="va">GOdata.BP</span> <span class="op">&lt;-</span> <span class="fu">new</span><span class="op">(</span><span class="st">"topGOdata"</span>, ontology <span class="op">=</span> <span class="st">"BP"</span>, allGenes <span class="op">=</span> <span class="va">geneList</span>, annot <span class="op">=</span> <span class="va">annFUN.gene2GO</span>, gene2GO <span class="op">=</span> <span class="va">geneID2GO</span><span class="op">)</span>
<span class="va">test.stat</span> <span class="op">&lt;-</span> <span class="fu">new</span><span class="op">(</span><span class="st">"classicCount"</span>, testStatistic <span class="op">=</span> <span class="va">GOFisherTest</span>, name <span class="op">=</span> <span class="st">"Fisher test"</span><span class="op">)</span>
<span class="op">(</span> <span class="va">resultFisher.BP</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/topGO/man/getSigGroups.html">getSigGroups</a></span><span class="op">(</span><span class="va">GOdata.BP</span>, <span class="va">test.stat</span><span class="op">)</span> <span class="op">)</span>
<span class="va">topNodes</span> <span class="op">=</span> <span class="fl">20</span>
<span class="va">resultFis.BP</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/topGO/man/getSigGroups.html">runTest</a></span><span class="op">(</span><span class="va">GOdata.BP</span>, statistic <span class="op">=</span> <span class="st">"fisher"</span><span class="op">)</span> <span class="co"># weight01 is the default algorithm </span>
<span class="va">allRes.BP</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span> <span class="fu"><a href="https://rdrr.io/pkg/topGO/man/diagnosticMethods.html">GenTable</a></span><span class="op">(</span><span class="va">GOdata.BP</span>, classic <span class="op">=</span> <span class="va">resultFis.BP</span>, orderBy <span class="op">=</span> <span class="st">"weight"</span>, 
    ranksOf <span class="op">=</span> <span class="st">"classic"</span>, topNodes <span class="op">=</span> <span class="va">topNodes</span><span class="op">)</span>, Ontology<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">"BP"</span>, <span class="va">topNodes</span><span class="op">)</span> <span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/topGO/man/printGraph-methods.html">showSigOfNodes</a></span><span class="op">(</span><span class="va">GOdata.BP</span>, <span class="fu"><a href="https://rdrr.io/pkg/topGO/man/topGOresult-class.html">score</a></span><span class="op">(</span><span class="va">resultFis.BP</span><span class="op">)</span>, firstSigNodes <span class="op">=</span> <span class="fl">10</span>, useInfo <span class="op">=</span> <span class="st">"all"</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="gofuncr" class="section level1">
<h1 class="hasAnchor">
<a href="#gofuncr" class="anchor"></a>GOFuncR</h1>
<div id="custom-annotations" class="section level3">
<h3 class="hasAnchor">
<a href="#custom-annotations" class="anchor"></a>Custom annotations</h3>
<p>Besides using bioconductor’s annotation packages for the mapping of genes to GO-categories, it is also possible to provide the annotations directly as a dataframe with two columns: (1) genes and (2) GO-IDs (parameter annotations).</p>
<p>Take a look at the data</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## first element of go_enrich result has the stats</span>
<span class="va">stats</span> <span class="op">=</span> <span class="va">res_hyper_anno</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>
<span class="co">## top-GO categories</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">stats</span><span class="op">)</span>
<span class="co">## top GO-categories per domain</span>
<span class="fu"><a href="https://rdrr.io/r/base/by.html">by</a></span><span class="op">(</span><span class="va">stats</span>, <span class="va">stats</span><span class="op">$</span><span class="va">ontology</span>, <span class="va">head</span>, n<span class="op">=</span><span class="fl">3</span><span class="op">)</span>
<span class="co">#all valid input genes</span>
<span class="co"># res_hyper_anno[[2]]</span>
<span class="co">#The fourth element is a dataframe with the minimum p-values from the permutations, which are used to compute the FWER:</span>
<span class="co">## minimum p-values from randomsets</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">res_hyper_anno</span><span class="op">[[</span><span class="fl">4</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>
<span class="co">## hypergeometric test</span>
<span class="co"># top_gos_hyper = res_hyper_anno[[1]][1:5, 'node_id']</span>
<span class="co"># GO-categories with a high proportion of candidate genes</span>
<span class="co"># top_gos_hyper</span>
<span class="co"># plot_anno_scores(res_hyper_anno, top_gos_hyper,annotations = go_assoc)</span></code></pre></div>
</div>
</div>
<div id="session-information" class="section level1">
<h1 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session information</h1>
<pre><code>#&gt; R version 4.0.4 (2021-02-15)
#&gt; Platform: x86_64-redhat-linux-gnu (64-bit)
#&gt; Running under: Fedora 33 (Workstation Edition)
#&gt; 
#&gt; Matrix products: default
#&gt; BLAS/LAPACK: /usr/lib64/libflexiblas.so.3.0
#&gt; 
#&gt; attached base packages:
#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     
#&gt; 
#&gt; other attached packages:
#&gt; [1] mixtools_1.2.0 rcompass_0.0.5
#&gt; 
#&gt; loaded via a namespace (and not attached):
#&gt;  [1] tidyselect_1.1.0  xfun_0.22         bslib_0.2.4       purrr_0.3.4      
#&gt;  [5] kernlab_0.9-29    splines_4.0.4     lattice_0.20-41   vctrs_0.3.7      
#&gt;  [9] generics_0.1.0    htmltools_0.5.1.1 yaml_2.2.1        utf8_1.2.1       
#&gt; [13] survival_3.2-7    rlang_0.4.10      pkgdown_1.6.1     jquerylib_0.1.3  
#&gt; [17] pillar_1.6.0      DBI_1.1.1         httpcode_0.3.0    glue_1.4.2       
#&gt; [21] ghql_0.1.0        segmented_1.3-3   lifecycle_1.0.0   stringr_1.4.0    
#&gt; [25] graphql_1.5       ragg_1.1.2        memoise_2.0.0     evaluate_0.14    
#&gt; [29] knitr_1.32        fastmap_1.1.0     curl_4.3          fansi_0.4.2      
#&gt; [33] urltools_1.7.3    triebeard_0.3.0   highr_0.9         Rcpp_1.0.6       
#&gt; [37] cachem_1.0.4      desc_1.3.0        jsonlite_1.7.2    systemfonts_1.0.1
#&gt; [41] fs_1.5.0          textshaping_0.3.3 digest_0.6.27     stringi_1.5.3    
#&gt; [45] dplyr_1.0.5       RJSONIO_1.3-1.4   rprojroot_2.0.2   grid_4.0.4       
#&gt; [49] tools_4.0.4       magrittr_2.0.1    sass_0.3.1        tibble_3.1.1     
#&gt; [53] crul_1.1.0        crayon_1.4.1      pkgconfig_2.0.3   MASS_7.3-53      
#&gt; [57] ellipsis_0.3.1    Matrix_1.3-2      assertthat_0.2.1  rmarkdown_2.7    
#&gt; [61] R6_2.5.0          compiler_4.0.4</code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Paolo Sonego.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
